// @fluentui/react-textarea/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.textareaFieldClassNames = exports.TextareaField_unstable = exports.useTextarea_unstable = exports.useTextareaStyles_unstable = exports.textareaClassNames = exports.renderTextarea_unstable = exports.Textarea = void 0;

var Textarea_1 = /*#__PURE__*/require("./Textarea");

Object.defineProperty(exports, "Textarea", {
  enumerable: true,
  get: function () {
    return Textarea_1.Textarea;
  }
});
Object.defineProperty(exports, "renderTextarea_unstable", {
  enumerable: true,
  get: function () {
    return Textarea_1.renderTextarea_unstable;
  }
});
Object.defineProperty(exports, "textareaClassNames", {
  enumerable: true,
  get: function () {
    return Textarea_1.textareaClassNames;
  }
});
Object.defineProperty(exports, "useTextareaStyles_unstable", {
  enumerable: true,
  get: function () {
    return Textarea_1.useTextareaStyles_unstable;
  }
});
Object.defineProperty(exports, "useTextarea_unstable", {
  enumerable: true,
  get: function () {
    return Textarea_1.useTextarea_unstable;
  }
});

var TextareaField_1 = /*#__PURE__*/require("./TextareaField");

Object.defineProperty(exports, "TextareaField_unstable", {
  enumerable: true,
  get: function () {
    return TextareaField_1.TextareaField;
  }
});
Object.defineProperty(exports, "textareaFieldClassNames", {
  enumerable: true,
  get: function () {
    return TextareaField_1.textareaFieldClassNames;
  }
});
