// @fluentui/example-data/lib/listItems.js
import { __spreadArray } from "tslib";
import { lorem } from './lorem';
var DATA = {
    color: ['red', 'blue', 'green', 'yellow'],
    shape: ['circle', 'square', 'triangle'],
    location: ['Seattle', 'New York', 'Chicago', 'Los Angeles', 'Portland'],
};
/** @internal */
export function createListItems(count, startIndex) {
    if (startIndex === void 0) { startIndex = 0; }
    return __spreadArray([], Array(count)).map(function (item, index) {
        var size = 150 + Math.round(Math.random() * 100);
        return {
            thumbnail: "//via.placeholder.com/" + size + "x" + size,
            key: 'item-' + (index + startIndex) + ' ' + lorem(4),
            name: lorem(5),
            description: lorem(10 + Math.round(Math.random() * 50)),
            color: _randWord(DATA.color),
            shape: _randWord(DATA.shape),
            location: _randWord(DATA.location),
            width: size,
            height: size,
        };
    });
}
/** @internal */
export function createGroups(groupCount, groupDepth, startIndex, itemsPerGroup, level, key, isCollapsed) {
    if (level === void 0) { level = 0; }
    if (key === void 0) { key = ''; }
    if (key !== '') {
        key = key + '-';
    }
    var count = Math.pow(itemsPerGroup, groupDepth);
    return __spreadArray([], Array(groupCount)).map(function (value, index) {
        return {
            count: count,
            key: 'group' + key + index,
            name: 'group ' + key + index,
            startIndex: index * count + startIndex,
            level: level,
            isCollapsed: isCollapsed,
            children: groupDepth > 1
                ? createGroups(groupCount, groupDepth - 1, index * count + startIndex, itemsPerGroup, level + 1, key + index)
                : [],
        };
    });
}
/** @internal */
export function isGroupable(key) {
    return key === 'color' || key === 'shape' || key === 'location';
}
function _randWord(array) {
    var index = Math.floor(Math.random() * array.length);
    return array[index];
}
