// @fluentui/react-button/lib-commonjs/utils/useToggleState.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useToggleState = void 0;

const React = /*#__PURE__*/require("react");

const react_utilities_1 = /*#__PURE__*/require("@fluentui/react-utilities");

function useToggleState(props, state) {
  const {
    checked,
    defaultChecked,
    disabled,
    disabledFocusable
  } = props;
  const {
    onClick,
    role
  } = state.root;
  const [checkedValue, setCheckedValue] = react_utilities_1.useControllableState({
    state: checked,
    defaultState: defaultChecked,
    initialState: false
  });
  const isCheckboxTypeRole = role === 'menuitemcheckbox' || role === 'checkbox';
  const onToggleClick = React.useCallback(ev => {
    if (!disabled && !disabledFocusable) {
      if (ev.defaultPrevented) {
        return;
      }

      setCheckedValue(!checkedValue);
    }
  }, [checkedValue, disabled, disabledFocusable, setCheckedValue]);
  return { ...state,
    checked: checkedValue,
    root: { ...state.root,
      [isCheckboxTypeRole ? 'aria-checked' : 'aria-pressed']: checkedValue,
      onClick: react_utilities_1.useEventCallback(react_utilities_1.mergeCallbacks(onClick, onToggleClick))
    }
  };
}

exports.useToggleState = useToggleState;
