// @fluentui/react-components/lib/AccessibilityScenarios/utils.js
import * as React from 'react';
const APP_TITLE = 'Accessibility Scenarios';
const APP_TITLE_SEPARATOR = ' | '; // https://storybook.js.org/addons/@storybook/addon-links does not allow opening a story in new tab
// so this is a naive attempt for opening a story in full screen

export const FullscreenLink = props => /*#__PURE__*/React.createElement("a", {
  className: "sbdocs sbdocs-a",
  href: `iframe.html?id=${props.parent}--${props.story}`,
  target: "_blank"
}, props.content);
export const ScenariosListLink = props => /*#__PURE__*/React.createElement("a", {
  className: "sbdocs sbdocs-a",
  href: `iframe.html?id=concepts-developer-accessibility-stories-list-of-scenarios--page`
}, props.children);
export const BackLink = () => /*#__PURE__*/React.createElement(ScenariosListLink, null, "Go back to main menu");
export const Scenario = ({
  pageTitle,
  children
}) => {
  React.useEffect(() => {
    document.title = pageTitle + APP_TITLE_SEPARATOR + APP_TITLE;
  }, [pageTitle]);
  return /*#__PURE__*/React.createElement("div", {
    role: "main"
  }, /*#__PURE__*/React.createElement(BackLink, null), /*#__PURE__*/React.createElement("br", null), children);
};
