// @fluentui/react-menu/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MenuPopover = exports.useMenuList_unstable = exports.useMenuListStyles_unstable = exports.useMenuListContextValues_unstable = exports.renderMenuList_unstable = exports.menuListClassNames = exports.MenuList = exports.useMenuItemRadio_unstable = exports.useMenuItemRadioStyles_unstable = exports.renderMenuItemRadio_unstable = exports.menuItemRadioClassNames = exports.MenuItemRadio = exports.useMenuItemCheckbox_unstable = exports.useMenuItemCheckboxStyles_unstable = exports.renderMenuItemCheckbox_unstable = exports.menuItemCheckboxClassNames = exports.MenuItemCheckbox = exports.useMenuItem_unstable = exports.useMenuItemStyles_unstable = exports.renderMenuItem_unstable = exports.menuItemClassNames = exports.MenuItem = exports.useMenuGroupHeader_unstable = exports.useMenuGroupHeaderStyles_unstable = exports.renderMenuGroupHeader_unstable = exports.menuGroupHeaderClassNames = exports.MenuGroupHeader = exports.useMenuGroup_unstable = exports.useMenuGroupStyles_unstable = exports.useMenuGroupContextValues_unstable = exports.renderMenuGroup_unstable = exports.menuGroupClassNames = exports.MenuGroup = exports.useMenuDivider_unstable = exports.useMenuDividerStyles_unstable = exports.renderMenuDivider_unstable = exports.menuDividerClassNames = exports.MenuDivider = exports.useMenu_unstable = exports.useMenuContextValues_unstable = exports.renderMenu_unstable = exports.Menu = exports.useMenuListContext_unstable = exports.MenuListProvider = exports.useMenuGroupContext_unstable = exports.MenuGroupContextProvider = exports.useMenuTriggerContext_unstable = exports.MenuTriggerContextProvider = exports.useMenuContext_unstable = exports.MenuProvider = void 0;
exports.useCheckmarkStyles_unstable = exports.useMenuTrigger_unstable = exports.renderMenuTrigger_unstable = exports.MenuTrigger = exports.useMenuSplitGroup_unstable = exports.useMenuSplitGroupStyles_unstable = exports.renderMenuSplitGroup_unstable = exports.menuSplitGroupClassNames = exports.MenuSplitGroup = exports.useMenuPopover_unstable = exports.useMenuPopoverStyles_unstable = exports.renderMenuPopover_unstable = exports.menuPopoverClassNames = void 0;

var menuContext_1 = /*#__PURE__*/require("./contexts/menuContext");

Object.defineProperty(exports, "MenuProvider", {
  enumerable: true,
  get: function () {
    return menuContext_1.MenuProvider;
  }
});
Object.defineProperty(exports, "useMenuContext_unstable", {
  enumerable: true,
  get: function () {
    return menuContext_1.useMenuContext_unstable;
  }
});

var menuTriggerContext_1 = /*#__PURE__*/require("./contexts/menuTriggerContext");

Object.defineProperty(exports, "MenuTriggerContextProvider", {
  enumerable: true,
  get: function () {
    return menuTriggerContext_1.MenuTriggerContextProvider;
  }
});
Object.defineProperty(exports, "useMenuTriggerContext_unstable", {
  enumerable: true,
  get: function () {
    return menuTriggerContext_1.useMenuTriggerContext_unstable;
  }
});

var menuGroupContext_1 = /*#__PURE__*/require("./contexts/menuGroupContext");

Object.defineProperty(exports, "MenuGroupContextProvider", {
  enumerable: true,
  get: function () {
    return menuGroupContext_1.MenuGroupContextProvider;
  }
});
Object.defineProperty(exports, "useMenuGroupContext_unstable", {
  enumerable: true,
  get: function () {
    return menuGroupContext_1.useMenuGroupContext_unstable;
  }
});

var menuListContext_1 = /*#__PURE__*/require("./contexts/menuListContext");

Object.defineProperty(exports, "MenuListProvider", {
  enumerable: true,
  get: function () {
    return menuListContext_1.MenuListProvider;
  }
});
Object.defineProperty(exports, "useMenuListContext_unstable", {
  enumerable: true,
  get: function () {
    return menuListContext_1.useMenuListContext_unstable;
  }
});

var Menu_1 = /*#__PURE__*/require("./Menu");

Object.defineProperty(exports, "Menu", {
  enumerable: true,
  get: function () {
    return Menu_1.Menu;
  }
});
Object.defineProperty(exports, "renderMenu_unstable", {
  enumerable: true,
  get: function () {
    return Menu_1.renderMenu_unstable;
  }
});
Object.defineProperty(exports, "useMenuContextValues_unstable", {
  enumerable: true,
  get: function () {
    return Menu_1.useMenuContextValues_unstable;
  }
});
Object.defineProperty(exports, "useMenu_unstable", {
  enumerable: true,
  get: function () {
    return Menu_1.useMenu_unstable;
  }
});

var MenuDivider_1 = /*#__PURE__*/require("./MenuDivider");

Object.defineProperty(exports, "MenuDivider", {
  enumerable: true,
  get: function () {
    return MenuDivider_1.MenuDivider;
  }
});
Object.defineProperty(exports, "menuDividerClassNames", {
  enumerable: true,
  get: function () {
    return MenuDivider_1.menuDividerClassNames;
  }
});
Object.defineProperty(exports, "renderMenuDivider_unstable", {
  enumerable: true,
  get: function () {
    return MenuDivider_1.renderMenuDivider_unstable;
  }
});
Object.defineProperty(exports, "useMenuDividerStyles_unstable", {
  enumerable: true,
  get: function () {
    return MenuDivider_1.useMenuDividerStyles_unstable;
  }
});
Object.defineProperty(exports, "useMenuDivider_unstable", {
  enumerable: true,
  get: function () {
    return MenuDivider_1.useMenuDivider_unstable;
  }
});

var MenuGroup_1 = /*#__PURE__*/require("./MenuGroup");

Object.defineProperty(exports, "MenuGroup", {
  enumerable: true,
  get: function () {
    return MenuGroup_1.MenuGroup;
  }
});
Object.defineProperty(exports, "menuGroupClassNames", {
  enumerable: true,
  get: function () {
    return MenuGroup_1.menuGroupClassNames;
  }
});
Object.defineProperty(exports, "renderMenuGroup_unstable", {
  enumerable: true,
  get: function () {
    return MenuGroup_1.renderMenuGroup_unstable;
  }
});
Object.defineProperty(exports, "useMenuGroupContextValues_unstable", {
  enumerable: true,
  get: function () {
    return MenuGroup_1.useMenuGroupContextValues_unstable;
  }
});
Object.defineProperty(exports, "useMenuGroupStyles_unstable", {
  enumerable: true,
  get: function () {
    return MenuGroup_1.useMenuGroupStyles_unstable;
  }
});
Object.defineProperty(exports, "useMenuGroup_unstable", {
  enumerable: true,
  get: function () {
    return MenuGroup_1.useMenuGroup_unstable;
  }
});

var MenuGroupHeader_1 = /*#__PURE__*/require("./MenuGroupHeader");

Object.defineProperty(exports, "MenuGroupHeader", {
  enumerable: true,
  get: function () {
    return MenuGroupHeader_1.MenuGroupHeader;
  }
});
Object.defineProperty(exports, "menuGroupHeaderClassNames", {
  enumerable: true,
  get: function () {
    return MenuGroupHeader_1.menuGroupHeaderClassNames;
  }
});
Object.defineProperty(exports, "renderMenuGroupHeader_unstable", {
  enumerable: true,
  get: function () {
    return MenuGroupHeader_1.renderMenuGroupHeader_unstable;
  }
});
Object.defineProperty(exports, "useMenuGroupHeaderStyles_unstable", {
  enumerable: true,
  get: function () {
    return MenuGroupHeader_1.useMenuGroupHeaderStyles_unstable;
  }
});
Object.defineProperty(exports, "useMenuGroupHeader_unstable", {
  enumerable: true,
  get: function () {
    return MenuGroupHeader_1.useMenuGroupHeader_unstable;
  }
});

var MenuItem_1 = /*#__PURE__*/require("./MenuItem");

Object.defineProperty(exports, "MenuItem", {
  enumerable: true,
  get: function () {
    return MenuItem_1.MenuItem;
  }
});
Object.defineProperty(exports, "menuItemClassNames", {
  enumerable: true,
  get: function () {
    return MenuItem_1.menuItemClassNames;
  }
});
Object.defineProperty(exports, "renderMenuItem_unstable", {
  enumerable: true,
  get: function () {
    return MenuItem_1.renderMenuItem_unstable;
  }
});
Object.defineProperty(exports, "useMenuItemStyles_unstable", {
  enumerable: true,
  get: function () {
    return MenuItem_1.useMenuItemStyles_unstable;
  }
});
Object.defineProperty(exports, "useMenuItem_unstable", {
  enumerable: true,
  get: function () {
    return MenuItem_1.useMenuItem_unstable;
  }
});

var MenuItemCheckbox_1 = /*#__PURE__*/require("./MenuItemCheckbox");

Object.defineProperty(exports, "MenuItemCheckbox", {
  enumerable: true,
  get: function () {
    return MenuItemCheckbox_1.MenuItemCheckbox;
  }
});
Object.defineProperty(exports, "menuItemCheckboxClassNames", {
  enumerable: true,
  get: function () {
    return MenuItemCheckbox_1.menuItemCheckboxClassNames;
  }
});
Object.defineProperty(exports, "renderMenuItemCheckbox_unstable", {
  enumerable: true,
  get: function () {
    return MenuItemCheckbox_1.renderMenuItemCheckbox_unstable;
  }
});
Object.defineProperty(exports, "useMenuItemCheckboxStyles_unstable", {
  enumerable: true,
  get: function () {
    return MenuItemCheckbox_1.useMenuItemCheckboxStyles_unstable;
  }
});
Object.defineProperty(exports, "useMenuItemCheckbox_unstable", {
  enumerable: true,
  get: function () {
    return MenuItemCheckbox_1.useMenuItemCheckbox_unstable;
  }
});

var MenuItemRadio_1 = /*#__PURE__*/require("./MenuItemRadio");

Object.defineProperty(exports, "MenuItemRadio", {
  enumerable: true,
  get: function () {
    return MenuItemRadio_1.MenuItemRadio;
  }
});
Object.defineProperty(exports, "menuItemRadioClassNames", {
  enumerable: true,
  get: function () {
    return MenuItemRadio_1.menuItemRadioClassNames;
  }
});
Object.defineProperty(exports, "renderMenuItemRadio_unstable", {
  enumerable: true,
  get: function () {
    return MenuItemRadio_1.renderMenuItemRadio_unstable;
  }
});
Object.defineProperty(exports, "useMenuItemRadioStyles_unstable", {
  enumerable: true,
  get: function () {
    return MenuItemRadio_1.useMenuItemRadioStyles_unstable;
  }
});
Object.defineProperty(exports, "useMenuItemRadio_unstable", {
  enumerable: true,
  get: function () {
    return MenuItemRadio_1.useMenuItemRadio_unstable;
  }
});

var MenuList_1 = /*#__PURE__*/require("./MenuList");

Object.defineProperty(exports, "MenuList", {
  enumerable: true,
  get: function () {
    return MenuList_1.MenuList;
  }
});
Object.defineProperty(exports, "menuListClassNames", {
  enumerable: true,
  get: function () {
    return MenuList_1.menuListClassNames;
  }
});
Object.defineProperty(exports, "renderMenuList_unstable", {
  enumerable: true,
  get: function () {
    return MenuList_1.renderMenuList_unstable;
  }
});
Object.defineProperty(exports, "useMenuListContextValues_unstable", {
  enumerable: true,
  get: function () {
    return MenuList_1.useMenuListContextValues_unstable;
  }
});
Object.defineProperty(exports, "useMenuListStyles_unstable", {
  enumerable: true,
  get: function () {
    return MenuList_1.useMenuListStyles_unstable;
  }
});
Object.defineProperty(exports, "useMenuList_unstable", {
  enumerable: true,
  get: function () {
    return MenuList_1.useMenuList_unstable;
  }
});

var MenuPopover_1 = /*#__PURE__*/require("./MenuPopover");

Object.defineProperty(exports, "MenuPopover", {
  enumerable: true,
  get: function () {
    return MenuPopover_1.MenuPopover;
  }
});
Object.defineProperty(exports, "menuPopoverClassNames", {
  enumerable: true,
  get: function () {
    return MenuPopover_1.menuPopoverClassNames;
  }
});
Object.defineProperty(exports, "renderMenuPopover_unstable", {
  enumerable: true,
  get: function () {
    return MenuPopover_1.renderMenuPopover_unstable;
  }
});
Object.defineProperty(exports, "useMenuPopoverStyles_unstable", {
  enumerable: true,
  get: function () {
    return MenuPopover_1.useMenuPopoverStyles_unstable;
  }
});
Object.defineProperty(exports, "useMenuPopover_unstable", {
  enumerable: true,
  get: function () {
    return MenuPopover_1.useMenuPopover_unstable;
  }
});

var MenuSplitGroup_1 = /*#__PURE__*/require("./MenuSplitGroup");

Object.defineProperty(exports, "MenuSplitGroup", {
  enumerable: true,
  get: function () {
    return MenuSplitGroup_1.MenuSplitGroup;
  }
});
Object.defineProperty(exports, "menuSplitGroupClassNames", {
  enumerable: true,
  get: function () {
    return MenuSplitGroup_1.menuSplitGroupClassNames;
  }
});
Object.defineProperty(exports, "renderMenuSplitGroup_unstable", {
  enumerable: true,
  get: function () {
    return MenuSplitGroup_1.renderMenuSplitGroup_unstable;
  }
});
Object.defineProperty(exports, "useMenuSplitGroupStyles_unstable", {
  enumerable: true,
  get: function () {
    return MenuSplitGroup_1.useMenuSplitGroupStyles_unstable;
  }
});
Object.defineProperty(exports, "useMenuSplitGroup_unstable", {
  enumerable: true,
  get: function () {
    return MenuSplitGroup_1.useMenuSplitGroup_unstable;
  }
});

var MenuTrigger_1 = /*#__PURE__*/require("./MenuTrigger");

Object.defineProperty(exports, "MenuTrigger", {
  enumerable: true,
  get: function () {
    return MenuTrigger_1.MenuTrigger;
  }
});
Object.defineProperty(exports, "renderMenuTrigger_unstable", {
  enumerable: true,
  get: function () {
    return MenuTrigger_1.renderMenuTrigger_unstable;
  }
});
Object.defineProperty(exports, "useMenuTrigger_unstable", {
  enumerable: true,
  get: function () {
    return MenuTrigger_1.useMenuTrigger_unstable;
  }
});

var index_1 = /*#__PURE__*/require("./selectable/index");

Object.defineProperty(exports, "useCheckmarkStyles_unstable", {
  enumerable: true,
  get: function () {
    return index_1.useCheckmarkStyles_unstable;
  }
});
