// @fluentui/react-popover/lib-amd/components/Popover/renderPopover.js
define(["require", "exports", "react", "../../popoverContext"], function (require, exports, React, popoverContext_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.renderPopover_unstable = void 0;
    /**
     * Render the final JSX of Popover
     */
    var renderPopover_unstable = function (state) {
        var appearance = state.appearance, arrowRef = state.arrowRef, contentRef = state.contentRef, inline = state.inline, mountNode = state.mountNode, open = state.open, openOnContext = state.openOnContext, openOnHover = state.openOnHover, setOpen = state.setOpen, size = state.size, toggleOpen = state.toggleOpen, trapFocus = state.trapFocus, triggerRef = state.triggerRef, withArrow = state.withArrow, legacyTrapFocus = state.legacyTrapFocus;
        return (React.createElement(popoverContext_1.PopoverContext.Provider, { value: {
                appearance: appearance,
                arrowRef: arrowRef,
                contentRef: contentRef,
                inline: inline,
                mountNode: mountNode,
                open: open,
                openOnContext: openOnContext,
                openOnHover: openOnHover,
                setOpen: setOpen,
                toggleOpen: toggleOpen,
                triggerRef: triggerRef,
                size: size,
                trapFocus: trapFocus,
                legacyTrapFocus: legacyTrapFocus,
                withArrow: withArrow,
            } },
            state.popoverTrigger,
            state.open && state.popoverSurface));
    };
    exports.renderPopover_unstable = renderPopover_unstable;
});
