// @fluentui/react-positioning/lib-amd/utils/fromFloatingUIPlacement.js
define(["require", "exports", "./parseFloatingUIPlacement"], function (require, exports, parseFloatingUIPlacement_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.fromFloatingUIPlacement = void 0;
    var getPositionMap = function () { return ({
        top: 'above',
        bottom: 'below',
        right: 'after',
        left: 'before',
    }); };
    // Floating UI automatically flips alignment
    // https://github.com/floating-ui/floating-ui/issues/1563
    var getAlignmentMap = function (position) {
        if (position === 'above' || position === 'below') {
            return {
                start: 'start',
                end: 'end',
            };
        }
        return {
            start: 'top',
            end: 'bottom',
        };
    };
    /**
     * Maps Floating UI placement to positioning values
     * @see positioningHelper.test.ts for expected placement values
     */
    var fromFloatingUIPlacement = function (placement) {
        var _a = parseFloatingUIPlacement_1.parseFloatingUIPlacement(placement), side = _a.side, floatingUIAlignment = _a.alignment;
        var position = getPositionMap()[side];
        var alignment = floatingUIAlignment && getAlignmentMap(position)[floatingUIAlignment];
        return { position: position, alignment: alignment };
    };
    exports.fromFloatingUIPlacement = fromFloatingUIPlacement;
});
