// @fluentui/react-label/lib-amd/components/Label/useLabel.js
define(["require", "exports", "tslib", "@fluentui/react-utilities", "@fluentui/react-utilities"], function (require, exports, tslib_1, react_utilities_1, react_utilities_2) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useLabel_unstable = void 0;
    /**
     * Create the state required to render Label.
     *
     * The returned state can be modified with hooks such as useLabelStyles_unstable,
     * before being passed to renderLabel_unstable.
     *
     * @param props - props from this instance of Label
     * @param ref - reference to root HTMLElement of Label
     */
    var useLabel_unstable = function (props, ref) {
        var _a = props.disabled, disabled = _a === void 0 ? false : _a, _b = props.required, required = _b === void 0 ? false : _b, _c = props.weight, weight = _c === void 0 ? 'regular' : _c, _d = props.size, size = _d === void 0 ? 'medium' : _d;
        return {
            disabled: disabled,
            required: react_utilities_2.resolveShorthand(required === true ? '*' : required || undefined, {
                defaultProps: { 'aria-hidden': 'true' },
            }),
            weight: weight,
            size: size,
            components: { root: 'label', required: 'span' },
            root: react_utilities_1.getNativeElementProps('label', tslib_1.__assign({ ref: ref }, props)),
        };
    };
    exports.useLabel_unstable = useLabel_unstable;
});
