// @fluentui/react-accordion/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useAccordionPanel_unstable = exports.useAccordionPanelStyles_unstable = exports.renderAccordionPanel_unstable = exports.accordionPanelClassNames = exports.AccordionPanel = exports.useAccordionHeader_unstable = exports.useAccordionHeaderStyles_unstable = exports.useAccordionHeaderContextValues_unstable = exports.renderAccordionHeader_unstable = exports.accordionHeaderClassNames = exports.AccordionHeader = exports.useAccordionItem_unstable = exports.useAccordionItemStyles_unstable = exports.useAccordionItemContext_unstable = exports.useAccordionItemContextValues_unstable = exports.renderAccordionItem_unstable = exports.accordionItemClassNames = exports.AccordionItemProvider = exports.AccordionItem = exports.useAccordion_unstable = exports.useAccordionStyles_unstable = exports.useAccordionContextValues_unstable = exports.useAccordionContext_unstable = exports.renderAccordion_unstable = exports.accordionClassNames = exports.AccordionProvider = exports.Accordion = void 0;

var Accordion_1 = /*#__PURE__*/require("./Accordion");

Object.defineProperty(exports, "Accordion", {
  enumerable: true,
  get: function () {
    return Accordion_1.Accordion;
  }
});
Object.defineProperty(exports, "AccordionProvider", {
  enumerable: true,
  get: function () {
    return Accordion_1.AccordionProvider;
  }
});
Object.defineProperty(exports, "accordionClassNames", {
  enumerable: true,
  get: function () {
    return Accordion_1.accordionClassNames;
  }
});
Object.defineProperty(exports, "renderAccordion_unstable", {
  enumerable: true,
  get: function () {
    return Accordion_1.renderAccordion_unstable;
  }
});
Object.defineProperty(exports, "useAccordionContext_unstable", {
  enumerable: true,
  get: function () {
    return Accordion_1.useAccordionContext_unstable;
  }
});
Object.defineProperty(exports, "useAccordionContextValues_unstable", {
  enumerable: true,
  get: function () {
    return Accordion_1.useAccordionContextValues_unstable;
  }
});
Object.defineProperty(exports, "useAccordionStyles_unstable", {
  enumerable: true,
  get: function () {
    return Accordion_1.useAccordionStyles_unstable;
  }
});
Object.defineProperty(exports, "useAccordion_unstable", {
  enumerable: true,
  get: function () {
    return Accordion_1.useAccordion_unstable;
  }
});

var AccordionItem_1 = /*#__PURE__*/require("./AccordionItem");

Object.defineProperty(exports, "AccordionItem", {
  enumerable: true,
  get: function () {
    return AccordionItem_1.AccordionItem;
  }
});
Object.defineProperty(exports, "AccordionItemProvider", {
  enumerable: true,
  get: function () {
    return AccordionItem_1.AccordionItemProvider;
  }
});
Object.defineProperty(exports, "accordionItemClassNames", {
  enumerable: true,
  get: function () {
    return AccordionItem_1.accordionItemClassNames;
  }
});
Object.defineProperty(exports, "renderAccordionItem_unstable", {
  enumerable: true,
  get: function () {
    return AccordionItem_1.renderAccordionItem_unstable;
  }
});
Object.defineProperty(exports, "useAccordionItemContextValues_unstable", {
  enumerable: true,
  get: function () {
    return AccordionItem_1.useAccordionItemContextValues_unstable;
  }
});
Object.defineProperty(exports, "useAccordionItemContext_unstable", {
  enumerable: true,
  get: function () {
    return AccordionItem_1.useAccordionItemContext_unstable;
  }
});
Object.defineProperty(exports, "useAccordionItemStyles_unstable", {
  enumerable: true,
  get: function () {
    return AccordionItem_1.useAccordionItemStyles_unstable;
  }
});
Object.defineProperty(exports, "useAccordionItem_unstable", {
  enumerable: true,
  get: function () {
    return AccordionItem_1.useAccordionItem_unstable;
  }
});

var AccordionHeader_1 = /*#__PURE__*/require("./AccordionHeader");

Object.defineProperty(exports, "AccordionHeader", {
  enumerable: true,
  get: function () {
    return AccordionHeader_1.AccordionHeader;
  }
});
Object.defineProperty(exports, "accordionHeaderClassNames", {
  enumerable: true,
  get: function () {
    return AccordionHeader_1.accordionHeaderClassNames;
  }
});
Object.defineProperty(exports, "renderAccordionHeader_unstable", {
  enumerable: true,
  get: function () {
    return AccordionHeader_1.renderAccordionHeader_unstable;
  }
});
Object.defineProperty(exports, "useAccordionHeaderContextValues_unstable", {
  enumerable: true,
  get: function () {
    return AccordionHeader_1.useAccordionHeaderContextValues_unstable;
  }
});
Object.defineProperty(exports, "useAccordionHeaderStyles_unstable", {
  enumerable: true,
  get: function () {
    return AccordionHeader_1.useAccordionHeaderStyles_unstable;
  }
});
Object.defineProperty(exports, "useAccordionHeader_unstable", {
  enumerable: true,
  get: function () {
    return AccordionHeader_1.useAccordionHeader_unstable;
  }
});

var AccordionPanel_1 = /*#__PURE__*/require("./AccordionPanel");

Object.defineProperty(exports, "AccordionPanel", {
  enumerable: true,
  get: function () {
    return AccordionPanel_1.AccordionPanel;
  }
});
Object.defineProperty(exports, "accordionPanelClassNames", {
  enumerable: true,
  get: function () {
    return AccordionPanel_1.accordionPanelClassNames;
  }
});
Object.defineProperty(exports, "renderAccordionPanel_unstable", {
  enumerable: true,
  get: function () {
    return AccordionPanel_1.renderAccordionPanel_unstable;
  }
});
Object.defineProperty(exports, "useAccordionPanelStyles_unstable", {
  enumerable: true,
  get: function () {
    return AccordionPanel_1.useAccordionPanelStyles_unstable;
  }
});
Object.defineProperty(exports, "useAccordionPanel_unstable", {
  enumerable: true,
  get: function () {
    return AccordionPanel_1.useAccordionPanel_unstable;
  }
});
