// @fluentui/react-radio/lib-amd/components/RadioGroup/useRadioGroup.js
define(["require", "exports", "tslib", "@fluentui/react-utilities"], function (require, exports, tslib_1, react_utilities_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useRadioGroup_unstable = void 0;
    /**
     * Create the state required to render RadioGroup.
     *
     * The returned state can be modified with hooks such as useRadioGroupStyles_unstable,
     * before being passed to renderRadioGroup_unstable.
     *
     * @param props - props from this instance of RadioGroup
     * @param ref - reference to root HTMLElement of RadioGroup
     */
    var useRadioGroup_unstable = function (props, ref) {
        var generatedName = react_utilities_1.useId('radiogroup-');
        var _a = props.name, name = _a === void 0 ? generatedName : _a, value = props.value, defaultValue = props.defaultValue, disabled = props.disabled, _b = props.layout, layout = _b === void 0 ? 'vertical' : _b, onChange = props.onChange, required = props.required;
        return {
            layout: layout,
            name: name,
            value: value,
            defaultValue: defaultValue,
            disabled: disabled,
            required: required,
            components: {
                root: 'div',
            },
            root: tslib_1.__assign(tslib_1.__assign({ ref: ref, role: 'radiogroup' }, react_utilities_1.getNativeElementProps('div', props, /*excludedPropNames:*/ ['onChange', 'name'])), { onChange: react_utilities_1.useEventCallback(function (ev) {
                    if (onChange && ev.target instanceof HTMLInputElement && ev.target.type === 'radio') {
                        onChange(ev, { value: ev.target.value });
                    }
                }) }),
        };
    };
    exports.useRadioGroup_unstable = useRadioGroup_unstable;
});
