// @fluentui/tokens/lib-commonjs/global/typographyStyles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.typographyStyles = void 0;

const tokens_1 = /*#__PURE__*/require("../tokens");
/**
 * Global typography styles (fontSize, fontWeight, and lineHeight)
 */


exports.typographyStyles = {
  body1: {
    fontFamily: tokens_1.tokens.fontFamilyBase,
    fontSize: tokens_1.tokens.fontSizeBase300,
    fontWeight: tokens_1.tokens.fontWeightRegular,
    lineHeight: tokens_1.tokens.lineHeightBase300
  },
  body1Strong: {
    fontFamily: tokens_1.tokens.fontFamilyBase,
    fontSize: tokens_1.tokens.fontSizeBase300,
    fontWeight: tokens_1.tokens.fontWeightSemibold,
    lineHeight: tokens_1.tokens.lineHeightBase300
  },
  body1Stronger: {
    fontFamily: tokens_1.tokens.fontFamilyBase,
    fontSize: tokens_1.tokens.fontSizeBase300,
    fontWeight: tokens_1.tokens.fontWeightBold,
    lineHeight: tokens_1.tokens.lineHeightBase300
  },
  body2: {
    fontFamily: tokens_1.tokens.fontFamilyBase,
    fontSize: tokens_1.tokens.fontSizeBase400,
    fontWeight: tokens_1.tokens.fontWeightRegular,
    lineHeight: tokens_1.tokens.lineHeightBase400
  },
  caption1: {
    fontFamily: tokens_1.tokens.fontFamilyBase,
    fontSize: tokens_1.tokens.fontSizeBase200,
    fontWeight: tokens_1.tokens.fontWeightRegular,
    lineHeight: tokens_1.tokens.lineHeightBase200
  },
  caption1Strong: {
    fontFamily: tokens_1.tokens.fontFamilyBase,
    fontSize: tokens_1.tokens.fontSizeBase200,
    fontWeight: tokens_1.tokens.fontWeightSemibold,
    lineHeight: tokens_1.tokens.lineHeightBase200
  },
  caption1Stronger: {
    fontFamily: tokens_1.tokens.fontFamilyBase,
    fontSize: tokens_1.tokens.fontSizeBase200,
    fontWeight: tokens_1.tokens.fontWeightBold,
    lineHeight: tokens_1.tokens.lineHeightBase200
  },
  caption2: {
    fontFamily: tokens_1.tokens.fontFamilyBase,
    fontSize: tokens_1.tokens.fontSizeBase100,
    fontWeight: tokens_1.tokens.fontWeightRegular,
    lineHeight: tokens_1.tokens.lineHeightBase100
  },
  caption2Strong: {
    fontFamily: tokens_1.tokens.fontFamilyBase,
    fontSize: tokens_1.tokens.fontSizeBase100,
    fontWeight: tokens_1.tokens.fontWeightSemibold,
    lineHeight: tokens_1.tokens.lineHeightBase100
  },
  subtitle1: {
    fontFamily: tokens_1.tokens.fontFamilyBase,
    fontSize: tokens_1.tokens.fontSizeBase500,
    fontWeight: tokens_1.tokens.fontWeightSemibold,
    lineHeight: tokens_1.tokens.lineHeightBase500
  },
  subtitle2: {
    fontFamily: tokens_1.tokens.fontFamilyBase,
    fontSize: tokens_1.tokens.fontSizeBase400,
    fontWeight: tokens_1.tokens.fontWeightSemibold,
    lineHeight: tokens_1.tokens.lineHeightBase400
  },
  subtitle2Stronger: {
    fontFamily: tokens_1.tokens.fontFamilyBase,
    fontSize: tokens_1.tokens.fontSizeBase400,
    fontWeight: tokens_1.tokens.fontWeightBold,
    lineHeight: tokens_1.tokens.lineHeightBase400
  },
  title1: {
    fontFamily: tokens_1.tokens.fontFamilyBase,
    fontSize: tokens_1.tokens.fontSizeHero800,
    fontWeight: tokens_1.tokens.fontWeightSemibold,
    lineHeight: tokens_1.tokens.lineHeightHero800
  },
  title2: {
    fontFamily: tokens_1.tokens.fontFamilyBase,
    fontSize: tokens_1.tokens.fontSizeHero700,
    fontWeight: tokens_1.tokens.fontWeightSemibold,
    lineHeight: tokens_1.tokens.lineHeightHero700
  },
  title3: {
    fontFamily: tokens_1.tokens.fontFamilyBase,
    fontSize: tokens_1.tokens.fontSizeBase600,
    fontWeight: tokens_1.tokens.fontWeightSemibold,
    lineHeight: tokens_1.tokens.lineHeightBase600
  },
  largeTitle: {
    fontFamily: tokens_1.tokens.fontFamilyBase,
    fontSize: tokens_1.tokens.fontSizeHero900,
    fontWeight: tokens_1.tokens.fontWeightSemibold,
    lineHeight: tokens_1.tokens.lineHeightHero900
  },
  display: {
    fontFamily: tokens_1.tokens.fontFamilyBase,
    fontSize: tokens_1.tokens.fontSizeHero1000,
    fontWeight: tokens_1.tokens.fontWeightSemibold,
    lineHeight: tokens_1.tokens.lineHeightHero1000
  }
};
