// @fluentui/react-link/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useLink_unstable = exports.useLinkStyles_unstable = exports.useLinkState_unstable = exports.renderLink_unstable = exports.linkClassNames = exports.Link = void 0;

var Link_1 = /*#__PURE__*/require("./Link");

Object.defineProperty(exports, "Link", {
  enumerable: true,
  get: function () {
    return Link_1.Link;
  }
});
Object.defineProperty(exports, "linkClassNames", {
  enumerable: true,
  get: function () {
    return Link_1.linkClassNames;
  }
});
Object.defineProperty(exports, "renderLink_unstable", {
  enumerable: true,
  get: function () {
    return Link_1.renderLink_unstable;
  }
});
Object.defineProperty(exports, "useLinkState_unstable", {
  enumerable: true,
  get: function () {
    return Link_1.useLinkState_unstable;
  }
});
Object.defineProperty(exports, "useLinkStyles_unstable", {
  enumerable: true,
  get: function () {
    return Link_1.useLinkStyles_unstable;
  }
});
Object.defineProperty(exports, "useLink_unstable", {
  enumerable: true,
  get: function () {
    return Link_1.useLink_unstable;
  }
});
