// @fluentui/react-badge/lib-amd/components/Badge/useBadge.js
define(["require", "exports", "tslib", "@fluentui/react-utilities"], function (require, exports, tslib_1, react_utilities_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useBadge_unstable = void 0;
    /**
     * Returns the props and state required to render the component
     */
    var useBadge_unstable = function (props, ref) {
        var _a = props.shape, shape = _a === void 0 ? 'circular' : _a, _b = props.size, size = _b === void 0 ? 'medium' : _b, _c = props.iconPosition, iconPosition = _c === void 0 ? 'before' : _c, _d = props.appearance, appearance = _d === void 0 ? 'filled' : _d, _e = props.color, color = _e === void 0 ? 'brand' : _e;
        var state = {
            shape: shape,
            size: size,
            iconPosition: iconPosition,
            appearance: appearance,
            color: color,
            components: {
                root: 'div',
                icon: 'span',
            },
            root: react_utilities_1.getNativeElementProps('div', tslib_1.__assign({ ref: ref }, props)),
            icon: react_utilities_1.resolveShorthand(props.icon),
        };
        return state;
    };
    exports.useBadge_unstable = useBadge_unstable;
});
