// @fluentui/react-menu/lib-amd/components/MenuGroupHeader/useMenuGroupHeaderStyles.js
define(["require", "exports", "@griffel/react", "@fluentui/react-theme"], function (require, exports, react_1, react_theme_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useMenuGroupHeaderStyles_unstable = exports.menuGroupHeaderClassNames = void 0;
    exports.menuGroupHeaderClassNames = {
        root: 'fui-MenuGroupHeader',
    };
    var useStyles = react_1.makeStyles({
        root: {
            fontSize: react_theme_1.tokens.fontSizeBase200,
            color: react_theme_1.tokens.colorNeutralForeground3,
            paddingLeft: '12px',
            paddingRight: '12px',
            fontWeight: react_theme_1.tokens.fontWeightSemibold,
            height: '32px',
            display: 'flex',
            alignItems: 'center',
        },
    });
    var useMenuGroupHeaderStyles_unstable = function (state) {
        var styles = useStyles();
        state.root.className = react_1.mergeClasses(exports.menuGroupHeaderClassNames.root, styles.root, state.root.className);
        return state;
    };
    exports.useMenuGroupHeaderStyles_unstable = useMenuGroupHeaderStyles_unstable;
});
