// @fluentui/react-utilities/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.isFluentTrigger = exports.getTriggerChild = exports.applyTriggerPropsToChildren = exports.isInteractiveHTMLElement = exports.isHTMLElement = exports.mergeCallbacks = exports.getRTLSafeKey = exports.getPartitionedNativeProps = exports.getNativeElementProps = exports.clamp = exports.SSRProvider = exports.useIsSSR = exports.canUseDOM = exports.useTimeout = exports.usePrevious = exports.useOnScrollOutside = exports.useOnClickOutside = exports.useMergedRefs = exports.useIsomorphicLayoutEffect = exports.useId = exports.useForceUpdate = exports.useFirstMount = exports.useEventCallback = exports.useControllableState = exports.resetIdsForTests = exports.isResolvedShorthand = exports.resolveShorthand = exports.getSlots = void 0;

var index_1 = /*#__PURE__*/require("./compose/index");

Object.defineProperty(exports, "getSlots", {
  enumerable: true,
  get: function () {
    return index_1.getSlots;
  }
});
Object.defineProperty(exports, "resolveShorthand", {
  enumerable: true,
  get: function () {
    return index_1.resolveShorthand;
  }
});
Object.defineProperty(exports, "isResolvedShorthand", {
  enumerable: true,
  get: function () {
    return index_1.isResolvedShorthand;
  }
});

var index_2 = /*#__PURE__*/require("./hooks/index");

Object.defineProperty(exports, "resetIdsForTests", {
  enumerable: true,
  get: function () {
    return index_2.resetIdsForTests;
  }
});
Object.defineProperty(exports, "useControllableState", {
  enumerable: true,
  get: function () {
    return index_2.useControllableState;
  }
});
Object.defineProperty(exports, "useEventCallback", {
  enumerable: true,
  get: function () {
    return index_2.useEventCallback;
  }
});
Object.defineProperty(exports, "useFirstMount", {
  enumerable: true,
  get: function () {
    return index_2.useFirstMount;
  }
});
Object.defineProperty(exports, "useForceUpdate", {
  enumerable: true,
  get: function () {
    return index_2.useForceUpdate;
  }
});
Object.defineProperty(exports, "useId", {
  enumerable: true,
  get: function () {
    return index_2.useId;
  }
});
Object.defineProperty(exports, "useIsomorphicLayoutEffect", {
  enumerable: true,
  get: function () {
    return index_2.useIsomorphicLayoutEffect;
  }
});
Object.defineProperty(exports, "useMergedRefs", {
  enumerable: true,
  get: function () {
    return index_2.useMergedRefs;
  }
});
Object.defineProperty(exports, "useOnClickOutside", {
  enumerable: true,
  get: function () {
    return index_2.useOnClickOutside;
  }
});
Object.defineProperty(exports, "useOnScrollOutside", {
  enumerable: true,
  get: function () {
    return index_2.useOnScrollOutside;
  }
});
Object.defineProperty(exports, "usePrevious", {
  enumerable: true,
  get: function () {
    return index_2.usePrevious;
  }
});
Object.defineProperty(exports, "useTimeout", {
  enumerable: true,
  get: function () {
    return index_2.useTimeout;
  }
});

var index_3 = /*#__PURE__*/require("./ssr/index");

Object.defineProperty(exports, "canUseDOM", {
  enumerable: true,
  get: function () {
    return index_3.canUseDOM;
  }
});
Object.defineProperty(exports, "useIsSSR", {
  enumerable: true,
  get: function () {
    return index_3.useIsSSR;
  }
});
Object.defineProperty(exports, "SSRProvider", {
  enumerable: true,
  get: function () {
    return index_3.SSRProvider;
  }
});

var index_4 = /*#__PURE__*/require("./utils/index");

Object.defineProperty(exports, "clamp", {
  enumerable: true,
  get: function () {
    return index_4.clamp;
  }
});
Object.defineProperty(exports, "getNativeElementProps", {
  enumerable: true,
  get: function () {
    return index_4.getNativeElementProps;
  }
});
Object.defineProperty(exports, "getPartitionedNativeProps", {
  enumerable: true,
  get: function () {
    return index_4.getPartitionedNativeProps;
  }
});
Object.defineProperty(exports, "getRTLSafeKey", {
  enumerable: true,
  get: function () {
    return index_4.getRTLSafeKey;
  }
});
Object.defineProperty(exports, "mergeCallbacks", {
  enumerable: true,
  get: function () {
    return index_4.mergeCallbacks;
  }
});
Object.defineProperty(exports, "isHTMLElement", {
  enumerable: true,
  get: function () {
    return index_4.isHTMLElement;
  }
});
Object.defineProperty(exports, "isInteractiveHTMLElement", {
  enumerable: true,
  get: function () {
    return index_4.isInteractiveHTMLElement;
  }
});

var index_5 = /*#__PURE__*/require("./trigger/index");

Object.defineProperty(exports, "applyTriggerPropsToChildren", {
  enumerable: true,
  get: function () {
    return index_5.applyTriggerPropsToChildren;
  }
});
Object.defineProperty(exports, "getTriggerChild", {
  enumerable: true,
  get: function () {
    return index_5.getTriggerChild;
  }
});
Object.defineProperty(exports, "isFluentTrigger", {
  enumerable: true,
  get: function () {
    return index_5.isFluentTrigger;
  }
});
