// @fluentui/react-menu/lib-amd/contexts/menuContext.js
define(["require", "exports", "@fluentui/react-context-selector"], function (require, exports, react_context_selector_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useMenuContext_unstable = exports.MenuProvider = exports.MenuContext = void 0;
    exports.MenuContext = react_context_selector_1.createContext(undefined);
    var menuContextDefaultValue = {
        open: false,
        setOpen: function () { return false; },
        checkedValues: {},
        onCheckedValueChange: function () { return null; },
        isSubmenu: false,
        triggerRef: { current: null },
        menuPopoverRef: { current: null },
        triggerId: '',
        openOnContext: false,
        openOnHover: false,
        hasIcons: false,
        hasCheckmarks: false,
        inline: false,
        persistOnItemClick: false,
    };
    exports.MenuProvider = exports.MenuContext.Provider;
    var useMenuContext_unstable = function (selector) {
        return react_context_selector_1.useContextSelector(exports.MenuContext, function (ctx) {
            if (ctx === void 0) { ctx = menuContextDefaultValue; }
            return selector(ctx);
        });
    };
    exports.useMenuContext_unstable = useMenuContext_unstable;
});
