// @fluentui/react-menu/lib-amd/contexts/menuListContext.js
define(["require", "exports", "@fluentui/react-context-selector"], function (require, exports, react_context_selector_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useMenuListContext_unstable = exports.MenuListProvider = exports.MenuListContext = void 0;
    exports.MenuListContext = react_context_selector_1.createContext(undefined);
    var menuListContextDefaultValue = {
        checkedValues: {},
        setFocusByFirstCharacter: function () { return null; },
        toggleCheckbox: function () { return null; },
        selectRadio: function () { return null; },
        hasIcons: false,
        hasCheckmarks: false,
    };
    exports.MenuListProvider = exports.MenuListContext.Provider;
    var useMenuListContext_unstable = function (selector) {
        return react_context_selector_1.useContextSelector(exports.MenuListContext, function (ctx) {
            if (ctx === void 0) { ctx = menuListContextDefaultValue; }
            return selector(ctx);
        });
    };
    exports.useMenuListContext_unstable = useMenuListContext_unstable;
});
