// @fluentui/react-positioning/lib/usePositioning.js
import { hide as hideMiddleware, arrow as arrowMiddleware } from '@floating-ui/dom';
import { useFluent_unstable as useFluent } from '@fluentui/react-shared-contexts';
import { canUseDOM, useIsomorphicLayoutEffect } from '@fluentui/react-utilities';
import * as React from 'react';
import { useCallbackRef, toFloatingUIPlacement, hasAutofocusFilter, hasScrollParent } from './utils';
import { shift as shiftMiddleware, flip as flipMiddleware, coverTarget as coverTargetMiddleware, maxSize as maxSizeMiddleware, offset as offsetMiddleware, intersecting as intersectingMiddleware } from './middleware';
import { createPositionManager } from './createPositionManager';
/**
 * @internal
 */

export function usePositioning(options) {
  const managerRef = React.useRef(null);
  const targetRef = React.useRef(null);
  const overrideTargetRef = React.useRef(null);
  const containerRef = React.useRef(null);
  const arrowRef = React.useRef(null);
  const {
    enabled = true
  } = options;
  const resolvePositioningOptions = usePositioningOptions(options);
  const updatePositionManager = React.useCallback(() => {
    var _a;

    if (managerRef.current) {
      managerRef.current.dispose();
    }

    managerRef.current = null;
    const target = (_a = overrideTargetRef.current) !== null && _a !== void 0 ? _a : targetRef.current;

    if (enabled && canUseDOM() && target && containerRef.current) {
      managerRef.current = createPositionManager({
        container: containerRef.current,
        target,
        arrow: arrowRef.current,
        ...resolvePositioningOptions(containerRef.current, arrowRef.current)
      });
    }
  }, [enabled, resolvePositioningOptions]);
  const setOverrideTarget = React.useCallback(target => {
    overrideTargetRef.current = target;
    updatePositionManager();
  }, [updatePositionManager]);
  React.useImperativeHandle(options.positioningRef, () => ({
    updatePosition: () => {
      var _a;

      return (_a = managerRef.current) === null || _a === void 0 ? void 0 : _a.updatePosition();
    },
    setTarget: target => {
      if (options.target && process.env.NODE_ENV !== 'production') {
        const err = new Error(); // eslint-disable-next-line no-console

        console.warn('Imperative setTarget should not be used at the same time as target option'); // eslint-disable-next-line no-console

        console.warn(err.stack);
      }

      setOverrideTarget(target);
    }
  }), [options.target, setOverrideTarget]);
  useIsomorphicLayoutEffect(() => {
    var _a;

    setOverrideTarget((_a = options.target) !== null && _a !== void 0 ? _a : null);
  }, [options.target, setOverrideTarget]);
  useIsomorphicLayoutEffect(() => {
    updatePositionManager();
  }, [updatePositionManager]);

  if (process.env.NODE_ENV !== 'production') {
    // This checked should run only in development mode
    // eslint-disable-next-line react-hooks/rules-of-hooks
    React.useEffect(() => {
      var _a;

      if (containerRef.current) {
        const contentNode = containerRef.current;
        const treeWalker = (_a = contentNode.ownerDocument) === null || _a === void 0 ? void 0 : _a.createTreeWalker(contentNode, NodeFilter.SHOW_ELEMENT, {
          acceptNode: hasAutofocusFilter
        });

        while (treeWalker.nextNode()) {
          const node = treeWalker.currentNode; // eslint-disable-next-line no-console

          console.warn('<Popper>:', node); // eslint-disable-next-line no-console

          console.warn(['<Popper>: ^ this node contains "autoFocus" prop on a React element. This can break the initial', 'positioning of an element and cause a window jump effect. This issue occurs because React polyfills', '"autoFocus" behavior to solve inconsistencies between different browsers:', 'https://github.com/facebook/react/issues/11851#issuecomment-351787078', '\n', 'However, ".focus()" in this case occurs before any other React effects will be executed', '(React.useEffect(), componentDidMount(), etc.) and we can not prevent this behavior. If you really', 'want to use "autoFocus" please add "position: fixed" to styles of the element that is wrapped by', '"Popper".', `In general, it's not recommended to use "autoFocus" as it may break accessibility aspects:`, 'https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-autofocus.md', '\n', 'We suggest to use the "trapFocus" prop on Fluent components or a catch "ref" and then use', '"ref.current.focus" in React.useEffect():', 'https://reactjs.org/docs/refs-and-the-dom.html#adding-a-ref-to-a-dom-element'].join(' '));
        }
      } // We run this check once, no need to add deps here
      // TODO: Should be rework to handle options.enabled and contentRef updates
      // eslint-disable-next-line react-hooks/exhaustive-deps

    }, []);
  }

  const setTarget = useCallbackRef(null, target => {
    if (targetRef.current !== target) {
      targetRef.current = target;
      updatePositionManager();
    }
  });
  const setContainer = useCallbackRef(null, container => {
    if (containerRef.current !== container) {
      containerRef.current = container;
      updatePositionManager();
    }
  });
  const setArrow = useCallbackRef(null, arrow => {
    if (arrowRef.current !== arrow) {
      arrowRef.current = arrow;
      updatePositionManager();
    }
  }); // Let users use callback refs so they feel like 'normal' DOM refs

  return {
    targetRef: setTarget,
    containerRef: setContainer,
    arrowRef: setArrow
  };
}

function usePositioningOptions(options) {
  const {
    align,
    arrowPadding,
    autoSize,
    coverTarget,
    flipBoundary,
    offset,
    overflowBoundary,
    pinned,
    position,
    unstable_disableTether: disableTether,
    positionFixed
  } = options;
  const {
    dir
  } = useFluent();
  const isRtl = dir === 'rtl';
  const strategy = positionFixed ? 'fixed' : 'absolute';
  return React.useCallback((container, arrow) => {
    const hasScrollableElement = hasScrollParent(container);
    const placement = toFloatingUIPlacement(align, position, isRtl);
    const middleware = [offset && offsetMiddleware(offset), coverTarget && coverTargetMiddleware(), !pinned && flipMiddleware({
      container,
      flipBoundary,
      hasScrollableElement
    }), shiftMiddleware({
      container,
      hasScrollableElement,
      overflowBoundary,
      disableTether
    }), autoSize && maxSizeMiddleware(autoSize), intersectingMiddleware(), arrow && arrowMiddleware({
      element: arrow,
      padding: arrowPadding
    }), hideMiddleware({
      strategy: 'referenceHidden'
    }), hideMiddleware({
      strategy: 'escaped'
    })].filter(Boolean);
    return {
      placement,
      middleware,
      strategy
    };
  }, [align, arrowPadding, autoSize, coverTarget, disableTether, flipBoundary, isRtl, offset, overflowBoundary, pinned, position, strategy]);
}
