// @fluentui/react-popover/lib-commonjs/components/PopoverSurface/usePopoverSurface.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.usePopoverSurface_unstable = void 0;

const react_utilities_1 = /*#__PURE__*/require("@fluentui/react-utilities");

const react_tabster_1 = /*#__PURE__*/require("@fluentui/react-tabster");

const popoverContext_1 = /*#__PURE__*/require("../../popoverContext");
/**
 * Create the state required to render PopoverSurface.
 *
 * The returned state can be modified with hooks such as usePopoverSurfaceStyles_unstable,
 * before being passed to renderPopoverSurface_unstable.
 *
 * @param props - props from this instance of PopoverSurface
 * @param ref - reference to root HTMLDivElement of PopoverSurface
 */


const usePopoverSurface_unstable = (props, ref) => {
  const contentRef = popoverContext_1.usePopoverContext_unstable(context => context.contentRef);
  const openOnHover = popoverContext_1.usePopoverContext_unstable(context => context.openOnHover);
  const setOpen = popoverContext_1.usePopoverContext_unstable(context => context.setOpen);
  const mountNode = popoverContext_1.usePopoverContext_unstable(context => context.mountNode);
  const arrowRef = popoverContext_1.usePopoverContext_unstable(context => context.arrowRef);
  const size = popoverContext_1.usePopoverContext_unstable(context => context.size);
  const withArrow = popoverContext_1.usePopoverContext_unstable(context => context.withArrow);
  const appearance = popoverContext_1.usePopoverContext_unstable(context => context.appearance);
  const trapFocus = popoverContext_1.usePopoverContext_unstable(context => context.trapFocus);
  const legacyTrapFocus = popoverContext_1.usePopoverContext_unstable(context => context.legacyTrapFocus);
  const inline = popoverContext_1.usePopoverContext_unstable(context => context.inline);
  const {
    modalAttributes
  } = react_tabster_1.useModalAttributes({
    trapFocus,
    legacyTrapFocus
  });
  const state = {
    inline,
    appearance,
    withArrow,
    size,
    arrowRef,
    mountNode,
    components: {
      root: 'div'
    },
    root: react_utilities_1.getNativeElementProps('div', {
      ref: react_utilities_1.useMergedRefs(ref, contentRef),
      role: trapFocus ? 'dialog' : 'group',
      'aria-modal': trapFocus ? true : undefined,
      ...modalAttributes,
      ...props
    })
  };
  const {
    onMouseEnter: onMouseEnterOriginal,
    onMouseLeave: onMouseLeaveOriginal,
    onKeyDown: onKeyDownOriginal
  } = state.root;

  state.root.onMouseEnter = e => {
    if (openOnHover) {
      setOpen(e, true);
    }

    onMouseEnterOriginal === null || onMouseEnterOriginal === void 0 ? void 0 : onMouseEnterOriginal(e);
  };

  state.root.onMouseLeave = e => {
    if (openOnHover) {
      setOpen(e, false);
    }

    onMouseLeaveOriginal === null || onMouseLeaveOriginal === void 0 ? void 0 : onMouseLeaveOriginal(e);
  };

  state.root.onKeyDown = e => {
    var _a; // only close if the event happened inside the current popover
    // If using a stack of inline popovers, the user should call `stopPropagation` to avoid dismissing the entire stack


    if (e.key === 'Escape' && ((_a = contentRef.current) === null || _a === void 0 ? void 0 : _a.contains(e.target))) {
      setOpen(e, false);
    }

    onKeyDownOriginal === null || onKeyDownOriginal === void 0 ? void 0 : onKeyDownOriginal(e);
  };

  return state;
};

exports.usePopoverSurface_unstable = usePopoverSurface_unstable;
