// @fluentui/react-table/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useTableCellActionsStyles_unstable = exports.tableCellActionsClassNames = exports.TableCellActions = exports.tableSelectionCellClassNames = exports.renderTableSelectionCell_unstable = exports.useTableSelectionCell_unstable = exports.useTableSelectionCellStyles_unstable = exports.TableSelectionCell = exports.useTableContext = exports.TableContextProvider = exports.renderTableHeaderCell_unstable = exports.useTableHeaderCell_unstable = exports.useTableHeaderCellStyles_unstable = exports.tableHeaderCellClassNames = exports.tableHeaderCellClassName = exports.TableHeaderCell = exports.renderTableHeader_unstable = exports.useTableHeader_unstable = exports.useTableHeaderStyles_unstable = exports.tableHeaderClassName = exports.tableHeaderClassNames = exports.TableHeader = exports.renderTable_unstable = exports.useTable_unstable = exports.useTableStyles_unstable = exports.tableClassNames = exports.tableClassName = exports.Table = exports.renderTableBody_unstable = exports.useTableBody_unstable = exports.useTableBodyStyles_unstable = exports.tableBodyClassNames = exports.tableBodyClassName = exports.TableBody = exports.renderTableRow_unstable = exports.useTableRow_unstable = exports.useTableRowStyles_unstable = exports.tableRowClassName = exports.tableRowClassNames = exports.TableRow = exports.renderTableCell_unstable = exports.useTableCell_unstable = exports.useTableCellStyles_unstable = exports.tableCellClassName = exports.tableCellClassNames = exports.TableCell = exports.createColumn = exports.useTableSort = exports.useTableSelection = exports.useTableFeatures = void 0;
exports.dataGridSelectionCellClassNames = exports.renderDataGridSelectionCell_unstable = exports.useDataGridSelectionCell_unstable = exports.useDataGridSelectionCellStyles_unstable = exports.DataGridSelectionCell = exports.renderDataGridHeaderCell_unstable = exports.useDataGridHeaderCell_unstable = exports.useDataGridHeaderCellStyles_unstable = exports.dataGridHeaderCellClassNames = exports.DataGridHeaderCell = exports.renderDataGridHeader_unstable = exports.useDataGridHeader_unstable = exports.useDataGridHeaderStyles_unstable = exports.dataGridHeaderClassNames = exports.DataGridHeader = exports.renderDataGrid_unstable = exports.useDataGrid_unstable = exports.useDataGridStyles_unstable = exports.dataGridClassNames = exports.DataGrid = exports.renderDataGridBody_unstable = exports.useDataGridBody_unstable = exports.useDataGridBodyStyles_unstable = exports.dataGridBodyClassNames = exports.DataGridBody = exports.renderDataGridRow_unstable = exports.useDataGridRow_unstable = exports.useDataGridRowStyles_unstable = exports.dataGridRowClassNames = exports.DataGridRow = exports.renderDataGridCell_unstable = exports.useDataGridCell_unstable = exports.useDataGridCellStyles_unstable = exports.dataGridCellClassNames = exports.DataGridCell = exports.renderTableCellLayout_unstable = exports.useTableCellLayout_unstable = exports.useTableCellLayoutStyles_unstable = exports.tableCellLayoutClassNames = exports.TableCellLayout = exports.renderTableCellActions_unstable = exports.useTableCellActions_unstable = void 0;

var hooks_1 = /*#__PURE__*/require("./hooks");

Object.defineProperty(exports, "useTableFeatures", {
  enumerable: true,
  get: function () {
    return hooks_1.useTableFeatures;
  }
});
Object.defineProperty(exports, "useTableSelection", {
  enumerable: true,
  get: function () {
    return hooks_1.useTableSelection;
  }
});
Object.defineProperty(exports, "useTableSort", {
  enumerable: true,
  get: function () {
    return hooks_1.useTableSort;
  }
});
Object.defineProperty(exports, "createColumn", {
  enumerable: true,
  get: function () {
    return hooks_1.createColumn;
  }
});

var TableCell_1 = /*#__PURE__*/require("./TableCell");

Object.defineProperty(exports, "TableCell", {
  enumerable: true,
  get: function () {
    return TableCell_1.TableCell;
  }
});
Object.defineProperty(exports, "tableCellClassNames", {
  enumerable: true,
  get: function () {
    return TableCell_1.tableCellClassNames;
  }
});
Object.defineProperty(exports, "tableCellClassName", {
  enumerable: true,
  get: function () {
    return TableCell_1.tableCellClassName;
  }
});
Object.defineProperty(exports, "useTableCellStyles_unstable", {
  enumerable: true,
  get: function () {
    return TableCell_1.useTableCellStyles_unstable;
  }
});
Object.defineProperty(exports, "useTableCell_unstable", {
  enumerable: true,
  get: function () {
    return TableCell_1.useTableCell_unstable;
  }
});
Object.defineProperty(exports, "renderTableCell_unstable", {
  enumerable: true,
  get: function () {
    return TableCell_1.renderTableCell_unstable;
  }
});

var TableRow_1 = /*#__PURE__*/require("./TableRow");

Object.defineProperty(exports, "TableRow", {
  enumerable: true,
  get: function () {
    return TableRow_1.TableRow;
  }
});
Object.defineProperty(exports, "tableRowClassNames", {
  enumerable: true,
  get: function () {
    return TableRow_1.tableRowClassNames;
  }
});
Object.defineProperty(exports, "tableRowClassName", {
  enumerable: true,
  get: function () {
    return TableRow_1.tableRowClassName;
  }
});
Object.defineProperty(exports, "useTableRowStyles_unstable", {
  enumerable: true,
  get: function () {
    return TableRow_1.useTableRowStyles_unstable;
  }
});
Object.defineProperty(exports, "useTableRow_unstable", {
  enumerable: true,
  get: function () {
    return TableRow_1.useTableRow_unstable;
  }
});
Object.defineProperty(exports, "renderTableRow_unstable", {
  enumerable: true,
  get: function () {
    return TableRow_1.renderTableRow_unstable;
  }
});

var TableBody_1 = /*#__PURE__*/require("./TableBody");

Object.defineProperty(exports, "TableBody", {
  enumerable: true,
  get: function () {
    return TableBody_1.TableBody;
  }
});
Object.defineProperty(exports, "tableBodyClassName", {
  enumerable: true,
  get: function () {
    return TableBody_1.tableBodyClassName;
  }
});
Object.defineProperty(exports, "tableBodyClassNames", {
  enumerable: true,
  get: function () {
    return TableBody_1.tableBodyClassNames;
  }
});
Object.defineProperty(exports, "useTableBodyStyles_unstable", {
  enumerable: true,
  get: function () {
    return TableBody_1.useTableBodyStyles_unstable;
  }
});
Object.defineProperty(exports, "useTableBody_unstable", {
  enumerable: true,
  get: function () {
    return TableBody_1.useTableBody_unstable;
  }
});
Object.defineProperty(exports, "renderTableBody_unstable", {
  enumerable: true,
  get: function () {
    return TableBody_1.renderTableBody_unstable;
  }
});

var Table_1 = /*#__PURE__*/require("./Table");

Object.defineProperty(exports, "Table", {
  enumerable: true,
  get: function () {
    return Table_1.Table;
  }
});
Object.defineProperty(exports, "tableClassName", {
  enumerable: true,
  get: function () {
    return Table_1.tableClassName;
  }
});
Object.defineProperty(exports, "tableClassNames", {
  enumerable: true,
  get: function () {
    return Table_1.tableClassNames;
  }
});
Object.defineProperty(exports, "useTableStyles_unstable", {
  enumerable: true,
  get: function () {
    return Table_1.useTableStyles_unstable;
  }
});
Object.defineProperty(exports, "useTable_unstable", {
  enumerable: true,
  get: function () {
    return Table_1.useTable_unstable;
  }
});
Object.defineProperty(exports, "renderTable_unstable", {
  enumerable: true,
  get: function () {
    return Table_1.renderTable_unstable;
  }
});

var TableHeader_1 = /*#__PURE__*/require("./TableHeader");

Object.defineProperty(exports, "TableHeader", {
  enumerable: true,
  get: function () {
    return TableHeader_1.TableHeader;
  }
});
Object.defineProperty(exports, "tableHeaderClassNames", {
  enumerable: true,
  get: function () {
    return TableHeader_1.tableHeaderClassNames;
  }
});
Object.defineProperty(exports, "tableHeaderClassName", {
  enumerable: true,
  get: function () {
    return TableHeader_1.tableHeaderClassName;
  }
});
Object.defineProperty(exports, "useTableHeaderStyles_unstable", {
  enumerable: true,
  get: function () {
    return TableHeader_1.useTableHeaderStyles_unstable;
  }
});
Object.defineProperty(exports, "useTableHeader_unstable", {
  enumerable: true,
  get: function () {
    return TableHeader_1.useTableHeader_unstable;
  }
});
Object.defineProperty(exports, "renderTableHeader_unstable", {
  enumerable: true,
  get: function () {
    return TableHeader_1.renderTableHeader_unstable;
  }
});

var TableHeaderCell_1 = /*#__PURE__*/require("./TableHeaderCell");

Object.defineProperty(exports, "TableHeaderCell", {
  enumerable: true,
  get: function () {
    return TableHeaderCell_1.TableHeaderCell;
  }
});
Object.defineProperty(exports, "tableHeaderCellClassName", {
  enumerable: true,
  get: function () {
    return TableHeaderCell_1.tableHeaderCellClassName;
  }
});
Object.defineProperty(exports, "tableHeaderCellClassNames", {
  enumerable: true,
  get: function () {
    return TableHeaderCell_1.tableHeaderCellClassNames;
  }
});
Object.defineProperty(exports, "useTableHeaderCellStyles_unstable", {
  enumerable: true,
  get: function () {
    return TableHeaderCell_1.useTableHeaderCellStyles_unstable;
  }
});
Object.defineProperty(exports, "useTableHeaderCell_unstable", {
  enumerable: true,
  get: function () {
    return TableHeaderCell_1.useTableHeaderCell_unstable;
  }
});
Object.defineProperty(exports, "renderTableHeaderCell_unstable", {
  enumerable: true,
  get: function () {
    return TableHeaderCell_1.renderTableHeaderCell_unstable;
  }
});

var tableContext_1 = /*#__PURE__*/require("./contexts/tableContext");

Object.defineProperty(exports, "TableContextProvider", {
  enumerable: true,
  get: function () {
    return tableContext_1.TableContextProvider;
  }
});
Object.defineProperty(exports, "useTableContext", {
  enumerable: true,
  get: function () {
    return tableContext_1.useTableContext;
  }
});

var TableSelectionCell_1 = /*#__PURE__*/require("./TableSelectionCell");

Object.defineProperty(exports, "TableSelectionCell", {
  enumerable: true,
  get: function () {
    return TableSelectionCell_1.TableSelectionCell;
  }
});
Object.defineProperty(exports, "useTableSelectionCellStyles_unstable", {
  enumerable: true,
  get: function () {
    return TableSelectionCell_1.useTableSelectionCellStyles_unstable;
  }
});
Object.defineProperty(exports, "useTableSelectionCell_unstable", {
  enumerable: true,
  get: function () {
    return TableSelectionCell_1.useTableSelectionCell_unstable;
  }
});
Object.defineProperty(exports, "renderTableSelectionCell_unstable", {
  enumerable: true,
  get: function () {
    return TableSelectionCell_1.renderTableSelectionCell_unstable;
  }
});
Object.defineProperty(exports, "tableSelectionCellClassNames", {
  enumerable: true,
  get: function () {
    return TableSelectionCell_1.tableSelectionCellClassNames;
  }
});

var TableCellActions_1 = /*#__PURE__*/require("./TableCellActions");

Object.defineProperty(exports, "TableCellActions", {
  enumerable: true,
  get: function () {
    return TableCellActions_1.TableCellActions;
  }
});
Object.defineProperty(exports, "tableCellActionsClassNames", {
  enumerable: true,
  get: function () {
    return TableCellActions_1.tableCellActionsClassNames;
  }
});
Object.defineProperty(exports, "useTableCellActionsStyles_unstable", {
  enumerable: true,
  get: function () {
    return TableCellActions_1.useTableCellActionsStyles_unstable;
  }
});
Object.defineProperty(exports, "useTableCellActions_unstable", {
  enumerable: true,
  get: function () {
    return TableCellActions_1.useTableCellActions_unstable;
  }
});
Object.defineProperty(exports, "renderTableCellActions_unstable", {
  enumerable: true,
  get: function () {
    return TableCellActions_1.renderTableCellActions_unstable;
  }
});

var TableCellLayout_1 = /*#__PURE__*/require("./TableCellLayout");

Object.defineProperty(exports, "TableCellLayout", {
  enumerable: true,
  get: function () {
    return TableCellLayout_1.TableCellLayout;
  }
});
Object.defineProperty(exports, "tableCellLayoutClassNames", {
  enumerable: true,
  get: function () {
    return TableCellLayout_1.tableCellLayoutClassNames;
  }
});
Object.defineProperty(exports, "useTableCellLayoutStyles_unstable", {
  enumerable: true,
  get: function () {
    return TableCellLayout_1.useTableCellLayoutStyles_unstable;
  }
});
Object.defineProperty(exports, "useTableCellLayout_unstable", {
  enumerable: true,
  get: function () {
    return TableCellLayout_1.useTableCellLayout_unstable;
  }
});
Object.defineProperty(exports, "renderTableCellLayout_unstable", {
  enumerable: true,
  get: function () {
    return TableCellLayout_1.renderTableCellLayout_unstable;
  }
});

var DataGridCell_1 = /*#__PURE__*/require("./DataGridCell");

Object.defineProperty(exports, "DataGridCell", {
  enumerable: true,
  get: function () {
    return DataGridCell_1.DataGridCell;
  }
});
Object.defineProperty(exports, "dataGridCellClassNames", {
  enumerable: true,
  get: function () {
    return DataGridCell_1.dataGridCellClassNames;
  }
});
Object.defineProperty(exports, "useDataGridCellStyles_unstable", {
  enumerable: true,
  get: function () {
    return DataGridCell_1.useDataGridCellStyles_unstable;
  }
});
Object.defineProperty(exports, "useDataGridCell_unstable", {
  enumerable: true,
  get: function () {
    return DataGridCell_1.useDataGridCell_unstable;
  }
});
Object.defineProperty(exports, "renderDataGridCell_unstable", {
  enumerable: true,
  get: function () {
    return DataGridCell_1.renderDataGridCell_unstable;
  }
});

var DataGridRow_1 = /*#__PURE__*/require("./DataGridRow");

Object.defineProperty(exports, "DataGridRow", {
  enumerable: true,
  get: function () {
    return DataGridRow_1.DataGridRow;
  }
});
Object.defineProperty(exports, "dataGridRowClassNames", {
  enumerable: true,
  get: function () {
    return DataGridRow_1.dataGridRowClassNames;
  }
});
Object.defineProperty(exports, "useDataGridRowStyles_unstable", {
  enumerable: true,
  get: function () {
    return DataGridRow_1.useDataGridRowStyles_unstable;
  }
});
Object.defineProperty(exports, "useDataGridRow_unstable", {
  enumerable: true,
  get: function () {
    return DataGridRow_1.useDataGridRow_unstable;
  }
});
Object.defineProperty(exports, "renderDataGridRow_unstable", {
  enumerable: true,
  get: function () {
    return DataGridRow_1.renderDataGridRow_unstable;
  }
});

var DataGridBody_1 = /*#__PURE__*/require("./DataGridBody");

Object.defineProperty(exports, "DataGridBody", {
  enumerable: true,
  get: function () {
    return DataGridBody_1.DataGridBody;
  }
});
Object.defineProperty(exports, "dataGridBodyClassNames", {
  enumerable: true,
  get: function () {
    return DataGridBody_1.dataGridBodyClassNames;
  }
});
Object.defineProperty(exports, "useDataGridBodyStyles_unstable", {
  enumerable: true,
  get: function () {
    return DataGridBody_1.useDataGridBodyStyles_unstable;
  }
});
Object.defineProperty(exports, "useDataGridBody_unstable", {
  enumerable: true,
  get: function () {
    return DataGridBody_1.useDataGridBody_unstable;
  }
});
Object.defineProperty(exports, "renderDataGridBody_unstable", {
  enumerable: true,
  get: function () {
    return DataGridBody_1.renderDataGridBody_unstable;
  }
});

var DataGrid_1 = /*#__PURE__*/require("./DataGrid");

Object.defineProperty(exports, "DataGrid", {
  enumerable: true,
  get: function () {
    return DataGrid_1.DataGrid;
  }
});
Object.defineProperty(exports, "dataGridClassNames", {
  enumerable: true,
  get: function () {
    return DataGrid_1.dataGridClassNames;
  }
});
Object.defineProperty(exports, "useDataGridStyles_unstable", {
  enumerable: true,
  get: function () {
    return DataGrid_1.useDataGridStyles_unstable;
  }
});
Object.defineProperty(exports, "useDataGrid_unstable", {
  enumerable: true,
  get: function () {
    return DataGrid_1.useDataGrid_unstable;
  }
});
Object.defineProperty(exports, "renderDataGrid_unstable", {
  enumerable: true,
  get: function () {
    return DataGrid_1.renderDataGrid_unstable;
  }
});

var DataGridHeader_1 = /*#__PURE__*/require("./DataGridHeader");

Object.defineProperty(exports, "DataGridHeader", {
  enumerable: true,
  get: function () {
    return DataGridHeader_1.DataGridHeader;
  }
});
Object.defineProperty(exports, "dataGridHeaderClassNames", {
  enumerable: true,
  get: function () {
    return DataGridHeader_1.dataGridHeaderClassNames;
  }
});
Object.defineProperty(exports, "useDataGridHeaderStyles_unstable", {
  enumerable: true,
  get: function () {
    return DataGridHeader_1.useDataGridHeaderStyles_unstable;
  }
});
Object.defineProperty(exports, "useDataGridHeader_unstable", {
  enumerable: true,
  get: function () {
    return DataGridHeader_1.useDataGridHeader_unstable;
  }
});
Object.defineProperty(exports, "renderDataGridHeader_unstable", {
  enumerable: true,
  get: function () {
    return DataGridHeader_1.renderDataGridHeader_unstable;
  }
});

var DataGridHeaderCell_1 = /*#__PURE__*/require("./DataGridHeaderCell");

Object.defineProperty(exports, "DataGridHeaderCell", {
  enumerable: true,
  get: function () {
    return DataGridHeaderCell_1.DataGridHeaderCell;
  }
});
Object.defineProperty(exports, "dataGridHeaderCellClassNames", {
  enumerable: true,
  get: function () {
    return DataGridHeaderCell_1.dataGridHeaderCellClassNames;
  }
});
Object.defineProperty(exports, "useDataGridHeaderCellStyles_unstable", {
  enumerable: true,
  get: function () {
    return DataGridHeaderCell_1.useDataGridHeaderCellStyles_unstable;
  }
});
Object.defineProperty(exports, "useDataGridHeaderCell_unstable", {
  enumerable: true,
  get: function () {
    return DataGridHeaderCell_1.useDataGridHeaderCell_unstable;
  }
});
Object.defineProperty(exports, "renderDataGridHeaderCell_unstable", {
  enumerable: true,
  get: function () {
    return DataGridHeaderCell_1.renderDataGridHeaderCell_unstable;
  }
});

var DataGridSelectionCell_1 = /*#__PURE__*/require("./DataGridSelectionCell");

Object.defineProperty(exports, "DataGridSelectionCell", {
  enumerable: true,
  get: function () {
    return DataGridSelectionCell_1.DataGridSelectionCell;
  }
});
Object.defineProperty(exports, "useDataGridSelectionCellStyles_unstable", {
  enumerable: true,
  get: function () {
    return DataGridSelectionCell_1.useDataGridSelectionCellStyles_unstable;
  }
});
Object.defineProperty(exports, "useDataGridSelectionCell_unstable", {
  enumerable: true,
  get: function () {
    return DataGridSelectionCell_1.useDataGridSelectionCell_unstable;
  }
});
Object.defineProperty(exports, "renderDataGridSelectionCell_unstable", {
  enumerable: true,
  get: function () {
    return DataGridSelectionCell_1.renderDataGridSelectionCell_unstable;
  }
});
Object.defineProperty(exports, "dataGridSelectionCellClassNames", {
  enumerable: true,
  get: function () {
    return DataGridSelectionCell_1.dataGridSelectionCellClassNames;
  }
});
