// @fluentui/tokens/lib-amd/global/colorPalette.js
define(["require", "exports", "./colors"], function (require, exports, colors_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.personaSharedColors = exports.statusSharedColors = void 0;
    exports.statusSharedColors = {
        red: colors_1.red,
        green: colors_1.green,
        darkOrange: colors_1.darkOrange,
        yellow: colors_1.yellow,
        berry: colors_1.berry,
        lightGreen: colors_1.lightGreen,
        marigold: colors_1.marigold,
    };
    exports.personaSharedColors = {
        darkRed: colors_1.darkRed,
        cranberry: colors_1.cranberry,
        pumpkin: colors_1.pumpkin,
        peach: colors_1.peach,
        gold: colors_1.gold,
        brass: colors_1.brass,
        brown: colors_1.brown,
        forest: colors_1.forest,
        seafoam: colors_1.seafoam,
        darkGreen: colors_1.darkGreen,
        lightTeal: colors_1.lightTeal,
        teal: colors_1.teal,
        steel: colors_1.steel,
        blue: colors_1.blue,
        royalBlue: colors_1.royalBlue,
        cornflower: colors_1.cornflower,
        navy: colors_1.navy,
        lavender: colors_1.lavender,
        purple: colors_1.purple,
        grape: colors_1.grape,
        lilac: colors_1.lilac,
        pink: colors_1.pink,
        magenta: colors_1.magenta,
        plum: colors_1.plum,
        beige: colors_1.beige,
        mink: colors_1.mink,
        platinum: colors_1.platinum,
        anchor: colors_1.anchor,
    };
});
