// @fluentui/tokens/lib-amd/global/spacings.js
define(["require", "exports"], function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.verticalSpacings = exports.horizontalSpacings = void 0;
    // Intentionally not exported! Use horizontalSpacings and verticalSpacings instead.
    var spacings = {
        none: '0',
        xxs: '2px',
        xs: '4px',
        sNudge: '6px',
        s: '8px',
        mNudge: '10px',
        m: '12px',
        l: '16px',
        xl: '20px',
        xxl: '24px',
        xxxl: '32px',
    };
    exports.horizontalSpacings = {
        spacingHorizontalNone: spacings.none,
        spacingHorizontalXXS: spacings.xxs,
        spacingHorizontalXS: spacings.xs,
        spacingHorizontalSNudge: spacings.sNudge,
        spacingHorizontalS: spacings.s,
        spacingHorizontalMNudge: spacings.mNudge,
        spacingHorizontalM: spacings.m,
        spacingHorizontalL: spacings.l,
        spacingHorizontalXL: spacings.xl,
        spacingHorizontalXXL: spacings.xxl,
        spacingHorizontalXXXL: spacings.xxxl,
    };
    exports.verticalSpacings = {
        spacingVerticalNone: spacings.none,
        spacingVerticalXXS: spacings.xxs,
        spacingVerticalXS: spacings.xs,
        spacingVerticalSNudge: spacings.sNudge,
        spacingVerticalS: spacings.s,
        spacingVerticalMNudge: spacings.mNudge,
        spacingVerticalM: spacings.m,
        spacingVerticalL: spacings.l,
        spacingVerticalXL: spacings.xl,
        spacingVerticalXXL: spacings.xxl,
        spacingVerticalXXXL: spacings.xxxl,
    };
});
