// @fluentui/react-dialog/lib-amd/components/Dialog/useDialogContextValues.js
define(["require", "exports"], function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useDialogContextValues_unstable = void 0;
    function useDialogContextValues_unstable(state) {
        var modalType = state.modalType, open = state.open, dialogRef = state.dialogRef, dialogTitleId = state.dialogTitleId, isNestedDialog = state.isNestedDialog, requestOpenChange = state.requestOpenChange;
        /**
         * This context is created with "@fluentui/react-context-selector",
         * there is no sense to memoize it
         */
        var dialog = {
            open: open,
            modalType: modalType,
            dialogRef: dialogRef,
            dialogTitleId: dialogTitleId,
            isNestedDialog: isNestedDialog,
            requestOpenChange: requestOpenChange,
        };
        var dialogSurface = false;
        return { dialog: dialog, dialogSurface: dialogSurface };
    }
    exports.useDialogContextValues_unstable = useDialogContextValues_unstable;
});
