// @fluentui/react-shared-contexts/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useFluent_unstable = exports.Provider_unstable = exports.useTooltipVisibility_unstable = exports.TooltipVisibilityProvider_unstable = exports.useThemeClassName_unstable = exports.ThemeClassNameProvider_unstable = exports.ThemeProvider_unstable = exports.ThemeContext_unstable = void 0;

var ThemeContext_1 = /*#__PURE__*/require("./ThemeContext");

Object.defineProperty(exports, "ThemeContext_unstable", {
  enumerable: true,
  get: function () {
    return ThemeContext_1.ThemeContext;
  }
});
Object.defineProperty(exports, "ThemeProvider_unstable", {
  enumerable: true,
  get: function () {
    return ThemeContext_1.ThemeProvider;
  }
});

var ThemeClassNameContext_1 = /*#__PURE__*/require("./ThemeClassNameContext");

Object.defineProperty(exports, "ThemeClassNameProvider_unstable", {
  enumerable: true,
  get: function () {
    return ThemeClassNameContext_1.ThemeClassNameProvider;
  }
});
Object.defineProperty(exports, "useThemeClassName_unstable", {
  enumerable: true,
  get: function () {
    return ThemeClassNameContext_1.useThemeClassName;
  }
});

var TooltipVisibilityContext_1 = /*#__PURE__*/require("./TooltipVisibilityContext");

Object.defineProperty(exports, "TooltipVisibilityProvider_unstable", {
  enumerable: true,
  get: function () {
    return TooltipVisibilityContext_1.TooltipVisibilityProvider;
  }
});
Object.defineProperty(exports, "useTooltipVisibility_unstable", {
  enumerable: true,
  get: function () {
    return TooltipVisibilityContext_1.useTooltipVisibility;
  }
});

var ProviderContext_1 = /*#__PURE__*/require("./ProviderContext");

Object.defineProperty(exports, "Provider_unstable", {
  enumerable: true,
  get: function () {
    return ProviderContext_1.Provider;
  }
});
Object.defineProperty(exports, "useFluent_unstable", {
  enumerable: true,
  get: function () {
    return ProviderContext_1.useFluent;
  }
});
