// @fluentui/react-text/lib-amd/components/Text/useText.js
define(["require", "exports", "tslib", "@fluentui/react-utilities"], function (require, exports, tslib_1, react_utilities_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useText_unstable = void 0;
    /**
     * Create the state required to render Text.
     *
     * The returned state can be modified with hooks such as useTextStyles_unstable,
     * before being passed to renderText_unstable.
     *
     * @param props - props from this instance of Text
     * @param ref - reference to root HTMLElement of Text
     */
    var useText_unstable = function (props, ref) {
        var _a;
        var wrap = props.wrap, truncate = props.truncate, block = props.block, italic = props.italic, underline = props.underline, strikethrough = props.strikethrough, size = props.size, font = props.font, weight = props.weight, align = props.align;
        var as = (_a = props.as) !== null && _a !== void 0 ? _a : 'span';
        var state = {
            align: align !== null && align !== void 0 ? align : 'start',
            block: block !== null && block !== void 0 ? block : false,
            font: font !== null && font !== void 0 ? font : 'base',
            italic: italic !== null && italic !== void 0 ? italic : false,
            size: size !== null && size !== void 0 ? size : 300,
            strikethrough: strikethrough !== null && strikethrough !== void 0 ? strikethrough : false,
            truncate: truncate !== null && truncate !== void 0 ? truncate : false,
            underline: underline !== null && underline !== void 0 ? underline : false,
            weight: weight !== null && weight !== void 0 ? weight : 'regular',
            wrap: wrap !== null && wrap !== void 0 ? wrap : true,
            components: { root: 'span' },
            root: react_utilities_1.getNativeElementProps(as, tslib_1.__assign(tslib_1.__assign({ ref: ref }, props), { as: as })),
        };
        return state;
    };
    exports.useText_unstable = useText_unstable;
});
