// @fluentui/react-textarea/lib-amd/components/Textarea/useTextarea.js
define(["require", "exports", "tslib", "@fluentui/react-utilities"], function (require, exports, tslib_1, react_utilities_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useTextarea_unstable = void 0;
    /**
     * Create the state required to render Textarea.
     *
     * The returned state can be modified with hooks such as useTextareaStyles_unstable,
     * before being passed to renderTextarea_unstable.
     *
     * @param props - props from this instance of Textarea
     * @param ref - reference to root HTMLElement of Textarea
     */
    var useTextarea_unstable = function (props, ref) {
        var _a = props.size, size = _a === void 0 ? 'medium' : _a, _b = props.appearance, appearance = _b === void 0 ? 'outline' : _b, _c = props.resize, resize = _c === void 0 ? 'none' : _c, onChange = props.onChange;
        if (process.env.NODE_ENV !== 'production' &&
            (appearance === 'filled-darker-shadow' || appearance === 'filled-lighter-shadow')) {
            // eslint-disable-next-line no-console
            console.error("The 'filled-darker-shadow' and 'filled-lighter-shadow' appearances are deprecated and will be removed in the" +
                ' future.');
        }
        var _d = react_utilities_1.useControllableState({
            state: props.value,
            defaultState: props.defaultValue,
            initialState: undefined,
        }), value = _d[0], setValue = _d[1];
        var nativeProps = react_utilities_1.getPartitionedNativeProps({
            props: props,
            primarySlotTagName: 'textarea',
            excludedPropNames: ['onChange', 'value', 'defaultValue'],
        });
        var state = {
            size: size,
            appearance: appearance,
            resize: resize,
            components: {
                root: 'span',
                textarea: 'textarea',
            },
            textarea: react_utilities_1.resolveShorthand(props.textarea, {
                required: true,
                defaultProps: tslib_1.__assign({ ref: ref }, nativeProps.primary),
            }),
            root: react_utilities_1.resolveShorthand(props.root, {
                required: true,
                defaultProps: nativeProps.root,
            }),
        };
        state.textarea.value = value;
        state.textarea.onChange = react_utilities_1.useEventCallback(function (ev) {
            var newValue = ev.target.value;
            onChange === null || onChange === void 0 ? void 0 : onChange(ev, { value: newValue });
            setValue(newValue);
        });
        return state;
    };
    exports.useTextarea_unstable = useTextarea_unstable;
});
