// @fluentui/react-positioning/lib-amd/createVirtualElementFromClick.js
define(["require", "exports"], function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.createVirtualElementFromClick = void 0;
    /**
     * Creates a virtual element based on the position of a click event
     * Can be used as a target for popper in scenarios such as context menus
     */
    function createVirtualElementFromClick(nativeEvent) {
        var left = nativeEvent.clientX;
        var top = nativeEvent.clientY;
        var right = left + 1;
        var bottom = top + 1;
        function getBoundingClientRect() {
            return {
                left: left,
                top: top,
                right: right,
                bottom: bottom,
                x: left,
                y: top,
                height: 1,
                width: 1,
            };
        }
        return {
            getBoundingClientRect: getBoundingClientRect,
        };
    }
    exports.createVirtualElementFromClick = createVirtualElementFromClick;
});
