// @fluentui/react-menu/lib-commonjs/components/MenuItemCheckbox/useMenuItemCheckbox.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useMenuItemCheckbox_unstable = void 0;

const React = /*#__PURE__*/require("react");

const react_utilities_1 = /*#__PURE__*/require("@fluentui/react-utilities");

const react_icons_1 = /*#__PURE__*/require("@fluentui/react-icons");

const menuListContext_1 = /*#__PURE__*/require("../../contexts/menuListContext");

const useMenuItem_1 = /*#__PURE__*/require("../MenuItem/useMenuItem");
/** Returns the props and state required to render the component */


const useMenuItemCheckbox_unstable = (props, ref) => {
  const toggleCheckbox = menuListContext_1.useMenuListContext_unstable(context => context.toggleCheckbox);
  const {
    name,
    value
  } = props;
  const checked = menuListContext_1.useMenuListContext_unstable(context => {
    var _a;

    const checkedItems = ((_a = context.checkedValues) === null || _a === void 0 ? void 0 : _a[name]) || [];
    return checkedItems.indexOf(value) !== -1;
  });
  const state = { ...useMenuItem_1.useMenuItem_unstable({
      role: 'menuitemcheckbox',
      persistOnClick: true,
      ...props,
      'aria-checked': checked,
      checkmark: react_utilities_1.resolveShorthand(props.checkmark, {
        defaultProps: {
          children: React.createElement(react_icons_1.Checkmark16Filled, null)
        },
        required: true
      }),
      onClick: e => {
        var _a;

        toggleCheckbox === null || toggleCheckbox === void 0 ? void 0 : toggleCheckbox(e, name, value, checked);
        (_a = props.onClick) === null || _a === void 0 ? void 0 : _a.call(props, e);
      }
    }, ref),
    name,
    value,
    checked
  };
  return state;
};

exports.useMenuItemCheckbox_unstable = useMenuItemCheckbox_unstable;
