// @fluentui/react-badge/lib-amd/components/CounterBadge/useCounterBadgeStyles.js
define(["require", "exports", "tslib", "@griffel/react", "../Badge/useBadgeStyles"], function (require, exports, tslib_1, react_1, useBadgeStyles_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useCounterBadgeStyles_unstable = exports.counterBadgeClassNames = void 0;
    exports.counterBadgeClassNames = {
        root: 'fui-CounterBadge',
        icon: 'fui-CounterBadge__icon',
    };
    var useStyles = react_1.makeStyles({
        dot: tslib_1.__assign({ minWidth: 'auto', width: '6px', height: '6px' }, react_1.shorthands.padding('0')),
        hide: {
            display: 'none',
        },
    });
    /**
     * Applies style classnames to slots
     */
    var useCounterBadgeStyles_unstable = function (state) {
        var styles = useStyles();
        state.root.className = react_1.mergeClasses(exports.counterBadgeClassNames.root, state.dot && styles.dot, !state.root.children && !state.dot && styles.hide, state.root.className);
        if (state.icon) {
            state.icon.className = react_1.mergeClasses(exports.counterBadgeClassNames.icon, state.icon.className);
        }
        return useBadgeStyles_1.useBadgeStyles_unstable(state);
    };
    exports.useCounterBadgeStyles_unstable = useCounterBadgeStyles_unstable;
});
