// @fluentui/react-avatar/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useAvatarGroupContext_unstable = exports.useAvatarContext = exports.AvatarGroupProvider = exports.AvatarContextProvider = exports.useAvatarGroupPopover_unstable = exports.useAvatarGroupPopoverStyles_unstable = exports.renderAvatarGroupPopover_unstable = exports.avatarGroupPopoverClassNames = exports.AvatarGroupPopover = exports.useAvatarGroupItem_unstable = exports.useAvatarGroupItemStyles_unstable = exports.renderAvatarGroupItem_unstable = exports.avatarGroupItemClassNames = exports.AvatarGroupItem = exports.useAvatarGroup_unstable = exports.useAvatarGroupStyles_unstable = exports.useAvatarGroupContextValues = exports.renderAvatarGroup_unstable = exports.avatarGroupClassNames = exports.AvatarGroup = exports.partitionAvatarGroupItems = exports.getInitials = exports.useAvatar_unstable = exports.useAvatarStyles_unstable = exports.renderAvatar_unstable = exports.avatarClassNames = exports.Avatar = void 0;

var Avatar_1 = /*#__PURE__*/require("./Avatar");

Object.defineProperty(exports, "Avatar", {
  enumerable: true,
  get: function () {
    return Avatar_1.Avatar;
  }
});
Object.defineProperty(exports, "avatarClassNames", {
  enumerable: true,
  get: function () {
    return Avatar_1.avatarClassNames;
  }
});
Object.defineProperty(exports, "renderAvatar_unstable", {
  enumerable: true,
  get: function () {
    return Avatar_1.renderAvatar_unstable;
  }
});
Object.defineProperty(exports, "useAvatarStyles_unstable", {
  enumerable: true,
  get: function () {
    return Avatar_1.useAvatarStyles_unstable;
  }
});
Object.defineProperty(exports, "useAvatar_unstable", {
  enumerable: true,
  get: function () {
    return Avatar_1.useAvatar_unstable;
  }
});

var index_1 = /*#__PURE__*/require("./utils/index");

Object.defineProperty(exports, "getInitials", {
  enumerable: true,
  get: function () {
    return index_1.getInitials;
  }
});
Object.defineProperty(exports, "partitionAvatarGroupItems", {
  enumerable: true,
  get: function () {
    return index_1.partitionAvatarGroupItems;
  }
});

var AvatarGroup_1 = /*#__PURE__*/require("./AvatarGroup");

Object.defineProperty(exports, "AvatarGroup", {
  enumerable: true,
  get: function () {
    return AvatarGroup_1.AvatarGroup;
  }
});
Object.defineProperty(exports, "avatarGroupClassNames", {
  enumerable: true,
  get: function () {
    return AvatarGroup_1.avatarGroupClassNames;
  }
});
Object.defineProperty(exports, "renderAvatarGroup_unstable", {
  enumerable: true,
  get: function () {
    return AvatarGroup_1.renderAvatarGroup_unstable;
  }
});
Object.defineProperty(exports, "useAvatarGroupContextValues", {
  enumerable: true,
  get: function () {
    return AvatarGroup_1.useAvatarGroupContextValues;
  }
});
Object.defineProperty(exports, "useAvatarGroupStyles_unstable", {
  enumerable: true,
  get: function () {
    return AvatarGroup_1.useAvatarGroupStyles_unstable;
  }
});
Object.defineProperty(exports, "useAvatarGroup_unstable", {
  enumerable: true,
  get: function () {
    return AvatarGroup_1.useAvatarGroup_unstable;
  }
});

var AvatarGroupItem_1 = /*#__PURE__*/require("./AvatarGroupItem");

Object.defineProperty(exports, "AvatarGroupItem", {
  enumerable: true,
  get: function () {
    return AvatarGroupItem_1.AvatarGroupItem;
  }
});
Object.defineProperty(exports, "avatarGroupItemClassNames", {
  enumerable: true,
  get: function () {
    return AvatarGroupItem_1.avatarGroupItemClassNames;
  }
});
Object.defineProperty(exports, "renderAvatarGroupItem_unstable", {
  enumerable: true,
  get: function () {
    return AvatarGroupItem_1.renderAvatarGroupItem_unstable;
  }
});
Object.defineProperty(exports, "useAvatarGroupItemStyles_unstable", {
  enumerable: true,
  get: function () {
    return AvatarGroupItem_1.useAvatarGroupItemStyles_unstable;
  }
});
Object.defineProperty(exports, "useAvatarGroupItem_unstable", {
  enumerable: true,
  get: function () {
    return AvatarGroupItem_1.useAvatarGroupItem_unstable;
  }
});

var AvatarGroupPopover_1 = /*#__PURE__*/require("./AvatarGroupPopover");

Object.defineProperty(exports, "AvatarGroupPopover", {
  enumerable: true,
  get: function () {
    return AvatarGroupPopover_1.AvatarGroupPopover;
  }
});
Object.defineProperty(exports, "avatarGroupPopoverClassNames", {
  enumerable: true,
  get: function () {
    return AvatarGroupPopover_1.avatarGroupPopoverClassNames;
  }
});
Object.defineProperty(exports, "renderAvatarGroupPopover_unstable", {
  enumerable: true,
  get: function () {
    return AvatarGroupPopover_1.renderAvatarGroupPopover_unstable;
  }
});
Object.defineProperty(exports, "useAvatarGroupPopoverStyles_unstable", {
  enumerable: true,
  get: function () {
    return AvatarGroupPopover_1.useAvatarGroupPopoverStyles_unstable;
  }
});
Object.defineProperty(exports, "useAvatarGroupPopover_unstable", {
  enumerable: true,
  get: function () {
    return AvatarGroupPopover_1.useAvatarGroupPopover_unstable;
  }
});

var index_2 = /*#__PURE__*/require("./contexts/index");

Object.defineProperty(exports, "AvatarContextProvider", {
  enumerable: true,
  get: function () {
    return index_2.AvatarContextProvider;
  }
});
Object.defineProperty(exports, "AvatarGroupProvider", {
  enumerable: true,
  get: function () {
    return index_2.AvatarGroupProvider;
  }
});
Object.defineProperty(exports, "useAvatarContext", {
  enumerable: true,
  get: function () {
    return index_2.useAvatarContext;
  }
});
Object.defineProperty(exports, "useAvatarGroupContext_unstable", {
  enumerable: true,
  get: function () {
    return index_2.useAvatarGroupContext_unstable;
  }
});
