// @fluentui/react-divider/lib-amd/components/Divider/useDivider.js
define(["require", "exports", "tslib", "@fluentui/react-utilities"], function (require, exports, tslib_1, react_utilities_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useDivider_unstable = void 0;
    /**
     * Returns the props and state required to render the component
     * @param props - User-provided props to the Divider component.
     * @param ref - User-provided ref to be passed to the Divider component.
     */
    var useDivider_unstable = function (props, ref) {
        var _a = props.alignContent, alignContent = _a === void 0 ? 'center' : _a, _b = props.appearance, appearance = _b === void 0 ? 'default' : _b, _c = props.inset, inset = _c === void 0 ? false : _c, _d = props.vertical, vertical = _d === void 0 ? false : _d, wrapper = props.wrapper;
        var dividerId = react_utilities_1.useId('divider-');
        return {
            // Props passed at the top-level
            alignContent: alignContent,
            appearance: appearance,
            inset: inset,
            vertical: vertical,
            // Slots definition
            components: {
                root: 'div',
                wrapper: 'div',
            },
            root: react_utilities_1.getNativeElementProps('div', tslib_1.__assign(tslib_1.__assign({ role: 'separator', 'aria-orientation': vertical ? 'vertical' : 'horizontal', 'aria-labelledby': props.children ? dividerId : undefined }, props), { ref: ref })),
            wrapper: react_utilities_1.resolveShorthand(wrapper, {
                required: true,
                defaultProps: {
                    id: dividerId,
                    children: props.children,
                },
            }),
        };
    };
    exports.useDivider_unstable = useDivider_unstable;
});
