// @fluentui/react-card/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useCardContext_unstable = exports.CardProvider = exports.useCardPreview_unstable = exports.useCardPreviewStyles_unstable = exports.renderCardPreview_unstable = exports.cardPreviewClassNames = exports.CardPreview = exports.useCardHeader_unstable = exports.useCardHeaderStyles_unstable = exports.renderCardHeader_unstable = exports.cardHeaderCSSVars = exports.cardHeaderClassNames = exports.CardHeader = exports.useCardFooter_unstable = exports.useCardFooterStyles_unstable = exports.renderCardFooter_unstable = exports.cardFooterClassNames = exports.CardFooter = exports.useCard_unstable = exports.useCardStyles_unstable = exports.renderCard_unstable = exports.cardCSSVars = exports.cardClassNames = exports.Card = void 0;

var Card_1 = /*#__PURE__*/require("./Card");

Object.defineProperty(exports, "Card", {
  enumerable: true,
  get: function () {
    return Card_1.Card;
  }
});
Object.defineProperty(exports, "cardClassNames", {
  enumerable: true,
  get: function () {
    return Card_1.cardClassNames;
  }
});
Object.defineProperty(exports, "cardCSSVars", {
  enumerable: true,
  get: function () {
    return Card_1.cardCSSVars;
  }
});
Object.defineProperty(exports, "renderCard_unstable", {
  enumerable: true,
  get: function () {
    return Card_1.renderCard_unstable;
  }
});
Object.defineProperty(exports, "useCardStyles_unstable", {
  enumerable: true,
  get: function () {
    return Card_1.useCardStyles_unstable;
  }
});
Object.defineProperty(exports, "useCard_unstable", {
  enumerable: true,
  get: function () {
    return Card_1.useCard_unstable;
  }
});

var CardFooter_1 = /*#__PURE__*/require("./CardFooter");

Object.defineProperty(exports, "CardFooter", {
  enumerable: true,
  get: function () {
    return CardFooter_1.CardFooter;
  }
});
Object.defineProperty(exports, "cardFooterClassNames", {
  enumerable: true,
  get: function () {
    return CardFooter_1.cardFooterClassNames;
  }
});
Object.defineProperty(exports, "renderCardFooter_unstable", {
  enumerable: true,
  get: function () {
    return CardFooter_1.renderCardFooter_unstable;
  }
});
Object.defineProperty(exports, "useCardFooterStyles_unstable", {
  enumerable: true,
  get: function () {
    return CardFooter_1.useCardFooterStyles_unstable;
  }
});
Object.defineProperty(exports, "useCardFooter_unstable", {
  enumerable: true,
  get: function () {
    return CardFooter_1.useCardFooter_unstable;
  }
});

var CardHeader_1 = /*#__PURE__*/require("./CardHeader");

Object.defineProperty(exports, "CardHeader", {
  enumerable: true,
  get: function () {
    return CardHeader_1.CardHeader;
  }
});
Object.defineProperty(exports, "cardHeaderClassNames", {
  enumerable: true,
  get: function () {
    return CardHeader_1.cardHeaderClassNames;
  }
});
Object.defineProperty(exports, "cardHeaderCSSVars", {
  enumerable: true,
  get: function () {
    return CardHeader_1.cardHeaderCSSVars;
  }
});
Object.defineProperty(exports, "renderCardHeader_unstable", {
  enumerable: true,
  get: function () {
    return CardHeader_1.renderCardHeader_unstable;
  }
});
Object.defineProperty(exports, "useCardHeaderStyles_unstable", {
  enumerable: true,
  get: function () {
    return CardHeader_1.useCardHeaderStyles_unstable;
  }
});
Object.defineProperty(exports, "useCardHeader_unstable", {
  enumerable: true,
  get: function () {
    return CardHeader_1.useCardHeader_unstable;
  }
});

var CardPreview_1 = /*#__PURE__*/require("./CardPreview");

Object.defineProperty(exports, "CardPreview", {
  enumerable: true,
  get: function () {
    return CardPreview_1.CardPreview;
  }
});
Object.defineProperty(exports, "cardPreviewClassNames", {
  enumerable: true,
  get: function () {
    return CardPreview_1.cardPreviewClassNames;
  }
});
Object.defineProperty(exports, "renderCardPreview_unstable", {
  enumerable: true,
  get: function () {
    return CardPreview_1.renderCardPreview_unstable;
  }
});
Object.defineProperty(exports, "useCardPreviewStyles_unstable", {
  enumerable: true,
  get: function () {
    return CardPreview_1.useCardPreviewStyles_unstable;
  }
});
Object.defineProperty(exports, "useCardPreview_unstable", {
  enumerable: true,
  get: function () {
    return CardPreview_1.useCardPreview_unstable;
  }
});

var Card_2 = /*#__PURE__*/require("./Card");

Object.defineProperty(exports, "CardProvider", {
  enumerable: true,
  get: function () {
    return Card_2.CardProvider;
  }
});
Object.defineProperty(exports, "useCardContext_unstable", {
  enumerable: true,
  get: function () {
    return Card_2.useCardContext_unstable;
  }
});
