// @fluentui/react-checkbox/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.checkboxFieldClassNames = exports.CheckboxField_unstable = exports.useCheckbox_unstable = exports.useCheckboxStyles_unstable = exports.renderCheckbox_unstable = exports.checkboxClassNames = exports.Checkbox = void 0;

var Checkbox_1 = /*#__PURE__*/require("./Checkbox");

Object.defineProperty(exports, "Checkbox", {
  enumerable: true,
  get: function () {
    return Checkbox_1.Checkbox;
  }
});
Object.defineProperty(exports, "checkboxClassNames", {
  enumerable: true,
  get: function () {
    return Checkbox_1.checkboxClassNames;
  }
});
Object.defineProperty(exports, "renderCheckbox_unstable", {
  enumerable: true,
  get: function () {
    return Checkbox_1.renderCheckbox_unstable;
  }
});
Object.defineProperty(exports, "useCheckboxStyles_unstable", {
  enumerable: true,
  get: function () {
    return Checkbox_1.useCheckboxStyles_unstable;
  }
});
Object.defineProperty(exports, "useCheckbox_unstable", {
  enumerable: true,
  get: function () {
    return Checkbox_1.useCheckbox_unstable;
  }
});

var CheckboxField_1 = /*#__PURE__*/require("./CheckboxField");

Object.defineProperty(exports, "CheckboxField_unstable", {
  enumerable: true,
  get: function () {
    return CheckboxField_1.CheckboxField;
  }
});
Object.defineProperty(exports, "checkboxFieldClassNames", {
  enumerable: true,
  get: function () {
    return CheckboxField_1.checkboxFieldClassNames;
  }
});
