// @fluentui/react-combobox/lib-commonjs/components/OptionGroup/useOptionGroup.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useOptionGroup_unstable = void 0;

const react_utilities_1 = /*#__PURE__*/require("@fluentui/react-utilities");
/**
 * Create the state required to render OptionGroup.
 *
 * The returned state can be modified with hooks such as useOptionGroupStyles_unstable,
 * before being passed to renderOptionGroup_unstable.
 *
 * @param props - props from this instance of OptionGroup
 * @param ref - reference to root HTMLElement of OptionGroup
 */


const useOptionGroup_unstable = (props, ref) => {
  const labelId = react_utilities_1.useId('group-label');
  const {
    label
  } = props;
  return {
    components: {
      root: 'div',
      label: 'span'
    },
    root: react_utilities_1.getNativeElementProps('div', {
      ref,
      role: 'group',
      'aria-labelledby': label ? labelId : undefined,
      ...props
    }),
    label: react_utilities_1.resolveShorthand(label, {
      defaultProps: {
        id: labelId,
        role: 'presentation'
      }
    })
  };
};

exports.useOptionGroup_unstable = useOptionGroup_unstable;
