// @fluentui/react-dialog/lib-amd/contexts/dialogContext.js
define(["require", "exports", "@fluentui/react-context-selector"], function (require, exports, react_context_selector_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useDialogContext_unstable = exports.DialogProvider = exports.DialogContext = void 0;
    var defaultContextValue = {
        open: false,
        modalType: 'modal',
        isNestedDialog: false,
        dialogRef: { current: null },
        requestOpenChange: function () {
            /* noop */
        },
    };
    // Contexts should default to undefined
    exports.DialogContext = react_context_selector_1.createContext(undefined);
    exports.DialogProvider = exports.DialogContext.Provider;
    var useDialogContext_unstable = function (selector) {
        return react_context_selector_1.useContextSelector(exports.DialogContext, function (ctx) {
            if (ctx === void 0) { ctx = defaultContextValue; }
            return selector(ctx);
        });
    };
    exports.useDialogContext_unstable = useDialogContext_unstable;
});
