// @fluentui/tokens/lib-commonjs/alias/highContrastColor.js
"use strict";
/* !!! DO NOT EDIT !!! */

/* This file has been generated by the token pipeline */

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.generateColorTokens = void 0;

const colors_1 = /*#__PURE__*/require("../global/colors");

const generateColorTokens = () => ({
  colorNeutralForeground1: colors_1.hcCanvasText,
  colorNeutralForeground1Hover: colors_1.hcHighlightText,
  colorNeutralForeground1Pressed: colors_1.hcHighlightText,
  colorNeutralForeground1Selected: colors_1.hcHighlightText,
  colorNeutralForeground2: colors_1.hcCanvasText,
  colorNeutralForeground2Hover: colors_1.hcHighlightText,
  colorNeutralForeground2Pressed: colors_1.hcHighlightText,
  colorNeutralForeground2Selected: colors_1.hcHighlightText,
  colorNeutralForeground2BrandHover: colors_1.hcHighlightText,
  colorNeutralForeground2BrandPressed: colors_1.hcHighlightText,
  colorNeutralForeground2BrandSelected: colors_1.hcHighlightText,
  colorNeutralForeground3: colors_1.hcCanvasText,
  colorNeutralForeground3Hover: colors_1.hcHighlightText,
  colorNeutralForeground3Pressed: colors_1.hcHighlightText,
  colorNeutralForeground3Selected: colors_1.hcHighlightText,
  colorNeutralForeground3BrandHover: colors_1.hcHighlightText,
  colorNeutralForeground3BrandPressed: colors_1.hcHighlightText,
  colorNeutralForeground3BrandSelected: colors_1.hcHighlightText,
  colorNeutralForeground4: colors_1.hcCanvasText,
  colorNeutralForegroundDisabled: colors_1.hcDisabled,
  colorNeutralForegroundInvertedDisabled: colors_1.hcDisabled,
  colorBrandForegroundLink: colors_1.hcHyperlink,
  colorBrandForegroundLinkHover: colors_1.hcHyperlink,
  colorBrandForegroundLinkPressed: colors_1.hcHyperlink,
  colorBrandForegroundLinkSelected: colors_1.hcHyperlink,
  colorNeutralForeground2Link: colors_1.hcHyperlink,
  colorNeutralForeground2LinkHover: colors_1.hcHyperlink,
  colorNeutralForeground2LinkPressed: colors_1.hcHyperlink,
  colorNeutralForeground2LinkSelected: colors_1.hcHyperlink,
  colorCompoundBrandForeground1: colors_1.hcHighlight,
  colorCompoundBrandForeground1Hover: colors_1.hcHighlight,
  colorCompoundBrandForeground1Pressed: colors_1.hcHighlight,
  colorBrandForeground1: colors_1.hcCanvasText,
  colorBrandForeground2: colors_1.hcButtonText,
  colorNeutralForeground1Static: colors_1.hcCanvas,
  colorNeutralForegroundStaticInverted: colors_1.hcCanvasText,
  colorNeutralForegroundInverted: colors_1.hcHighlightText,
  colorNeutralForegroundInvertedHover: colors_1.hcHighlightText,
  colorNeutralForegroundInvertedPressed: colors_1.hcHighlightText,
  colorNeutralForegroundInvertedSelected: colors_1.hcHighlightText,
  colorNeutralForegroundInverted2: colors_1.hcCanvasText,
  colorNeutralForegroundOnBrand: colors_1.hcButtonText,
  colorNeutralForegroundInvertedLink: colors_1.hcHyperlink,
  colorNeutralForegroundInvertedLinkHover: colors_1.hcHyperlink,
  colorNeutralForegroundInvertedLinkPressed: colors_1.hcHyperlink,
  colorNeutralForegroundInvertedLinkSelected: colors_1.hcHyperlink,
  colorBrandForegroundInverted: colors_1.hcCanvasText,
  colorBrandForegroundInvertedHover: colors_1.hcHighlightText,
  colorBrandForegroundInvertedPressed: colors_1.hcHighlightText,
  colorBrandForegroundOnLight: colors_1.hcButtonText,
  colorBrandForegroundOnLightHover: colors_1.hcHighlightText,
  colorBrandForegroundOnLightPressed: colors_1.hcHighlightText,
  colorBrandForegroundOnLightSelected: colors_1.hcHighlightText,
  colorNeutralBackground1: colors_1.hcCanvas,
  colorNeutralBackground1Hover: colors_1.hcHighlight,
  colorNeutralBackground1Pressed: colors_1.hcHighlight,
  colorNeutralBackground1Selected: colors_1.hcHighlight,
  colorNeutralBackground2: colors_1.hcCanvas,
  colorNeutralBackground2Hover: colors_1.hcHighlight,
  colorNeutralBackground2Pressed: colors_1.hcHighlight,
  colorNeutralBackground2Selected: colors_1.hcHighlight,
  colorNeutralBackground3: colors_1.hcCanvas,
  colorNeutralBackground3Hover: colors_1.hcHighlight,
  colorNeutralBackground3Pressed: colors_1.hcHighlight,
  colorNeutralBackground3Selected: colors_1.hcHighlight,
  colorNeutralBackground4: colors_1.hcCanvas,
  colorNeutralBackground4Hover: colors_1.hcHighlight,
  colorNeutralBackground4Pressed: colors_1.hcHighlight,
  colorNeutralBackground4Selected: colors_1.hcHighlight,
  colorNeutralBackground5: colors_1.hcCanvas,
  colorNeutralBackground5Hover: colors_1.hcHighlight,
  colorNeutralBackground5Pressed: colors_1.hcHighlight,
  colorNeutralBackground5Selected: colors_1.hcHighlight,
  colorNeutralBackground6: colors_1.hcCanvas,
  colorNeutralBackgroundInverted: colors_1.hcCanvas,
  colorNeutralBackgroundStatic: colors_1.hcCanvas,
  colorSubtleBackground: 'transparent',
  colorSubtleBackgroundHover: colors_1.hcHighlight,
  colorSubtleBackgroundPressed: colors_1.hcHighlight,
  colorSubtleBackgroundSelected: colors_1.hcHighlight,
  colorSubtleBackgroundLightAlphaHover: colors_1.hcHighlight,
  colorSubtleBackgroundLightAlphaPressed: colors_1.hcHighlight,
  colorSubtleBackgroundLightAlphaSelected: colors_1.hcHighlight,
  colorSubtleBackgroundInverted: 'transparent',
  colorSubtleBackgroundInvertedHover: colors_1.hcHighlight,
  colorSubtleBackgroundInvertedPressed: colors_1.hcHighlight,
  colorSubtleBackgroundInvertedSelected: colors_1.hcHighlight,
  colorTransparentBackground: 'transparent',
  colorTransparentBackgroundHover: colors_1.hcHighlight,
  colorTransparentBackgroundPressed: colors_1.hcHighlight,
  colorTransparentBackgroundSelected: colors_1.hcHighlight,
  colorNeutralBackgroundDisabled: colors_1.hcCanvas,
  colorNeutralBackgroundInvertedDisabled: colors_1.hcCanvas,
  colorNeutralStencil1: colors_1.hcCanvasText,
  colorNeutralStencil2: colors_1.hcCanvasText,
  colorNeutralStencil1Alpha: colors_1.hcCanvasText,
  colorNeutralStencil2Alpha: colors_1.hcCanvasText,
  colorBackgroundOverlay: colors_1.blackAlpha[50],
  colorScrollbarOverlay: colors_1.hcButtonFace,
  colorBrandBackground: colors_1.hcButtonFace,
  colorBrandBackgroundHover: colors_1.hcHighlight,
  colorBrandBackgroundPressed: colors_1.hcHighlight,
  colorBrandBackgroundSelected: colors_1.hcHighlight,
  colorCompoundBrandBackground: colors_1.hcHighlight,
  colorCompoundBrandBackgroundHover: colors_1.hcHighlight,
  colorCompoundBrandBackgroundPressed: colors_1.hcHighlight,
  colorBrandBackgroundStatic: colors_1.hcCanvas,
  colorBrandBackground2: colors_1.hcButtonFace,
  colorBrandBackgroundInverted: colors_1.hcButtonFace,
  colorBrandBackgroundInvertedHover: colors_1.hcHighlight,
  colorBrandBackgroundInvertedPressed: colors_1.hcHighlight,
  colorBrandBackgroundInvertedSelected: colors_1.hcHighlight,
  colorNeutralStrokeAccessible: colors_1.hcCanvasText,
  colorNeutralStrokeAccessibleHover: colors_1.hcHighlight,
  colorNeutralStrokeAccessiblePressed: colors_1.hcHighlight,
  colorNeutralStrokeAccessibleSelected: colors_1.hcHighlight,
  colorNeutralStroke1: colors_1.hcCanvasText,
  colorNeutralStroke1Hover: colors_1.hcHighlight,
  colorNeutralStroke1Pressed: colors_1.hcHighlight,
  colorNeutralStroke1Selected: colors_1.hcHighlight,
  colorNeutralStroke2: colors_1.hcCanvasText,
  colorNeutralStroke3: colors_1.hcCanvasText,
  colorNeutralStrokeOnBrand: colors_1.hcCanvas,
  colorNeutralStrokeOnBrand2: colors_1.hcCanvasText,
  colorNeutralStrokeOnBrand2Hover: colors_1.hcCanvasText,
  colorNeutralStrokeOnBrand2Pressed: colors_1.hcCanvasText,
  colorNeutralStrokeOnBrand2Selected: colors_1.hcCanvasText,
  colorBrandStroke1: colors_1.hcCanvasText,
  colorBrandStroke2: colors_1.hcCanvasText,
  colorCompoundBrandStroke: colors_1.hcHighlight,
  colorCompoundBrandStrokeHover: colors_1.hcHighlight,
  colorCompoundBrandStrokePressed: colors_1.hcHighlight,
  colorNeutralStrokeDisabled: colors_1.hcDisabled,
  colorNeutralStrokeInvertedDisabled: colors_1.hcDisabled,
  colorTransparentStroke: colors_1.hcCanvasText,
  colorTransparentStrokeInteractive: colors_1.hcHighlight,
  colorTransparentStrokeDisabled: colors_1.hcDisabled,
  colorStrokeFocus1: colors_1.hcCanvas,
  colorStrokeFocus2: colors_1.hcHighlight,
  colorNeutralShadowAmbient: 'rgba(0,0,0,0.24)',
  colorNeutralShadowKey: 'rgba(0,0,0,0.28)',
  colorNeutralShadowAmbientLighter: 'rgba(0,0,0,0.12)',
  colorNeutralShadowKeyLighter: 'rgba(0,0,0,0.14)',
  colorNeutralShadowAmbientDarker: 'rgba(0,0,0,0.40)',
  colorNeutralShadowKeyDarker: 'rgba(0,0,0,0.48)',
  colorBrandShadowAmbient: 'rgba(0,0,0,0.30)',
  colorBrandShadowKey: 'rgba(0,0,0,0.25)' // rgba(0,0,0,0.25) undefined

});

exports.generateColorTokens = generateColorTokens;
