// @fluentui/react-badge/lib-amd/components/PresenceBadge/usePresenceBadge.js
define(["require", "exports", "tslib", "react", "@fluentui/react-utilities", "./presenceIcons", "../Badge/index"], function (require, exports, tslib_1, React, react_utilities_1, presenceIcons_1, index_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.usePresenceBadge_unstable = void 0;
    var iconMap = function (status, outOfOffice, size) {
        switch (status) {
            case 'available':
                return outOfOffice ? presenceIcons_1.presenceAvailableRegular[size] : presenceIcons_1.presenceAvailableFilled[size];
            case 'away':
                return outOfOffice ? presenceIcons_1.presenceOfflineRegular[size] : presenceIcons_1.presenceAwayFilled[size];
            case 'busy':
                return outOfOffice ? presenceIcons_1.presenceUnknownRegular[size] : presenceIcons_1.presenceBusyFilled[size];
            case 'do-not-disturb':
                return outOfOffice ? presenceIcons_1.presenceDndRegular[size] : presenceIcons_1.presenceDndFilled[size];
            case 'offline':
                return presenceIcons_1.presenceOfflineRegular[size];
            case 'out-of-office':
                return presenceIcons_1.presenceOofRegular[size];
            case 'unknown':
                return presenceIcons_1.presenceUnknownRegular[size];
        }
    };
    var DEFAULT_STRINGS = {
        busy: 'busy',
        'out-of-office': 'out of office',
        away: 'away',
        available: 'available',
        offline: 'offline',
        'do-not-disturb': 'do not disturb',
        unknown: 'unknown',
    };
    /**
     * Returns the props and state required to render the component
     */
    var usePresenceBadge_unstable = function (props, ref) {
        var _a = props.size, size = _a === void 0 ? 'medium' : _a, _b = props.status, status = _b === void 0 ? 'available' : _b, _c = props.outOfOffice, outOfOffice = _c === void 0 ? false : _c;
        var statusText = DEFAULT_STRINGS[status];
        var oofText = props.outOfOffice && props.status !== 'out-of-office' ? " " + DEFAULT_STRINGS['out-of-office'] : '';
        var IconElement = iconMap(status, outOfOffice, size);
        var state = tslib_1.__assign(tslib_1.__assign({}, index_1.useBadge_unstable(tslib_1.__assign(tslib_1.__assign({ 'aria-label': statusText + oofText, role: 'img' }, props), { size: size, icon: react_utilities_1.resolveShorthand(props.icon, {
                defaultProps: {
                    children: IconElement ? React.createElement(IconElement, null) : null,
                },
                required: true,
            }) }), ref)), { status: status, outOfOffice: outOfOffice });
        return state;
    };
    exports.usePresenceBadge_unstable = usePresenceBadge_unstable;
});
