// @fluentui/react-menu/lib-amd/components/Menu/useMenuContextValues.js
define(["require", "exports"], function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useMenuContextValues_unstable = void 0;
    function useMenuContextValues_unstable(state) {
        var checkedValues = state.checkedValues, hasCheckmarks = state.hasCheckmarks, hasIcons = state.hasIcons, inline = state.inline, isSubmenu = state.isSubmenu, menuPopoverRef = state.menuPopoverRef, onCheckedValueChange = state.onCheckedValueChange, open = state.open, openOnContext = state.openOnContext, openOnHover = state.openOnHover, persistOnItemClick = state.persistOnItemClick, setOpen = state.setOpen, triggerId = state.triggerId, triggerRef = state.triggerRef;
        // This context is created with "@fluentui/react-context-selector", these is no sense to memoize it
        var menu = {
            checkedValues: checkedValues,
            hasCheckmarks: hasCheckmarks,
            hasIcons: hasIcons,
            inline: inline,
            isSubmenu: isSubmenu,
            menuPopoverRef: menuPopoverRef,
            onCheckedValueChange: onCheckedValueChange,
            open: open,
            openOnContext: openOnContext,
            openOnHover: openOnHover,
            persistOnItemClick: persistOnItemClick,
            setOpen: setOpen,
            triggerId: triggerId,
            triggerRef: triggerRef,
        };
        return { menu: menu };
    }
    exports.useMenuContextValues_unstable = useMenuContextValues_unstable;
});
