// @fluentui/react-button/lib-commonjs/components/MenuButton/useMenuButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useMenuButton_unstable = void 0;

const React = /*#__PURE__*/require("react");

const react_icons_1 = /*#__PURE__*/require("@fluentui/react-icons");

const react_utilities_1 = /*#__PURE__*/require("@fluentui/react-utilities");

const index_1 = /*#__PURE__*/require("../Button/index");
/**
 * Given user props, returns the final state for a MenuButton.
 */


const useMenuButton_unstable = ({
  menuIcon,
  ...props
}, ref) => {
  var _a;

  const buttonState = index_1.useButton_unstable(props, ref);
  buttonState.root['aria-expanded'] = (_a = props['aria-expanded']) !== null && _a !== void 0 ? _a : false;
  return { // Button state
    ...buttonState,
    // State calculated from a set of props
    iconOnly: Boolean(!props.children),
    // Slots definition
    components: {
      root: 'button',
      icon: 'span',
      menuIcon: 'span'
    },
    menuIcon: react_utilities_1.resolveShorthand(menuIcon, {
      defaultProps: {
        children: React.createElement(react_icons_1.ChevronDownRegular, null)
      },
      required: true
    })
  };
};

exports.useMenuButton_unstable = useMenuButton_unstable;
