// @fluentui/react-components/lib-commonjs/AccessibilityScenarios/utils.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Scenario = exports.BackLink = exports.ScenariosListLink = exports.FullscreenLink = void 0;

const React = /*#__PURE__*/require("react");

const APP_TITLE = 'Accessibility Scenarios';
const APP_TITLE_SEPARATOR = ' | '; // https://storybook.js.org/addons/@storybook/addon-links does not allow opening a story in new tab
// so this is a naive attempt for opening a story in full screen

const FullscreenLink = props => React.createElement("a", {
  className: "sbdocs sbdocs-a",
  href: `iframe.html?id=${props.parent}--${props.story}`,
  target: "_blank"
}, props.content);

exports.FullscreenLink = FullscreenLink;

const ScenariosListLink = props => React.createElement("a", {
  className: "sbdocs sbdocs-a",
  href: `iframe.html?id=concepts-developer-accessibility-stories-list-of-scenarios--page`
}, props.children);

exports.ScenariosListLink = ScenariosListLink;

const BackLink = () => React.createElement(exports.ScenariosListLink, null, "Go back to main menu");

exports.BackLink = BackLink;

const Scenario = ({
  pageTitle,
  children
}) => {
  React.useEffect(() => {
    document.title = pageTitle + APP_TITLE_SEPARATOR + APP_TITLE;
  }, [pageTitle]);
  return React.createElement("div", {
    role: "main"
  }, React.createElement(exports.BackLink, null), React.createElement("br", null), children);
};

exports.Scenario = Scenario;
