// @fluentui/react-tabs/lib-amd/components/TabList/useTabListStyles.js
define(["require", "exports", "@griffel/react"], function (require, exports, react_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useTabListStyles_unstable = exports.tabListClassNames = void 0;
    exports.tabListClassNames = {
        root: 'fui-TabList',
    };
    /**
     * Styles for the root slot
     */
    var useStyles = react_1.makeStyles({
        root: {
            display: 'flex',
            flexDirection: 'row',
            flexShrink: 0,
            flexWrap: 'nowrap',
            position: 'relative',
        },
        horizontal: {
            alignItems: 'stretch',
            flexDirection: 'row',
        },
        vertical: {
            alignItems: 'stretch',
            flexDirection: 'column',
        },
    });
    /**
     * Apply styling to the TabList slots based on the state
     */
    var useTabListStyles_unstable = function (state) {
        var vertical = state.vertical;
        var styles = useStyles();
        state.root.className = react_1.mergeClasses(exports.tabListClassNames.root, styles.root, vertical ? styles.vertical : styles.horizontal, state.root.className);
        return state;
    };
    exports.useTabListStyles_unstable = useTabListStyles_unstable;
});
