// @fluentui/react-overflow/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.OverflowItem = exports.useOverflowMenu = exports.useOverflowItem = exports.useOverflowCount = exports.useOverflowContainer = exports.useIsOverflowItemVisible = exports.useIsOverflowGroupVisible = exports.DATA_OVERFLOW_MENU = exports.DATA_OVERFLOW_ITEM = exports.DATA_OVERFLOWING = exports.Overflow = void 0;

var Overflow_1 = /*#__PURE__*/require("./components/Overflow");

Object.defineProperty(exports, "Overflow", {
  enumerable: true,
  get: function () {
    return Overflow_1.Overflow;
  }
});

var constants_1 = /*#__PURE__*/require("./constants");

Object.defineProperty(exports, "DATA_OVERFLOWING", {
  enumerable: true,
  get: function () {
    return constants_1.DATA_OVERFLOWING;
  }
});
Object.defineProperty(exports, "DATA_OVERFLOW_ITEM", {
  enumerable: true,
  get: function () {
    return constants_1.DATA_OVERFLOW_ITEM;
  }
});
Object.defineProperty(exports, "DATA_OVERFLOW_MENU", {
  enumerable: true,
  get: function () {
    return constants_1.DATA_OVERFLOW_MENU;
  }
});

var useIsOverflowGroupVisible_1 = /*#__PURE__*/require("./useIsOverflowGroupVisible");

Object.defineProperty(exports, "useIsOverflowGroupVisible", {
  enumerable: true,
  get: function () {
    return useIsOverflowGroupVisible_1.useIsOverflowGroupVisible;
  }
});

var useIsOverflowItemVisible_1 = /*#__PURE__*/require("./useIsOverflowItemVisible");

Object.defineProperty(exports, "useIsOverflowItemVisible", {
  enumerable: true,
  get: function () {
    return useIsOverflowItemVisible_1.useIsOverflowItemVisible;
  }
});

var useOverflowContainer_1 = /*#__PURE__*/require("./useOverflowContainer");

Object.defineProperty(exports, "useOverflowContainer", {
  enumerable: true,
  get: function () {
    return useOverflowContainer_1.useOverflowContainer;
  }
});

var useOverflowCount_1 = /*#__PURE__*/require("./useOverflowCount");

Object.defineProperty(exports, "useOverflowCount", {
  enumerable: true,
  get: function () {
    return useOverflowCount_1.useOverflowCount;
  }
});

var useOverflowItem_1 = /*#__PURE__*/require("./useOverflowItem");

Object.defineProperty(exports, "useOverflowItem", {
  enumerable: true,
  get: function () {
    return useOverflowItem_1.useOverflowItem;
  }
});

var useOverflowMenu_1 = /*#__PURE__*/require("./useOverflowMenu");

Object.defineProperty(exports, "useOverflowMenu", {
  enumerable: true,
  get: function () {
    return useOverflowMenu_1.useOverflowMenu;
  }
});

var OverflowItem_1 = /*#__PURE__*/require("./components/OverflowItem/OverflowItem");

Object.defineProperty(exports, "OverflowItem", {
  enumerable: true,
  get: function () {
    return OverflowItem_1.OverflowItem;
  }
});
