// @fluentui/react-field/lib/components/Field/useFieldStyles.js
import { tokens, typographyStyles } from '@fluentui/react-theme';
import { __styles, mergeClasses } from '@griffel/react';
export const getFieldClassNames = name => ({
  root: `fui-${name}`,
  control: `fui-${name}__control`,
  label: `fui-${name}__label`,
  validationMessage: `fui-${name}__validationMessage`,
  validationMessageIcon: `fui-${name}__validationMessageIcon`,
  hint: `fui-${name}__hint`
});
/**
 * Styles for the root slot
 */

const useRootStyles = /*#__PURE__*/__styles({
  "base": {
    "mc9l5x": "f13qh94s",
    "Bxotwcr": "f14np9u9",
    "B7hvi0a": "f1m2n5bn"
  },
  "horizontal": {
    "wkccdc": "fai812u",
    "Budl1dq": "fckz59x"
  },
  "secondColumn": {
    "Br312pm": "fd46tj4"
  }
}, {
  "d": [".f13qh94s{display:grid;}", ".f14np9u9{grid-auto-flow:row;}", ".f1m2n5bn{justify-items:start;}", ".fai812u{grid-template-rows:auto auto auto auto;}", ".fckz59x{grid-template-columns:1fr 2fr;}", ".fd46tj4{grid-column-start:2;}"]
});

const useLabelStyles = /*#__PURE__*/__styles({
  "base": {
    "B6of3ja": "fg30ohd",
    "jrapky": "fyacil5"
  },
  "horizontal": {
    "Ijaq50": "f16hsg94",
    "nk6f5a": "f1nzqi2z",
    "t21cq0": ["fkujibs", "f199hnxi"],
    "qb2dma": "f9h729m",
    "Bdqf98w": "fhb5wj7"
  }
}, {
  "d": [".fg30ohd{margin-top:var(--spacingVerticalXXS);}", ".fyacil5{margin-bottom:var(--spacingVerticalXXS);}", ".f16hsg94{grid-row-start:1;}", ".f1nzqi2z{grid-row-end:-1;}", ".fkujibs{margin-right:var(--spacingHorizontalM);}", ".f199hnxi{margin-left:var(--spacingHorizontalM);}", ".f9h729m{-webkit-align-self:start;-ms-flex-item-align:start;align-self:start;}", ".fhb5wj7{justify-self:stretch;}"]
});

const useSecondaryTextStyles = /*#__PURE__*/__styles({
  "base": {
    "B6of3ja": "fg30ohd",
    "sj55zd": "f11d4kpn",
    "Bahqtrf": "fk6fouc",
    "Be2twd7": "fy9rknc",
    "Bhrd7zp": "figsok6",
    "Bg96gwp": "fwrc4pm"
  },
  "error": {
    "sj55zd": "f1hcrxcs"
  }
}, {
  "d": [".fg30ohd{margin-top:var(--spacingVerticalXXS);}", ".f11d4kpn{color:var(--colorNeutralForeground3);}", ".fk6fouc{font-family:var(--fontFamilyBase);}", ".fy9rknc{font-size:var(--fontSizeBase200);}", ".figsok6{font-weight:var(--fontWeightRegular);}", ".fwrc4pm{line-height:var(--lineHeightBase200);}", ".f1hcrxcs{color:var(--colorPaletteRedForeground1);}"]
});

const useValidationMessageIconStyles = /*#__PURE__*/__styles({
  "base": {
    "Be2twd7": "f1ugzwwg",
    "Bg96gwp": "fp4gqsa",
    "ha4doy": "fmrv4ls",
    "t21cq0": ["fm0x6gh", "fbyavb5"]
  },
  "error": {
    "sj55zd": "f1hcrxcs"
  },
  "warning": {
    "sj55zd": "f1k5f75o"
  },
  "success": {
    "sj55zd": "ffmvakt"
  }
}, {
  "d": [".f1ugzwwg{font-size:12px;}", ".fp4gqsa{line-height:12px;}", ".fmrv4ls{vertical-align:middle;}", ".fm0x6gh{margin-right:var(--spacingHorizontalXS);}", ".fbyavb5{margin-left:var(--spacingHorizontalXS);}", ".f1hcrxcs{color:var(--colorPaletteRedForeground1);}", ".f1k5f75o{color:var(--colorPaletteDarkOrangeForeground1);}", ".ffmvakt{color:var(--colorPaletteGreenForeground1);}"]
});
/**
 * Apply styling to the Field slots based on the state
 */


export const useFieldStyles_unstable = state => {
  const classNames = state.classNames;
  const validationState = state.validationState;
  const horizontal = state.orientation === 'horizontal';
  const rootStyles = useRootStyles();
  state.root.className = mergeClasses(classNames.root, rootStyles.base, horizontal && rootStyles.horizontal, state.root.className);

  if (state.control) {
    state.control.className = mergeClasses(classNames.control, horizontal && rootStyles.secondColumn, state.control.className);
  }

  const labelStyles = useLabelStyles();

  if (state.label) {
    state.label.className = mergeClasses(classNames.label, labelStyles.base, horizontal && labelStyles.horizontal, state.label.className);
  }

  const validationMessageIconStyles = useValidationMessageIconStyles();

  if (state.validationMessageIcon) {
    state.validationMessageIcon.className = mergeClasses(classNames.validationMessageIcon, validationMessageIconStyles.base, !!validationState && validationMessageIconStyles[validationState], state.validationMessageIcon.className);
  }

  const secondaryTextStyles = useSecondaryTextStyles();

  if (state.validationMessage) {
    state.validationMessage.className = mergeClasses(classNames.validationMessage, secondaryTextStyles.base, horizontal && rootStyles.secondColumn, validationState === 'error' && secondaryTextStyles.error, state.validationMessage.className);
  }

  if (state.hint) {
    state.hint.className = mergeClasses(classNames.hint, secondaryTextStyles.base, horizontal && rootStyles.secondColumn, state.hint.className);
  }
};
