// @fluentui/react-theme/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.typographyStyles = exports.tokens = exports.themeToTokensObject = exports.createTeamsDarkTheme = exports.createLightTheme = exports.createHighContrastTheme = exports.createDarkTheme = exports.webLightTheme = exports.webDarkTheme = exports.teamsLightTheme = exports.teamsHighContrastTheme = exports.teamsDarkTheme = void 0;

var tokens_1 = /*#__PURE__*/require("@fluentui/tokens");

Object.defineProperty(exports, "teamsDarkTheme", {
  enumerable: true,
  get: function () {
    return tokens_1.teamsDarkTheme;
  }
});
Object.defineProperty(exports, "teamsHighContrastTheme", {
  enumerable: true,
  get: function () {
    return tokens_1.teamsHighContrastTheme;
  }
});
Object.defineProperty(exports, "teamsLightTheme", {
  enumerable: true,
  get: function () {
    return tokens_1.teamsLightTheme;
  }
});
Object.defineProperty(exports, "webDarkTheme", {
  enumerable: true,
  get: function () {
    return tokens_1.webDarkTheme;
  }
});
Object.defineProperty(exports, "webLightTheme", {
  enumerable: true,
  get: function () {
    return tokens_1.webLightTheme;
  }
});
Object.defineProperty(exports, "createDarkTheme", {
  enumerable: true,
  get: function () {
    return tokens_1.createDarkTheme;
  }
});
Object.defineProperty(exports, "createHighContrastTheme", {
  enumerable: true,
  get: function () {
    return tokens_1.createHighContrastTheme;
  }
});
Object.defineProperty(exports, "createLightTheme", {
  enumerable: true,
  get: function () {
    return tokens_1.createLightTheme;
  }
});
Object.defineProperty(exports, "createTeamsDarkTheme", {
  enumerable: true,
  get: function () {
    return tokens_1.createTeamsDarkTheme;
  }
});
Object.defineProperty(exports, "themeToTokensObject", {
  enumerable: true,
  get: function () {
    return tokens_1.themeToTokensObject;
  }
});
Object.defineProperty(exports, "tokens", {
  enumerable: true,
  get: function () {
    return tokens_1.tokens;
  }
});
Object.defineProperty(exports, "typographyStyles", {
  enumerable: true,
  get: function () {
    return tokens_1.typographyStyles;
  }
});
