// @fluentui/react-menu/lib-commonjs/components/MenuItemRadio/useMenuItemRadioStyles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useMenuItemRadioStyles_unstable = exports.menuItemRadioClassNames = void 0;

const react_1 = /*#__PURE__*/require("@griffel/react");

const index_1 = /*#__PURE__*/require("../../selectable/index");

const useMenuItemStyles_1 = /*#__PURE__*/require("../MenuItem/useMenuItemStyles");

exports.menuItemRadioClassNames = {
  root: 'fui-MenuItemRadio',
  icon: 'fui-MenuItemRadio__icon',
  checkmark: 'fui-MenuItemRadio__checkmark',
  content: 'fui-MenuItemRadio__content',
  secondaryContent: 'fui-MenuItemRadio__secondaryContent'
};

const useMenuItemRadioStyles_unstable = state => {
  state.root.className = react_1.mergeClasses(exports.menuItemRadioClassNames.root, state.root.className);

  if (state.content) {
    state.content.className = react_1.mergeClasses(exports.menuItemRadioClassNames.content, state.content.className);
  }

  if (state.secondaryContent) {
    state.secondaryContent.className = react_1.mergeClasses(exports.menuItemRadioClassNames.secondaryContent, state.secondaryContent.className);
  }

  if (state.icon) {
    state.icon.className = react_1.mergeClasses(exports.menuItemRadioClassNames.icon, state.icon.className);
  }

  if (state.checkmark) {
    state.checkmark.className = react_1.mergeClasses(exports.menuItemRadioClassNames.checkmark, state.checkmark.className);
  }

  useMenuItemStyles_1.useMenuItemStyles_unstable(state);
  index_1.useCheckmarkStyles_unstable(state);
};

exports.useMenuItemRadioStyles_unstable = useMenuItemRadioStyles_unstable;
