// @fluentui/react-menu/lib-commonjs/components/MenuItemRadio/useMenuItemRadio.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useMenuItemRadio_unstable = void 0;

const React = /*#__PURE__*/require("react");

const react_utilities_1 = /*#__PURE__*/require("@fluentui/react-utilities");

const react_icons_1 = /*#__PURE__*/require("@fluentui/react-icons");

const menuListContext_1 = /*#__PURE__*/require("../../contexts/menuListContext");

const useMenuItem_1 = /*#__PURE__*/require("../MenuItem/useMenuItem");
/**
 * Given user props, returns state and render function for a MenuItemRadio.
 */


const useMenuItemRadio_unstable = (props, ref) => {
  const {
    name,
    value
  } = props;
  const checked = menuListContext_1.useMenuListContext_unstable(context => {
    var _a;

    const checkedItems = ((_a = context.checkedValues) === null || _a === void 0 ? void 0 : _a[name]) || [];
    return checkedItems.indexOf(value) !== -1;
  });
  const selectRadio = menuListContext_1.useMenuListContext_unstable(context => context.selectRadio);
  return { ...useMenuItem_1.useMenuItem_unstable({ ...props,
      role: 'menuitemradio',
      'aria-checked': checked,
      checkmark: react_utilities_1.resolveShorthand(props.checkmark, {
        defaultProps: {
          children: React.createElement(react_icons_1.Checkmark16Filled, null)
        },
        required: true
      }),
      onClick: e => {
        var _a;

        selectRadio === null || selectRadio === void 0 ? void 0 : selectRadio(e, name, value, checked);
        (_a = props.onClick) === null || _a === void 0 ? void 0 : _a.call(props, e);
      }
    }, ref),
    checked,
    name,
    value
  };
};

exports.useMenuItemRadio_unstable = useMenuItemRadio_unstable;
