// @fluentui/react-avatar/lib-commonjs/components/AvatarGroupItem/useAvatarGroupItem.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useAvatarGroupItem_unstable = void 0;

const Avatar_1 = /*#__PURE__*/require("../Avatar/Avatar");

const AvatarGroupContext_1 = /*#__PURE__*/require("../../contexts/AvatarGroupContext");

const useAvatarGroup_1 = /*#__PURE__*/require("../AvatarGroup/useAvatarGroup");

const react_utilities_1 = /*#__PURE__*/require("@fluentui/react-utilities");

const react_context_selector_1 = /*#__PURE__*/require("@fluentui/react-context-selector");
/**
 * Create the state required to render AvatarGroupItem.
 *
 * The returned state can be modified with hooks such as useAvatarGroupItemStyles_unstable,
 * before being passed to renderAvatarGroupItem_unstable.
 *
 * @param props - props from this instance of AvatarGroupItem
 * @param ref - reference to root HTMLElement of AvatarGroupItem
 */


const useAvatarGroupItem_unstable = (props, ref) => {
  const groupIsOverflow = AvatarGroupContext_1.useAvatarGroupContext_unstable(ctx => ctx.isOverflow);
  const groupSize = AvatarGroupContext_1.useAvatarGroupContext_unstable(ctx => ctx.size);
  const layout = AvatarGroupContext_1.useAvatarGroupContext_unstable(ctx => ctx.layout); // Since the primary slot is not an intrinsic element, getPartitionedNativeProps cannot be used here.

  const {
    style,
    className,
    ...avatarSlotProps
  } = props;
  const size = groupSize !== null && groupSize !== void 0 ? groupSize : useAvatarGroup_1.defaultAvatarGroupSize;
  const hasAvatarGroupContext = react_context_selector_1.useHasParentContext(AvatarGroupContext_1.AvatarGroupContext);

  if (process.env.NODE_ENV !== 'production' && !hasAvatarGroupContext) {
    // eslint-disable-next-line no-console
    console.warn('AvatarGroupItem must only be used inside an AvatarGroup component.');
  }

  return {
    isOverflowItem: groupIsOverflow,
    layout,
    size,
    components: {
      root: groupIsOverflow ? 'li' : 'div',
      avatar: Avatar_1.Avatar,
      overflowLabel: 'span'
    },
    root: react_utilities_1.resolveShorthand(props.root, {
      required: true,
      defaultProps: {
        style,
        className
      }
    }),
    avatar: react_utilities_1.resolveShorthand(props.avatar, {
      required: true,
      defaultProps: {
        ref,
        size,
        color: 'colorful',
        ...avatarSlotProps
      }
    }),
    overflowLabel: react_utilities_1.resolveShorthand(props.overflowLabel, {
      required: true,
      defaultProps: {
        // Avatar already has its aria-label set to the name, this will prevent the name to be read twice.
        'aria-hidden': true,
        children: props.name
      }
    })
  };
};

exports.useAvatarGroupItem_unstable = useAvatarGroupItem_unstable;
