// @fluentui/react-card/lib-commonjs/components/CardHeader/useCardHeader.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useCardHeader_unstable = void 0;

const React = /*#__PURE__*/require("react");

const react_utilities_1 = /*#__PURE__*/require("@fluentui/react-utilities");

const CardContext_1 = /*#__PURE__*/require("../Card/CardContext");

const useCardHeaderStyles_1 = /*#__PURE__*/require("./useCardHeaderStyles");
/**
 * Create the state required to render CardHeader.
 *
 * The returned state can be modified with hooks such as useCardHeaderStyles_unstable,
 * before being passed to renderCardHeader_unstable.
 *
 * @param props - props from this instance of CardHeader
 * @param ref - reference to root HTMLElement of CardHeader
 */


const useCardHeader_unstable = (props, ref) => {
  const {
    image,
    header,
    description,
    action
  } = props;
  const {
    selectableA11yProps: {
      referenceId,
      setReferenceId
    }
  } = CardContext_1.useCardContext_unstable();
  const headerRef = React.useRef(null);
  const generatedId = react_utilities_1.useId(useCardHeaderStyles_1.cardHeaderClassNames.header, referenceId);
  React.useEffect(() => {
    if (header && headerRef.current) {
      const {
        id
      } = headerRef.current;
      setReferenceId(id ? id : generatedId);
    }
  }, [header, setReferenceId, generatedId]);
  return {
    components: {
      root: 'div',
      image: 'div',
      header: 'div',
      description: 'div',
      action: 'div'
    },
    root: react_utilities_1.getNativeElementProps('div', {
      ref,
      ...props
    }),
    image: react_utilities_1.resolveShorthand(image),
    header: react_utilities_1.resolveShorthand(header, {
      required: true,
      defaultProps: {
        ref: headerRef,
        id: referenceId
      }
    }),
    description: react_utilities_1.resolveShorthand(description),
    action: react_utilities_1.resolveShorthand(action)
  };
};

exports.useCardHeader_unstable = useCardHeader_unstable;
