// @fluentui/react-input/lib-commonjs/components/Input/useInput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useInput_unstable = void 0;

const react_utilities_1 = /*#__PURE__*/require("@fluentui/react-utilities");
/**
 * Create the state required to render Input.
 *
 * The returned state can be modified with hooks such as useInputStyles_unstable,
 * before being passed to renderInput_unstable.
 *
 * @param props - props from this instance of Input
 * @param ref - reference to `<input>` element of Input
 */


const useInput_unstable = (props, ref) => {
  const {
    size = 'medium',
    appearance = 'outline',
    onChange
  } = props;

  if (process.env.NODE_ENV !== 'production' && (appearance === 'filled-darker-shadow' || appearance === 'filled-lighter-shadow')) {
    // eslint-disable-next-line no-console
    console.error("The 'filled-darker-shadow' and 'filled-lighter-shadow' appearances are deprecated and will be removed in the" + ' future.');
  }

  const [value, setValue] = react_utilities_1.useControllableState({
    state: props.value,
    defaultState: props.defaultValue,
    initialState: ''
  });
  const nativeProps = react_utilities_1.getPartitionedNativeProps({
    props,
    primarySlotTagName: 'input',
    excludedPropNames: ['size', 'onChange', 'value', 'defaultValue']
  });
  const state = {
    size,
    appearance,
    components: {
      root: 'span',
      input: 'input',
      contentBefore: 'span',
      contentAfter: 'span'
    },
    input: react_utilities_1.resolveShorthand(props.input, {
      required: true,
      defaultProps: {
        type: 'text',
        ref,
        ...nativeProps.primary
      }
    }),
    contentAfter: react_utilities_1.resolveShorthand(props.contentAfter),
    contentBefore: react_utilities_1.resolveShorthand(props.contentBefore),
    root: react_utilities_1.resolveShorthand(props.root, {
      required: true,
      defaultProps: nativeProps.root
    })
  };
  state.input.value = value;
  state.input.onChange = react_utilities_1.useEventCallback(ev => {
    const newValue = ev.target.value;
    onChange === null || onChange === void 0 ? void 0 : onChange(ev, {
      value: newValue
    });
    setValue(newValue);
  });
  return state;
};

exports.useInput_unstable = useInput_unstable;
