// @fluentui/react-table/lib/components/Table/useTable.js
import { getNativeElementProps } from '@fluentui/react-utilities';
/**
 * Create the state required to render Table.
 *
 * The returned state can be modified with hooks such as useTableStyles_unstable,
 * before being passed to renderTable_unstable.
 *
 * @param props - props from this instance of Table
 * @param ref - reference to root HTMLElement of Table
 */

export const useTable_unstable = (props, ref) => {
  var _a, _b, _c, _d;

  const rootComponent = ((_a = props.as) !== null && _a !== void 0 ? _a : props.noNativeElements) ? 'div' : 'table';
  return {
    components: {
      root: rootComponent
    },
    root: getNativeElementProps(rootComponent, {
      ref,
      role: rootComponent === 'div' ? 'table' : undefined,
      ...props
    }),
    size: (_b = props.size) !== null && _b !== void 0 ? _b : 'medium',
    noNativeElements: (_c = props.noNativeElements) !== null && _c !== void 0 ? _c : false,
    sortable: (_d = props.sortable) !== null && _d !== void 0 ? _d : false
  };
};
