// @fluentui/tokens/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.typographyStyles = exports.tokens = exports.themeToTokensObject = exports.createTeamsDarkTheme = exports.createLightTheme = exports.createHighContrastTheme = exports.createDarkTheme = exports.webLightTheme = exports.webDarkTheme = exports.teamsLightTheme = exports.teamsHighContrastTheme = exports.teamsDarkTheme = void 0;

var index_1 = /*#__PURE__*/require("./themes/index");

Object.defineProperty(exports, "teamsDarkTheme", {
  enumerable: true,
  get: function () {
    return index_1.teamsDarkTheme;
  }
});
Object.defineProperty(exports, "teamsHighContrastTheme", {
  enumerable: true,
  get: function () {
    return index_1.teamsHighContrastTheme;
  }
});
Object.defineProperty(exports, "teamsLightTheme", {
  enumerable: true,
  get: function () {
    return index_1.teamsLightTheme;
  }
});
Object.defineProperty(exports, "webDarkTheme", {
  enumerable: true,
  get: function () {
    return index_1.webDarkTheme;
  }
});
Object.defineProperty(exports, "webLightTheme", {
  enumerable: true,
  get: function () {
    return index_1.webLightTheme;
  }
});

var index_2 = /*#__PURE__*/require("./utils/index");

Object.defineProperty(exports, "createDarkTheme", {
  enumerable: true,
  get: function () {
    return index_2.createDarkTheme;
  }
});
Object.defineProperty(exports, "createHighContrastTheme", {
  enumerable: true,
  get: function () {
    return index_2.createHighContrastTheme;
  }
});
Object.defineProperty(exports, "createLightTheme", {
  enumerable: true,
  get: function () {
    return index_2.createLightTheme;
  }
});
Object.defineProperty(exports, "createTeamsDarkTheme", {
  enumerable: true,
  get: function () {
    return index_2.createTeamsDarkTheme;
  }
});

var themeToTokensObject_1 = /*#__PURE__*/require("./themeToTokensObject");

Object.defineProperty(exports, "themeToTokensObject", {
  enumerable: true,
  get: function () {
    return themeToTokensObject_1.themeToTokensObject;
  }
});

var tokens_1 = /*#__PURE__*/require("./tokens");

Object.defineProperty(exports, "tokens", {
  enumerable: true,
  get: function () {
    return tokens_1.tokens;
  }
});

var index_3 = /*#__PURE__*/require("./global/index");

Object.defineProperty(exports, "typographyStyles", {
  enumerable: true,
  get: function () {
    return index_3.typographyStyles;
  }
});
