// @fluentui/react-menu/lib-commonjs/contexts/menuContext.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useMenuContext_unstable = exports.MenuProvider = exports.MenuContext = void 0;

const react_context_selector_1 = /*#__PURE__*/require("@fluentui/react-context-selector");

exports.MenuContext = /*#__PURE__*/react_context_selector_1.createContext(undefined);
const menuContextDefaultValue = {
  open: false,
  setOpen: () => false,
  checkedValues: {},
  onCheckedValueChange: () => null,
  isSubmenu: false,
  triggerRef: {
    current: null
  },
  menuPopoverRef: {
    current: null
  },
  triggerId: '',
  openOnContext: false,
  openOnHover: false,
  hasIcons: false,
  hasCheckmarks: false,
  inline: false,
  persistOnItemClick: false
};
exports.MenuProvider = exports.MenuContext.Provider;

const useMenuContext_unstable = selector => react_context_selector_1.useContextSelector(exports.MenuContext, (ctx = menuContextDefaultValue) => selector(ctx));

exports.useMenuContext_unstable = useMenuContext_unstable;
