// @fluentui/react-popover/lib-commonjs/components/PopoverTrigger/usePopoverTrigger.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.usePopoverTrigger_unstable = void 0;

const react_utilities_1 = /*#__PURE__*/require("@fluentui/react-utilities");

const react_tabster_1 = /*#__PURE__*/require("@fluentui/react-tabster");

const popoverContext_1 = /*#__PURE__*/require("../../popoverContext");

const react_aria_1 = /*#__PURE__*/require("@fluentui/react-aria");

const keyboard_keys_1 = /*#__PURE__*/require("@fluentui/keyboard-keys");
/**
 * Create the state required to render PopoverTrigger.
 *
 * The returned state can be modified with hooks such as usePopoverTriggerStyles,
 * before being passed to renderPopoverTrigger_unstable.
 *
 * @param props - props from this instance of PopoverTrigger
 */


const usePopoverTrigger_unstable = props => {
  const {
    children,
    disableButtonEnhancement = false
  } = props;
  const child = react_utilities_1.getTriggerChild(children);
  const open = popoverContext_1.usePopoverContext_unstable(context => context.open);
  const setOpen = popoverContext_1.usePopoverContext_unstable(context => context.setOpen);
  const toggleOpen = popoverContext_1.usePopoverContext_unstable(context => context.toggleOpen);
  const triggerRef = popoverContext_1.usePopoverContext_unstable(context => context.triggerRef);
  const openOnHover = popoverContext_1.usePopoverContext_unstable(context => context.openOnHover);
  const openOnContext = popoverContext_1.usePopoverContext_unstable(context => context.openOnContext);
  const {
    triggerAttributes
  } = react_tabster_1.useModalAttributes();

  const onContextMenu = e => {
    if (openOnContext) {
      e.preventDefault();
      setOpen(e, true);
    }
  };

  const onClick = e => {
    if (!openOnContext) {
      toggleOpen(e);
    }
  };

  const onKeyDown = e => {
    if (e.key === keyboard_keys_1.Escape && open) {
      setOpen(e, false); // stop propagation to avoid conflicting with other elements that listen for `Escape`
      // e,g: Dialog, Menu

      e.stopPropagation();
    }
  };

  const onMouseEnter = e => {
    if (openOnHover) {
      setOpen(e, true);
    }
  };

  const onMouseLeave = e => {
    if (openOnHover) {
      setOpen(e, false);
    }
  };

  const contextMenuProps = { ...triggerAttributes,
    'aria-expanded': `${open}`,
    ...(child === null || child === void 0 ? void 0 : child.props),
    onMouseEnter: react_utilities_1.useEventCallback(react_utilities_1.mergeCallbacks(child === null || child === void 0 ? void 0 : child.props.onMouseEnter, onMouseEnter)),
    onMouseLeave: react_utilities_1.useEventCallback(react_utilities_1.mergeCallbacks(child === null || child === void 0 ? void 0 : child.props.onMouseLeave, onMouseLeave)),
    onContextMenu: react_utilities_1.useEventCallback(react_utilities_1.mergeCallbacks(child === null || child === void 0 ? void 0 : child.props.onContextMenu, onContextMenu)),
    ref: react_utilities_1.useMergedRefs(triggerRef, child === null || child === void 0 ? void 0 : child.ref)
  };
  const triggerChildProps = { ...contextMenuProps,
    onClick: react_utilities_1.useEventCallback(react_utilities_1.mergeCallbacks(child === null || child === void 0 ? void 0 : child.props.onClick, onClick)),
    onKeyDown: react_utilities_1.useEventCallback(react_utilities_1.mergeCallbacks(child === null || child === void 0 ? void 0 : child.props.onKeyDown, onKeyDown))
  };
  const ariaButtonTriggerChildProps = react_aria_1.useARIAButtonProps((child === null || child === void 0 ? void 0 : child.type) === 'button' || (child === null || child === void 0 ? void 0 : child.type) === 'a' ? child.type : 'div', triggerChildProps);
  return {
    children: react_utilities_1.applyTriggerPropsToChildren(props.children, react_aria_1.useARIAButtonProps((child === null || child === void 0 ? void 0 : child.type) === 'button' || (child === null || child === void 0 ? void 0 : child.type) === 'a' ? child.type : 'div', openOnContext ? contextMenuProps : disableButtonEnhancement ? triggerChildProps : ariaButtonTriggerChildProps))
  };
};

exports.usePopoverTrigger_unstable = usePopoverTrigger_unstable;
