// @fluentui/tokens/lib-commonjs/alias/lightColor.js
"use strict";
/* !!! DO NOT EDIT !!! */

/* This file has been generated by the token pipeline */

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.generateColorTokens = void 0;

const colors_1 = /*#__PURE__*/require("../global/colors");

const generateColorTokens = brand => ({
  colorNeutralForeground1: colors_1.grey[14],
  colorNeutralForeground1Hover: colors_1.grey[14],
  colorNeutralForeground1Pressed: colors_1.grey[14],
  colorNeutralForeground1Selected: colors_1.grey[14],
  colorNeutralForeground2: colors_1.grey[26],
  colorNeutralForeground2Hover: colors_1.grey[14],
  colorNeutralForeground2Pressed: colors_1.grey[14],
  colorNeutralForeground2Selected: colors_1.grey[14],
  colorNeutralForeground2BrandHover: brand[80],
  colorNeutralForeground2BrandPressed: brand[70],
  colorNeutralForeground2BrandSelected: brand[80],
  colorNeutralForeground3: colors_1.grey[38],
  colorNeutralForeground3Hover: colors_1.grey[26],
  colorNeutralForeground3Pressed: colors_1.grey[26],
  colorNeutralForeground3Selected: colors_1.grey[26],
  colorNeutralForeground3BrandHover: brand[80],
  colorNeutralForeground3BrandPressed: brand[70],
  colorNeutralForeground3BrandSelected: brand[80],
  colorNeutralForeground4: colors_1.grey[44],
  colorNeutralForegroundDisabled: colors_1.grey[74],
  colorNeutralForegroundInvertedDisabled: colors_1.whiteAlpha[40],
  colorBrandForegroundLink: brand[70],
  colorBrandForegroundLinkHover: brand[60],
  colorBrandForegroundLinkPressed: brand[40],
  colorBrandForegroundLinkSelected: brand[70],
  colorNeutralForeground2Link: colors_1.grey[26],
  colorNeutralForeground2LinkHover: colors_1.grey[14],
  colorNeutralForeground2LinkPressed: colors_1.grey[14],
  colorNeutralForeground2LinkSelected: colors_1.grey[14],
  colorCompoundBrandForeground1: brand[80],
  colorCompoundBrandForeground1Hover: brand[70],
  colorCompoundBrandForeground1Pressed: brand[60],
  colorBrandForeground1: brand[80],
  colorBrandForeground2: brand[70],
  colorNeutralForeground1Static: colors_1.grey[14],
  colorNeutralForegroundStaticInverted: colors_1.white,
  colorNeutralForegroundInverted: colors_1.white,
  colorNeutralForegroundInvertedHover: colors_1.white,
  colorNeutralForegroundInvertedPressed: colors_1.white,
  colorNeutralForegroundInvertedSelected: colors_1.white,
  colorNeutralForegroundInverted2: colors_1.white,
  colorNeutralForegroundOnBrand: colors_1.white,
  colorNeutralForegroundInvertedLink: colors_1.white,
  colorNeutralForegroundInvertedLinkHover: colors_1.white,
  colorNeutralForegroundInvertedLinkPressed: colors_1.white,
  colorNeutralForegroundInvertedLinkSelected: colors_1.white,
  colorBrandForegroundInverted: brand[100],
  colorBrandForegroundInvertedHover: brand[110],
  colorBrandForegroundInvertedPressed: brand[100],
  colorBrandForegroundOnLight: brand[80],
  colorBrandForegroundOnLightHover: brand[70],
  colorBrandForegroundOnLightPressed: brand[50],
  colorBrandForegroundOnLightSelected: brand[60],
  colorNeutralBackground1: colors_1.white,
  colorNeutralBackground1Hover: colors_1.grey[96],
  colorNeutralBackground1Pressed: colors_1.grey[88],
  colorNeutralBackground1Selected: colors_1.grey[92],
  colorNeutralBackground2: colors_1.grey[98],
  colorNeutralBackground2Hover: colors_1.grey[94],
  colorNeutralBackground2Pressed: colors_1.grey[86],
  colorNeutralBackground2Selected: colors_1.grey[90],
  colorNeutralBackground3: colors_1.grey[96],
  colorNeutralBackground3Hover: colors_1.grey[92],
  colorNeutralBackground3Pressed: colors_1.grey[84],
  colorNeutralBackground3Selected: colors_1.grey[88],
  colorNeutralBackground4: colors_1.grey[94],
  colorNeutralBackground4Hover: colors_1.grey[98],
  colorNeutralBackground4Pressed: colors_1.grey[96],
  colorNeutralBackground4Selected: colors_1.white,
  colorNeutralBackground5: colors_1.grey[92],
  colorNeutralBackground5Hover: colors_1.grey[96],
  colorNeutralBackground5Pressed: colors_1.grey[94],
  colorNeutralBackground5Selected: colors_1.grey[98],
  colorNeutralBackground6: colors_1.grey[90],
  colorNeutralBackgroundInverted: colors_1.grey[16],
  colorNeutralBackgroundStatic: colors_1.grey[20],
  colorSubtleBackground: 'transparent',
  colorSubtleBackgroundHover: colors_1.grey[96],
  colorSubtleBackgroundPressed: colors_1.grey[88],
  colorSubtleBackgroundSelected: colors_1.grey[92],
  colorSubtleBackgroundLightAlphaHover: colors_1.whiteAlpha[70],
  colorSubtleBackgroundLightAlphaPressed: colors_1.whiteAlpha[50],
  colorSubtleBackgroundLightAlphaSelected: 'transparent',
  colorSubtleBackgroundInverted: 'transparent',
  colorSubtleBackgroundInvertedHover: colors_1.blackAlpha[10],
  colorSubtleBackgroundInvertedPressed: colors_1.blackAlpha[30],
  colorSubtleBackgroundInvertedSelected: colors_1.blackAlpha[20],
  colorTransparentBackground: 'transparent',
  colorTransparentBackgroundHover: 'transparent',
  colorTransparentBackgroundPressed: 'transparent',
  colorTransparentBackgroundSelected: 'transparent',
  colorNeutralBackgroundDisabled: colors_1.grey[94],
  colorNeutralBackgroundInvertedDisabled: colors_1.whiteAlpha[10],
  colorNeutralStencil1: colors_1.grey[90],
  colorNeutralStencil2: colors_1.grey[98],
  colorNeutralStencil1Alpha: colors_1.blackAlpha[10],
  colorNeutralStencil2Alpha: colors_1.blackAlpha[5],
  colorBackgroundOverlay: colors_1.blackAlpha[40],
  colorScrollbarOverlay: colors_1.blackAlpha[50],
  colorBrandBackground: brand[80],
  colorBrandBackgroundHover: brand[70],
  colorBrandBackgroundPressed: brand[40],
  colorBrandBackgroundSelected: brand[60],
  colorCompoundBrandBackground: brand[80],
  colorCompoundBrandBackgroundHover: brand[70],
  colorCompoundBrandBackgroundPressed: brand[60],
  colorBrandBackgroundStatic: brand[80],
  colorBrandBackground2: brand[160],
  colorBrandBackgroundInverted: colors_1.white,
  colorBrandBackgroundInvertedHover: brand[160],
  colorBrandBackgroundInvertedPressed: brand[140],
  colorBrandBackgroundInvertedSelected: brand[150],
  colorNeutralStrokeAccessible: colors_1.grey[38],
  colorNeutralStrokeAccessibleHover: colors_1.grey[34],
  colorNeutralStrokeAccessiblePressed: colors_1.grey[30],
  colorNeutralStrokeAccessibleSelected: brand[80],
  colorNeutralStroke1: colors_1.grey[82],
  colorNeutralStroke1Hover: colors_1.grey[78],
  colorNeutralStroke1Pressed: colors_1.grey[70],
  colorNeutralStroke1Selected: colors_1.grey[74],
  colorNeutralStroke2: colors_1.grey[88],
  colorNeutralStroke3: colors_1.grey[94],
  colorNeutralStrokeOnBrand: colors_1.white,
  colorNeutralStrokeOnBrand2: colors_1.white,
  colorNeutralStrokeOnBrand2Hover: colors_1.white,
  colorNeutralStrokeOnBrand2Pressed: colors_1.white,
  colorNeutralStrokeOnBrand2Selected: colors_1.white,
  colorBrandStroke1: brand[80],
  colorBrandStroke2: brand[140],
  colorCompoundBrandStroke: brand[80],
  colorCompoundBrandStrokeHover: brand[70],
  colorCompoundBrandStrokePressed: brand[60],
  colorNeutralStrokeDisabled: colors_1.grey[88],
  colorNeutralStrokeInvertedDisabled: colors_1.whiteAlpha[40],
  colorTransparentStroke: 'transparent',
  colorTransparentStrokeInteractive: 'transparent',
  colorTransparentStrokeDisabled: 'transparent',
  colorStrokeFocus1: colors_1.white,
  colorStrokeFocus2: colors_1.black,
  colorNeutralShadowAmbient: 'rgba(0,0,0,0.12)',
  colorNeutralShadowKey: 'rgba(0,0,0,0.14)',
  colorNeutralShadowAmbientLighter: 'rgba(0,0,0,0.06)',
  colorNeutralShadowKeyLighter: 'rgba(0,0,0,0.07)',
  colorNeutralShadowAmbientDarker: 'rgba(0,0,0,0.20)',
  colorNeutralShadowKeyDarker: 'rgba(0,0,0,0.24)',
  colorBrandShadowAmbient: 'rgba(0,0,0,0.30)',
  colorBrandShadowKey: 'rgba(0,0,0,0.25)' // rgba(0,0,0,0.25) undefined

});

exports.generateColorTokens = generateColorTokens;
