// @fluentui/react-image/lib-amd/components/Image/useImageStyles.js
define(["require", "exports", "tslib", "@griffel/react", "@fluentui/react-theme"], function (require, exports, tslib_1, react_1, react_theme_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useImageStyles_unstable = exports.imageClassNames = void 0;
    exports.imageClassNames = {
        root: 'fui-Image',
    };
    var useStyles = react_1.makeStyles({
        // Base styles
        base: tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({}, react_1.shorthands.borderColor(react_theme_1.tokens.colorNeutralStroke1)), react_1.shorthands.borderRadius(react_theme_1.tokens.borderRadiusNone)), { boxSizing: 'border-box', display: 'inline-block' }),
        // Bordered styles
        bordered: tslib_1.__assign(tslib_1.__assign({}, react_1.shorthands.borderStyle('solid')), react_1.shorthands.borderWidth(react_theme_1.tokens.strokeWidthThin)),
        // Shape variations
        circular: tslib_1.__assign({}, react_1.shorthands.borderRadius(react_theme_1.tokens.borderRadiusCircular)),
        rounded: tslib_1.__assign({}, react_1.shorthands.borderRadius(react_theme_1.tokens.borderRadiusMedium)),
        square: {
        /* The square styles are exactly the same as the base styles. */
        },
        // Shadow styles
        shadow: {
            boxShadow: react_theme_1.tokens.shadow4,
        },
        // Fit variations
        center: {
            objectFit: 'none',
            objectPosition: 'center',
            height: '100%',
            width: '100%',
        },
        contain: {
            objectFit: 'contain',
            objectPosition: 'center',
            height: '100%',
            width: '100%',
        },
        default: {
        /* The default styles are exactly the same as the base styles. */
        },
        cover: {
            objectFit: 'cover',
            objectPosition: 'center',
            height: '100%',
            width: '100%',
        },
        none: {
            objectFit: 'none',
            objectPosition: 'left top',
            height: '100%',
            width: '100%',
        },
        // Block styles
        block: {
            width: '100%',
        },
    });
    var useImageStyles_unstable = function (state) {
        var styles = useStyles();
        state.root.className = react_1.mergeClasses(exports.imageClassNames.root, styles.base, state.block && styles.block, state.bordered && styles.bordered, state.shadow && styles.shadow, styles[state.fit], styles[state.shape], state.root.className);
    };
    exports.useImageStyles_unstable = useImageStyles_unstable;
});
