// @fluentui/react-button/lib-commonjs/components/CompoundButton/useCompoundButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useCompoundButton_unstable = void 0;

const react_utilities_1 = /*#__PURE__*/require("@fluentui/react-utilities");

const index_1 = /*#__PURE__*/require("../Button/index");
/**
 * Given user props, defines default props for the CompoundButton, calls useButtonState, and returns processed state.
 * @param props - User provided props to the CompoundButton component.
 * @param ref - User provided ref to be passed to the CompoundButton component.
 */


const useCompoundButton_unstable = ({
  contentContainer,
  secondaryContent,
  ...props
}, ref) => {
  var _a, _b;

  const state = { // Button state
    ...index_1.useButton_unstable(props, ref),
    // Slots definition
    components: {
      root: 'button',
      icon: 'span',
      contentContainer: 'span',
      secondaryContent: 'span'
    },
    contentContainer: react_utilities_1.resolveShorthand(contentContainer, {
      required: true
    }),
    secondaryContent: react_utilities_1.resolveShorthand(secondaryContent)
  }; // Recalculate iconOnly to take into account secondaryContent.

  state.iconOnly = Boolean(((_a = state.icon) === null || _a === void 0 ? void 0 : _a.children) && !props.children && !((_b = state.secondaryContent) === null || _b === void 0 ? void 0 : _b.children));
  return state;
};

exports.useCompoundButton_unstable = useCompoundButton_unstable;
