// @fluentui/react-provider/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useFluentProviderThemeStyleTag = exports.useFluentProviderStyles_unstable = exports.useFluentProvider_unstable = exports.useFluentProviderContextValues_unstable = exports.renderFluentProvider_unstable = exports.FluentProvider = exports.fluentProviderClassNames = void 0;

var FluentProvider_1 = /*#__PURE__*/require("./FluentProvider");

Object.defineProperty(exports, "fluentProviderClassNames", {
  enumerable: true,
  get: function () {
    return FluentProvider_1.fluentProviderClassNames;
  }
});
Object.defineProperty(exports, "FluentProvider", {
  enumerable: true,
  get: function () {
    return FluentProvider_1.FluentProvider;
  }
});
Object.defineProperty(exports, "renderFluentProvider_unstable", {
  enumerable: true,
  get: function () {
    return FluentProvider_1.renderFluentProvider_unstable;
  }
});
Object.defineProperty(exports, "useFluentProviderContextValues_unstable", {
  enumerable: true,
  get: function () {
    return FluentProvider_1.useFluentProviderContextValues_unstable;
  }
});
Object.defineProperty(exports, "useFluentProvider_unstable", {
  enumerable: true,
  get: function () {
    return FluentProvider_1.useFluentProvider_unstable;
  }
});
Object.defineProperty(exports, "useFluentProviderStyles_unstable", {
  enumerable: true,
  get: function () {
    return FluentProvider_1.useFluentProviderStyles_unstable;
  }
});
Object.defineProperty(exports, "useFluentProviderThemeStyleTag", {
  enumerable: true,
  get: function () {
    return FluentProvider_1.useFluentProviderThemeStyleTag;
  }
});
