// @fluentui/react-menu/lib-commonjs/components/MenuItemCheckbox/useMenuItemCheckboxStyles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useMenuItemCheckboxStyles_unstable = exports.menuItemCheckboxClassNames = void 0;

const react_1 = /*#__PURE__*/require("@griffel/react");

const index_1 = /*#__PURE__*/require("../../selectable/index");

const useMenuItemStyles_1 = /*#__PURE__*/require("../MenuItem/useMenuItemStyles");

exports.menuItemCheckboxClassNames = {
  root: 'fui-MenuItemCheckbox',
  icon: 'fui-MenuItemCheckbox__icon',
  checkmark: 'fui-MenuItemCheckbox__checkmark',
  content: 'fui-MenuItemCheckbox__content',
  secondaryContent: 'fui-MenuItemCheckbox__secondaryContent'
};

const useMenuItemCheckboxStyles_unstable = state => {
  state.root.className = react_1.mergeClasses(exports.menuItemCheckboxClassNames.root, state.root.className);

  if (state.content) {
    state.content.className = react_1.mergeClasses(exports.menuItemCheckboxClassNames.content, state.content.className);
  }

  if (state.secondaryContent) {
    state.secondaryContent.className = react_1.mergeClasses(exports.menuItemCheckboxClassNames.secondaryContent, state.secondaryContent.className);
  }

  if (state.icon) {
    state.icon.className = react_1.mergeClasses(exports.menuItemCheckboxClassNames.icon, state.icon.className);
  }

  if (state.checkmark) {
    state.checkmark.className = react_1.mergeClasses(exports.menuItemCheckboxClassNames.checkmark, state.checkmark.className);
  }

  useMenuItemStyles_1.useMenuItemStyles_unstable(state);
  index_1.useCheckmarkStyles_unstable(state);
};

exports.useMenuItemCheckboxStyles_unstable = useMenuItemCheckboxStyles_unstable;
