// @fluentui/react-table/lib-commonjs/components/TableHeader/useTableHeaderStyles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useTableHeaderStyles_unstable = exports.tableHeaderClassNames = exports.tableHeaderClassName = void 0;

const react_1 = /*#__PURE__*/require("@griffel/react");

exports.tableHeaderClassName = 'fui-TableHeader';
exports.tableHeaderClassNames = {
  root: 'fui-TableHeader'
};

const useFlexLayoutStyles = /*#__PURE__*/react_1.__styles({
  "root": {
    "mc9l5x": "ftgm304"
  },
  "roottable": {
    "mc9l5x": "f1tp1avn"
  },
  "rootFlex": {
    "mc9l5x": "ftgm304"
  }
}, {
  "d": [".ftgm304{display:block;}", ".f1tp1avn{display:table-row-group;}"]
});

const useTableLayoutStyles = /*#__PURE__*/react_1.__styles({
  "root": {
    "mc9l5x": "f1tp1avn"
  }
}, {
  "d": [".f1tp1avn{display:table-row-group;}"]
});
/**
 * Apply styling to the TableHeader slots based on the state
 */


const useTableHeaderStyles_unstable = state => {
  const layoutStyles = {
    table: useTableLayoutStyles(),
    flex: useFlexLayoutStyles()
  };
  state.root.className = react_1.mergeClasses(exports.tableHeaderClassName, state.noNativeElements ? layoutStyles.flex.root : layoutStyles.table.root, state.root.className);
  return state;
};

exports.useTableHeaderStyles_unstable = useTableHeaderStyles_unstable;
