// @fluentui/react-image/lib-amd/components/Image/useImage.js
define(["require", "exports", "tslib", "@fluentui/react-utilities"], function (require, exports, tslib_1, react_utilities_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useImage_unstable = void 0;
    /**
     * Given user props, returns state and render function for an Image.
     */
    var useImage_unstable = function (props, ref) {
        var _a = props.bordered, bordered = _a === void 0 ? false : _a, _b = props.fit, fit = _b === void 0 ? 'default' : _b, _c = props.block, block = _c === void 0 ? false : _c, _d = props.shape, shape = _d === void 0 ? 'square' : _d, _e = props.shadow, shadow = _e === void 0 ? false : _e;
        var state = {
            bordered: bordered,
            fit: fit,
            block: block,
            shape: shape,
            shadow: shadow,
            components: {
                root: 'img',
            },
            root: react_utilities_1.getNativeElementProps('img', tslib_1.__assign({ ref: ref }, props)),
        };
        return state;
    };
    exports.useImage_unstable = useImage_unstable;
});
