// @fluentui/react-button/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useToggleState = exports.useToggleButton_unstable = exports.useToggleButtonStyles_unstable = exports.toggleButtonClassNames = exports.renderToggleButton_unstable = exports.ToggleButton = exports.useSplitButton_unstable = exports.useSplitButtonStyles_unstable = exports.splitButtonClassNames = exports.renderSplitButton_unstable = exports.SplitButton = exports.useMenuButton_unstable = exports.useMenuButtonStyles_unstable = exports.renderMenuButton_unstable = exports.menuButtonClassNames = exports.MenuButton = exports.useCompoundButton_unstable = exports.useCompoundButtonStyles_unstable = exports.renderCompoundButton_unstable = exports.compoundButtonClassNames = exports.CompoundButton = exports.useButton_unstable = exports.useButtonStyles_unstable = exports.renderButton_unstable = exports.buttonClassNames = exports.Button = void 0;

var Button_1 = /*#__PURE__*/require("./Button");

Object.defineProperty(exports, "Button", {
  enumerable: true,
  get: function () {
    return Button_1.Button;
  }
});
Object.defineProperty(exports, "buttonClassNames", {
  enumerable: true,
  get: function () {
    return Button_1.buttonClassNames;
  }
});
Object.defineProperty(exports, "renderButton_unstable", {
  enumerable: true,
  get: function () {
    return Button_1.renderButton_unstable;
  }
});
Object.defineProperty(exports, "useButtonStyles_unstable", {
  enumerable: true,
  get: function () {
    return Button_1.useButtonStyles_unstable;
  }
});
Object.defineProperty(exports, "useButton_unstable", {
  enumerable: true,
  get: function () {
    return Button_1.useButton_unstable;
  }
});

var CompoundButton_1 = /*#__PURE__*/require("./CompoundButton");

Object.defineProperty(exports, "CompoundButton", {
  enumerable: true,
  get: function () {
    return CompoundButton_1.CompoundButton;
  }
});
Object.defineProperty(exports, "compoundButtonClassNames", {
  enumerable: true,
  get: function () {
    return CompoundButton_1.compoundButtonClassNames;
  }
});
Object.defineProperty(exports, "renderCompoundButton_unstable", {
  enumerable: true,
  get: function () {
    return CompoundButton_1.renderCompoundButton_unstable;
  }
});
Object.defineProperty(exports, "useCompoundButtonStyles_unstable", {
  enumerable: true,
  get: function () {
    return CompoundButton_1.useCompoundButtonStyles_unstable;
  }
});
Object.defineProperty(exports, "useCompoundButton_unstable", {
  enumerable: true,
  get: function () {
    return CompoundButton_1.useCompoundButton_unstable;
  }
});

var MenuButton_1 = /*#__PURE__*/require("./MenuButton");

Object.defineProperty(exports, "MenuButton", {
  enumerable: true,
  get: function () {
    return MenuButton_1.MenuButton;
  }
});
Object.defineProperty(exports, "menuButtonClassNames", {
  enumerable: true,
  get: function () {
    return MenuButton_1.menuButtonClassNames;
  }
});
Object.defineProperty(exports, "renderMenuButton_unstable", {
  enumerable: true,
  get: function () {
    return MenuButton_1.renderMenuButton_unstable;
  }
});
Object.defineProperty(exports, "useMenuButtonStyles_unstable", {
  enumerable: true,
  get: function () {
    return MenuButton_1.useMenuButtonStyles_unstable;
  }
});
Object.defineProperty(exports, "useMenuButton_unstable", {
  enumerable: true,
  get: function () {
    return MenuButton_1.useMenuButton_unstable;
  }
});

var SplitButton_1 = /*#__PURE__*/require("./SplitButton");

Object.defineProperty(exports, "SplitButton", {
  enumerable: true,
  get: function () {
    return SplitButton_1.SplitButton;
  }
});
Object.defineProperty(exports, "renderSplitButton_unstable", {
  enumerable: true,
  get: function () {
    return SplitButton_1.renderSplitButton_unstable;
  }
});
Object.defineProperty(exports, "splitButtonClassNames", {
  enumerable: true,
  get: function () {
    return SplitButton_1.splitButtonClassNames;
  }
});
Object.defineProperty(exports, "useSplitButtonStyles_unstable", {
  enumerable: true,
  get: function () {
    return SplitButton_1.useSplitButtonStyles_unstable;
  }
});
Object.defineProperty(exports, "useSplitButton_unstable", {
  enumerable: true,
  get: function () {
    return SplitButton_1.useSplitButton_unstable;
  }
});

var ToggleButton_1 = /*#__PURE__*/require("./ToggleButton");

Object.defineProperty(exports, "ToggleButton", {
  enumerable: true,
  get: function () {
    return ToggleButton_1.ToggleButton;
  }
});
Object.defineProperty(exports, "renderToggleButton_unstable", {
  enumerable: true,
  get: function () {
    return ToggleButton_1.renderToggleButton_unstable;
  }
});
Object.defineProperty(exports, "toggleButtonClassNames", {
  enumerable: true,
  get: function () {
    return ToggleButton_1.toggleButtonClassNames;
  }
});
Object.defineProperty(exports, "useToggleButtonStyles_unstable", {
  enumerable: true,
  get: function () {
    return ToggleButton_1.useToggleButtonStyles_unstable;
  }
});
Object.defineProperty(exports, "useToggleButton_unstable", {
  enumerable: true,
  get: function () {
    return ToggleButton_1.useToggleButton_unstable;
  }
});

var index_1 = /*#__PURE__*/require("./utils/index");

Object.defineProperty(exports, "useToggleState", {
  enumerable: true,
  get: function () {
    return index_1.useToggleState;
  }
});
