// @fluentui/react-table/lib-commonjs/components/TableBody/useTableBodyStyles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useTableBodyStyles_unstable = exports.tableBodyClassNames = exports.tableBodyClassName = void 0;

const react_1 = /*#__PURE__*/require("@griffel/react");

const useTableLayoutStyles = /*#__PURE__*/react_1.__styles({
  "root": {
    "mc9l5x": "f1tp1avn"
  }
}, {
  "d": [".f1tp1avn{display:table-row-group;}"]
});

const useFlexLayoutStyles = /*#__PURE__*/react_1.__styles({
  "root": {
    "mc9l5x": "ftgm304"
  }
}, {
  "d": [".ftgm304{display:block;}"]
});

exports.tableBodyClassName = 'fui-TableBody';
exports.tableBodyClassNames = {
  root: 'fui-TableBody'
};
/**
 * Apply styling to the TableBody slots based on the state
 */

const useTableBodyStyles_unstable = state => {
  const layoutStyles = {
    table: useTableLayoutStyles(),
    flex: useFlexLayoutStyles()
  };
  state.root.className = react_1.mergeClasses(exports.tableBodyClassName, state.noNativeElements ? layoutStyles.flex.root : layoutStyles.table.root, state.root.className);
  return state;
};

exports.useTableBodyStyles_unstable = useTableBodyStyles_unstable;
