// @fluentui/react-positioning/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.mergeArrowOffset = exports.resolvePositioningShorthand = exports.usePositioningMouseTarget = exports.usePositioning = exports.createArrowStyles = exports.createArrowHeightStyles = exports.createVirtualElementFromClick = void 0;

var createVirtualElementFromClick_1 = /*#__PURE__*/require("./createVirtualElementFromClick");

Object.defineProperty(exports, "createVirtualElementFromClick", {
  enumerable: true,
  get: function () {
    return createVirtualElementFromClick_1.createVirtualElementFromClick;
  }
});

var createArrowStyles_1 = /*#__PURE__*/require("./createArrowStyles");

Object.defineProperty(exports, "createArrowHeightStyles", {
  enumerable: true,
  get: function () {
    return createArrowStyles_1.createArrowHeightStyles;
  }
});
Object.defineProperty(exports, "createArrowStyles", {
  enumerable: true,
  get: function () {
    return createArrowStyles_1.createArrowStyles;
  }
});

var usePositioning_1 = /*#__PURE__*/require("./usePositioning");

Object.defineProperty(exports, "usePositioning", {
  enumerable: true,
  get: function () {
    return usePositioning_1.usePositioning;
  }
});

var usePositioningMouseTarget_1 = /*#__PURE__*/require("./usePositioningMouseTarget");

Object.defineProperty(exports, "usePositioningMouseTarget", {
  enumerable: true,
  get: function () {
    return usePositioningMouseTarget_1.usePositioningMouseTarget;
  }
});

var index_1 = /*#__PURE__*/require("./utils/index");

Object.defineProperty(exports, "resolvePositioningShorthand", {
  enumerable: true,
  get: function () {
    return index_1.resolvePositioningShorthand;
  }
});
Object.defineProperty(exports, "mergeArrowOffset", {
  enumerable: true,
  get: function () {
    return index_1.mergeArrowOffset;
  }
});
