// @fluentui/react-table/lib-commonjs/components/TableCellLayout/useTableCellLayoutStyles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useTableCellLayoutStyles_unstable = exports.tableCellLayoutClassNames = void 0;

const react_1 = /*#__PURE__*/require("@griffel/react");

const react_theme_1 = /*#__PURE__*/require("@fluentui/react-theme");

const react_theme_2 = /*#__PURE__*/require("@fluentui/react-theme");

exports.tableCellLayoutClassNames = {
  root: 'fui-TableCellLayout',
  media: 'fui-TableCellLayout__media',
  main: 'fui-TableCellLayout__main',
  description: 'fui-TableCellLayout__description',
  wrapper: 'fui-TableCellLayout__wrapper'
};
/**
 * Styles for the root slot
 */

const useStyles = /*#__PURE__*/react_1.__styles({
  "root": {
    "mc9l5x": "f22iagw",
    "Bt984gj": "f122n59",
    "i8kkvl": "fsnqrgy",
    "Belr9w4": "fylz90v",
    "Bh6795r": "fqerorx",
    "Bnnss6s": "f1neuvcm",
    "xawz": "fkjuxzh"
  },
  "wrapper": {
    "mc9l5x": "f22iagw",
    "Beiy3e4": "f1vx9l62"
  },
  "media": {
    "mc9l5x": "f22iagw",
    "Bt984gj": "f122n59"
  },
  "mediaExtraSmall": {
    "Be2twd7": "f4ybsrx"
  },
  "mediaSmallAndMedium": {
    "Be2twd7": "fe5j1ua"
  },
  "mediaPrimary": {
    "Be2twd7": "f1rt2boy"
  },
  "mainPrimary": {
    "Bhrd7zp": "fl43uef"
  },
  "description": {
    "sj55zd": "fkfq4zb",
    "Bahqtrf": "fk6fouc",
    "Be2twd7": "fy9rknc",
    "Bhrd7zp": "figsok6",
    "Bg96gwp": "fwrc4pm"
  }
}, {
  "d": [".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}", ".f122n59{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}", ".fsnqrgy{-webkit-column-gap:var(--spacingHorizontalS);column-gap:var(--spacingHorizontalS);}", ".fylz90v{row-gap:var(--spacingHorizontalS);}", ".fqerorx{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}", ".f1neuvcm{-webkit-flex-shrink:1;-ms-flex-negative:1;flex-shrink:1;}", ".fkjuxzh{-webkit-flex-basis:0px;-ms-flex-preferred-size:0px;flex-basis:0px;}", ".f1vx9l62{-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}", ".f4ybsrx{font-size:16px;}", ".fe5j1ua{font-size:20px;}", ".f1rt2boy{font-size:24px;}", ".fl43uef{font-weight:var(--fontWeightSemibold);}", ".fkfq4zb{color:var(--colorNeutralForeground2);}", ".fk6fouc{font-family:var(--fontFamilyBase);}", ".fy9rknc{font-size:var(--fontSizeBase200);}", ".figsok6{font-weight:var(--fontWeightRegular);}", ".fwrc4pm{line-height:var(--lineHeightBase200);}"]
});
/**
 * Apply styling to the TableCellLayout slots based on the state
 */


const useTableCellLayoutStyles_unstable = state => {
  const styles = useStyles();
  state.root.className = react_1.mergeClasses(exports.tableCellLayoutClassNames.root, styles.root, state.root.className);
  const primary = state.appearance === 'primary';

  if (state.media) {
    const mediaSizedStyles = {
      small: styles.mediaSmallAndMedium,
      medium: styles.mediaSmallAndMedium,
      'extra-small': styles.mediaExtraSmall
    };
    state.media.className = react_1.mergeClasses(exports.tableCellLayoutClassNames.media, styles.media, mediaSizedStyles[state.size], primary && styles.mediaPrimary, state.media.className);
  }

  if (state.main) {
    state.main.className = react_1.mergeClasses(exports.tableCellLayoutClassNames.main, primary && styles.mainPrimary, state.main.className);
  }

  if (state.description) {
    state.description.className = react_1.mergeClasses(exports.tableCellLayoutClassNames.description, styles.description, state.description.className);
  }

  if (state.wrapper) {
    state.wrapper.className = react_1.mergeClasses(exports.tableCellLayoutClassNames.wrapper, styles.wrapper, state.wrapper.className);
  }

  return state;
};

exports.useTableCellLayoutStyles_unstable = useTableCellLayoutStyles_unstable;
