// @fluentui/react-table/lib-commonjs/components/TableCellLayout/useTableCellLayout.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useTableCellLayout_unstable = void 0;

const react_utilities_1 = /*#__PURE__*/require("@fluentui/react-utilities");

const tableContext_1 = /*#__PURE__*/require("../../contexts/tableContext");

const tableAvatarSizeMap = {
  medium: 32,
  small: 24,
  'extra-small': 20
};
/**
 * Create the state required to render TableCellLayout.
 *
 * The returned state can be modified with hooks such as useTableCellLayoutStyles_unstable,
 * before being passed to renderTableCellLayout_unstable.
 *
 * @param props - props from this instance of TableCellLayout
 * @param ref - reference to root HTMLElement of TableCellLayout
 */

const useTableCellLayout_unstable = (props, ref) => {
  const {
    size
  } = tableContext_1.useTableContext();
  return {
    components: {
      root: 'div',
      main: 'span',
      description: 'span',
      wrapper: 'div',
      media: 'span'
    },
    root: react_utilities_1.getNativeElementProps('div', {
      ref,
      ...props
    }),
    appearance: props.appearance,
    main: react_utilities_1.resolveShorthand(props.main, {
      required: true
    }),
    media: react_utilities_1.resolveShorthand(props.media),
    description: react_utilities_1.resolveShorthand(props.description),
    wrapper: react_utilities_1.resolveShorthand(props.wrapper, {
      required: !!props.description || !!props.children
    }),
    avatarSize: tableAvatarSizeMap[size],
    size
  };
};

exports.useTableCellLayout_unstable = useTableCellLayout_unstable;
