// @fluentui/react-menu/lib-amd/components/MenuSplitGroup/useMenuSplitGroupStyles.js
define(["require", "exports", "@griffel/react", "@fluentui/react-theme", "../MenuItem/useMenuItemStyles"], function (require, exports, react_1, react_theme_1, useMenuItemStyles_1) {
    "use strict";
    var _a;
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useMenuSplitGroupStyles_unstable = exports.menuSplitGroupClassNames = void 0;
    exports.menuSplitGroupClassNames = {
        root: 'fui-MenuSplitGroup',
    };
    /**
     * Styles for the root slot
     */
    var useStyles = react_1.makeStyles({
        root: (_a = {
                display: 'flex'
            },
            _a["& > ." + useMenuItemStyles_1.menuItemClassNames.root + ":nth-child(1)"] = {
                width: '100%',
            },
            _a["& > ." + useMenuItemStyles_1.menuItemClassNames.root + ":nth-child(2)"] = {
                borderTopLeftRadius: 0,
                borderBottomLeftRadius: 0,
                paddingLeft: 0,
                marginLeft: 'auto',
                '::before': {
                    content: '""',
                    width: react_theme_1.tokens.strokeWidthThin,
                    height: '24px',
                    backgroundColor: react_theme_1.tokens.colorNeutralStroke1,
                },
            },
            _a),
    });
    /**
     * Apply styling to the MenuSplitGroup slots based on the state
     */
    var useMenuSplitGroupStyles_unstable = function (state) {
        var styles = useStyles();
        state.root.className = react_1.mergeClasses(exports.menuSplitGroupClassNames.root, styles.root, state.root.className);
        return state;
    };
    exports.useMenuSplitGroupStyles_unstable = useMenuSplitGroupStyles_unstable;
});
