// @fluentui/react-input/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.inputFieldClassNames = exports.InputField_unstable = exports.useInput_unstable = exports.useInputStyles_unstable = exports.renderInput_unstable = exports.inputClassNames = exports.Input = void 0;

var Input_1 = /*#__PURE__*/require("./Input");

Object.defineProperty(exports, "Input", {
  enumerable: true,
  get: function () {
    return Input_1.Input;
  }
});
Object.defineProperty(exports, "inputClassNames", {
  enumerable: true,
  get: function () {
    return Input_1.inputClassNames;
  }
});
Object.defineProperty(exports, "renderInput_unstable", {
  enumerable: true,
  get: function () {
    return Input_1.renderInput_unstable;
  }
});
Object.defineProperty(exports, "useInputStyles_unstable", {
  enumerable: true,
  get: function () {
    return Input_1.useInputStyles_unstable;
  }
});
Object.defineProperty(exports, "useInput_unstable", {
  enumerable: true,
  get: function () {
    return Input_1.useInput_unstable;
  }
});

var InputField_1 = /*#__PURE__*/require("./InputField");

Object.defineProperty(exports, "InputField_unstable", {
  enumerable: true,
  get: function () {
    return InputField_1.InputField;
  }
});
Object.defineProperty(exports, "inputFieldClassNames", {
  enumerable: true,
  get: function () {
    return InputField_1.inputFieldClassNames;
  }
});
