// @fluentui/react-card/lib/components/CardHeader/useCardHeaderStyles.js
import { __styles, mergeClasses } from '@griffel/react';
/**
 * Static CSS class names used internally for the component slots.
 */

export const cardHeaderClassNames = {
  root: 'fui-CardHeader',
  image: 'fui-CardHeader__image',
  header: 'fui-CardHeader__header',
  description: 'fui-CardHeader__description',
  action: 'fui-CardHeader__action'
};
/**
 * CSS variable names used internally for uniform styling in CardHeader.
 */

export const cardHeaderCSSVars = {
  cardHeaderGapVar: '--fui-CardHeader--gap'
};

const useStyles = /*#__PURE__*/__styles({
  "root": {
    "Bkc6ea2": "fkufhic",
    "mc9l5x": "f13qh94s",
    "t4k1zu": "f8a668j",
    "v29qe6": "f4h39za",
    "Bt984gj": "f122n59"
  },
  "image": {
    "mc9l5x": "ftuwxu6",
    "t21cq0": ["fql5097", "f6yss9k"],
    "Br312pm": "fwpfdsa",
    "Ijaq50": "fldnz9j"
  },
  "header": {
    "Br312pm": "fd46tj4",
    "Ijaq50": "f16hsg94"
  },
  "description": {
    "Br312pm": "fd46tj4",
    "Ijaq50": "faunodf"
  },
  "action": {
    "Frg6f3": ["f6yss9k", "fql5097"],
    "Br312pm": "fis13di",
    "Ijaq50": "fldnz9j"
  }
}, {
  "d": [".fkufhic{--fui-CardHeader--gap:12px;}", ".f13qh94s{display:grid;}", ".f8a668j{grid-auto-columns:min-content 1fr min-content;}", ".f4h39za{grid-auto-rows:1fr min-content;}", ".f122n59{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}", ".ftuwxu6{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}", ".fql5097{margin-right:var(--fui-CardHeader--gap);}", ".f6yss9k{margin-left:var(--fui-CardHeader--gap);}", ".fwpfdsa{grid-column-start:1;}", ".fldnz9j{grid-row-start:span 2;}", ".fd46tj4{grid-column-start:2;}", ".f16hsg94{grid-row-start:1;}", ".faunodf{grid-row-start:2;}", ".fis13di{grid-column-start:3;}"]
});
/**
 * Apply styling to the CardHeader slots based on the state.
 */


export const useCardHeaderStyles_unstable = state => {
  const styles = useStyles();
  state.root.className = mergeClasses(cardHeaderClassNames.root, styles.root, state.root.className);

  if (state.image) {
    state.image.className = mergeClasses(cardHeaderClassNames.image, styles.image, state.image.className);
  }

  if (state.header) {
    state.header.className = mergeClasses(cardHeaderClassNames.header, styles.header, state.header.className);
  }

  if (state.description) {
    state.description.className = mergeClasses(cardHeaderClassNames.description, styles.description, state.description.className);
  }

  if (state.action) {
    state.action.className = mergeClasses(cardHeaderClassNames.action, styles.action, state.action.className);
  }

  return state;
};
