// @fluentui/react-accordion/lib-amd/components/AccordionItem/useAccordionItem.js
define(["require", "exports", "tslib", "react", "@fluentui/react-utilities", "../Accordion/AccordionContext"], function (require, exports, tslib_1, React, react_utilities_1, AccordionContext_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useAccordionItem_unstable = void 0;
    /**
     * Returns the props and state required to render the component
     * @param props - AccordionItem properties
     * @param ref - reference to root HTMLElement of AccordionItem
     */
    var useAccordionItem_unstable = function (props, ref) {
        var value = props.value, _a = props.disabled, disabled = _a === void 0 ? false : _a;
        var requestToggle = AccordionContext_1.useAccordionContext_unstable(function (ctx) { return ctx.requestToggle; });
        var open = AccordionContext_1.useAccordionContext_unstable(function (ctx) { return ctx.openItems.includes(value); });
        var onAccordionHeaderClick = React.useCallback(function (ev) { return requestToggle(ev, { value: value }); }, [
            requestToggle,
            value,
        ]);
        return {
            open: open,
            disabled: disabled,
            onHeaderClick: onAccordionHeaderClick,
            components: {
                root: 'div',
            },
            root: react_utilities_1.getNativeElementProps('div', tslib_1.__assign({ ref: ref }, props)),
        };
    };
    exports.useAccordionItem_unstable = useAccordionItem_unstable;
});
