// @fluentui/react-avatar/lib-amd/components/AvatarGroup/useAvatarGroupStyles.js
define(["require", "exports", "@griffel/react", "@fluentui/react-theme", "../Avatar/useAvatarStyles"], function (require, exports, react_1, react_theme_1, useAvatarStyles_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useAvatarGroupStyles_unstable = exports.avatarGroupClassNames = void 0;
    exports.avatarGroupClassNames = {
        root: 'fui-AvatarGroup',
    };
    /**
     * Styles for the root slot.
     */
    var useStyles = react_1.makeStyles({
        base: {
            display: 'inline-flex',
            position: 'relative',
        },
        pie: {
            clipPath: 'circle(50%)',
            backgroundColor: react_theme_1.tokens.colorTransparentStroke,
            '@media (forced-colors: active)': {
                backgroundColor: 'CanvasText',
            },
        },
    });
    /**
     * Apply styling to the AvatarGroup slots based on the state
     */
    var useAvatarGroupStyles_unstable = function (state) {
        var layout = state.layout, size = state.size;
        var styles = useStyles();
        var sizeStyles = useAvatarStyles_1.useSizeStyles();
        state.root.className = react_1.mergeClasses(exports.avatarGroupClassNames.root, styles.base, layout === 'pie' && sizeStyles[size], layout === 'pie' && styles.pie, state.root.className);
        return state;
    };
    exports.useAvatarGroupStyles_unstable = useAvatarGroupStyles_unstable;
});
