// @fluentui/react-tabs/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useTabList_unstable = exports.useTabListStyles_unstable = exports.tabListClassNames = exports.TabList = exports.renderTabList_unstable = exports.useTab_unstable = exports.useTabStyles_unstable = exports.tabClassNames = exports.Tab = exports.renderTab_unstable = void 0;

var Tab_1 = /*#__PURE__*/require("./Tab");

Object.defineProperty(exports, "renderTab_unstable", {
  enumerable: true,
  get: function () {
    return Tab_1.renderTab_unstable;
  }
});
Object.defineProperty(exports, "Tab", {
  enumerable: true,
  get: function () {
    return Tab_1.Tab;
  }
});
Object.defineProperty(exports, "tabClassNames", {
  enumerable: true,
  get: function () {
    return Tab_1.tabClassNames;
  }
});
Object.defineProperty(exports, "useTabStyles_unstable", {
  enumerable: true,
  get: function () {
    return Tab_1.useTabStyles_unstable;
  }
});
Object.defineProperty(exports, "useTab_unstable", {
  enumerable: true,
  get: function () {
    return Tab_1.useTab_unstable;
  }
});

var TabList_1 = /*#__PURE__*/require("./TabList");

Object.defineProperty(exports, "renderTabList_unstable", {
  enumerable: true,
  get: function () {
    return TabList_1.renderTabList_unstable;
  }
});
Object.defineProperty(exports, "TabList", {
  enumerable: true,
  get: function () {
    return TabList_1.TabList;
  }
});
Object.defineProperty(exports, "tabListClassNames", {
  enumerable: true,
  get: function () {
    return TabList_1.tabListClassNames;
  }
});
Object.defineProperty(exports, "useTabListStyles_unstable", {
  enumerable: true,
  get: function () {
    return TabList_1.useTabListStyles_unstable;
  }
});
Object.defineProperty(exports, "useTabList_unstable", {
  enumerable: true,
  get: function () {
    return TabList_1.useTabList_unstable;
  }
});
