// @fluentui/react-button/lib-amd/components/CompoundButton/useCompoundButton.js
define(["require", "exports", "tslib", "@fluentui/react-utilities", "../Button/index"], function (require, exports, tslib_1, react_utilities_1, index_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useCompoundButton_unstable = void 0;
    /**
     * Given user props, defines default props for the CompoundButton, calls useButtonState, and returns processed state.
     * @param props - User provided props to the CompoundButton component.
     * @param ref - User provided ref to be passed to the CompoundButton component.
     */
    var useCompoundButton_unstable = function (_a, ref) {
        var _b, _c;
        var contentContainer = _a.contentContainer, secondaryContent = _a.secondaryContent, props = tslib_1.__rest(_a, ["contentContainer", "secondaryContent"]);
        var state = tslib_1.__assign(tslib_1.__assign({}, index_1.useButton_unstable(props, ref)), { 
            // Slots definition
            components: {
                root: 'button',
                icon: 'span',
                contentContainer: 'span',
                secondaryContent: 'span',
            }, contentContainer: react_utilities_1.resolveShorthand(contentContainer, { required: true }), secondaryContent: react_utilities_1.resolveShorthand(secondaryContent) });
        // Recalculate iconOnly to take into account secondaryContent.
        state.iconOnly = Boolean(((_b = state.icon) === null || _b === void 0 ? void 0 : _b.children) && !props.children && !((_c = state.secondaryContent) === null || _c === void 0 ? void 0 : _c.children));
        return state;
    };
    exports.useCompoundButton_unstable = useCompoundButton_unstable;
});
