// @fluentui/react-popover/lib-amd/popoverContext.js
define(["require", "exports", "@fluentui/react-context-selector"], function (require, exports, react_context_selector_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.usePopoverContext_unstable = exports.PopoverProvider = exports.PopoverContext = void 0;
    exports.PopoverContext = react_context_selector_1.createContext(undefined);
    var popoverContextDefaultValue = {
        open: false,
        setOpen: function () { return null; },
        toggleOpen: function () { return null; },
        triggerRef: { current: null },
        contentRef: { current: null },
        arrowRef: { current: null },
        openOnContext: false,
        openOnHover: false,
        size: 'medium',
        trapFocus: false,
        inline: false,
    };
    exports.PopoverProvider = exports.PopoverContext.Provider;
    var usePopoverContext_unstable = function (selector) {
        return react_context_selector_1.useContextSelector(exports.PopoverContext, function (ctx) {
            if (ctx === void 0) { ctx = popoverContextDefaultValue; }
            return selector(ctx);
        });
    };
    exports.usePopoverContext_unstable = usePopoverContext_unstable;
});
