// @fluentui/react-tabster/lib-commonjs/focus/createFocusOutlineStyle.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.createFocusOutlineStyle = void 0;

const react_theme_1 = /*#__PURE__*/require("@fluentui/react-theme");

const react_1 = /*#__PURE__*/require("@griffel/react");

const createCustomFocusIndicatorStyle_1 = /*#__PURE__*/require("./createCustomFocusIndicatorStyle");

const constants_1 = /*#__PURE__*/require("./constants");
/**
 * NOTE: the element with the focus outline needs to have `position: relative` so that the
 * pseudo element can be properly positioned.
 *
 * @param options - Configures the style of the focus outline
 * @returns focus outline styles object
 */


const getFocusOutlineStyles = options => {
  var _a, _b, _c, _d;

  const {
    outlineRadius,
    outlineColor,
    outlineOffset,
    outlineWidth
  } = options;
  const outlineOffsetTop = ((_a = outlineOffset) === null || _a === void 0 ? void 0 : _a.top) || outlineOffset;
  const outlineOffsetBottom = ((_b = outlineOffset) === null || _b === void 0 ? void 0 : _b.bottom) || outlineOffset;
  const outlineOffsetLeft = ((_c = outlineOffset) === null || _c === void 0 ? void 0 : _c.left) || outlineOffset;
  const outlineOffsetRight = ((_d = outlineOffset) === null || _d === void 0 ? void 0 : _d.right) || outlineOffset;
  return { ...react_1.shorthands.borderColor('transparent'),
    '::after': {
      content: '""',
      position: 'absolute',
      pointerEvents: 'none',
      zIndex: 1,
      ...react_1.shorthands.borderStyle('solid'),
      ...react_1.shorthands.borderWidth(outlineWidth),
      ...react_1.shorthands.borderRadius(outlineRadius),
      ...react_1.shorthands.borderColor(outlineColor),
      top: !outlineOffset ? `-${outlineWidth}` : `calc(0px - ${outlineWidth} - ${outlineOffsetTop})`,
      bottom: !outlineOffset ? `-${outlineWidth}` : `calc(0px - ${outlineWidth} - ${outlineOffsetBottom})`,
      left: !outlineOffset ? `-${outlineWidth}` : `calc(0px - ${outlineWidth} - ${outlineOffsetLeft})`,
      right: !outlineOffset ? `-${outlineWidth}` : `calc(0px - ${outlineWidth} - ${outlineOffsetRight})`
    }
  };
};
/**
 * NOTE: The element with the focus outline needs to have `position: relative` so that the
 * pseudo element can be properly positioned.
 *
 * @param options - Configure the style of the focus outline
 * @returns focus outline styles object for @see makeStyles
 */


const createFocusOutlineStyle = ({
  selector = constants_1.defaultOptions.selector,
  style = constants_1.defaultOptions.style
} = constants_1.defaultOptions) => createCustomFocusIndicatorStyle_1.createCustomFocusIndicatorStyle(getFocusOutlineStyles({
  outlineColor: react_theme_1.tokens.colorStrokeFocus2,
  outlineRadius: react_theme_1.tokens.borderRadiusMedium,
  // FIXME: tokens.strokeWidthThick causes some weird bugs
  outlineWidth: '2px',
  ...style
}), {
  selector
});

exports.createFocusOutlineStyle = createFocusOutlineStyle;
