// @fluentui/react-tabs/lib-commonjs/components/Tab/useTab.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useTab_unstable = void 0;

const React = /*#__PURE__*/require("react");

const react_utilities_1 = /*#__PURE__*/require("@fluentui/react-utilities");

const TabListContext_1 = /*#__PURE__*/require("../TabList/TabListContext");

const react_context_selector_1 = /*#__PURE__*/require("@fluentui/react-context-selector");
/**
 * Create the state required to render Tab.
 *
 * The returned state can be modified with hooks such as useTabStyles_unstable,
 * before being passed to renderTab_unstable.
 *
 * @param props - props from this instance of Tab
 * @param ref - reference to root HTMLElement of Tab
 */


const useTab_unstable = (props, ref) => {
  const {
    content,
    disabled: tabDisabled = false,
    icon,
    value
  } = props;
  const appearance = react_context_selector_1.useContextSelector(TabListContext_1.TabListContext, ctx => ctx.appearance);
  const reserveSelectedTabSpace = react_context_selector_1.useContextSelector(TabListContext_1.TabListContext, ctx => ctx.reserveSelectedTabSpace);
  const listDisabled = react_context_selector_1.useContextSelector(TabListContext_1.TabListContext, ctx => ctx.disabled);
  const selected = react_context_selector_1.useContextSelector(TabListContext_1.TabListContext, ctx => ctx.selectedValue === value);
  const onRegister = react_context_selector_1.useContextSelector(TabListContext_1.TabListContext, ctx => ctx.onRegister);
  const onUnregister = react_context_selector_1.useContextSelector(TabListContext_1.TabListContext, ctx => ctx.onUnregister);
  const onSelect = react_context_selector_1.useContextSelector(TabListContext_1.TabListContext, ctx => ctx.onSelect);
  const size = react_context_selector_1.useContextSelector(TabListContext_1.TabListContext, ctx => ctx.size);
  const vertical = react_context_selector_1.useContextSelector(TabListContext_1.TabListContext, ctx => !!ctx.vertical);
  const disabled = listDisabled || tabDisabled;
  const innerRef = React.useRef(null);
  const onClick = react_utilities_1.useEventCallback(event => onSelect(event, {
    value
  }));
  React.useEffect(() => {
    onRegister({
      value,
      ref: innerRef
    });
    return () => {
      onUnregister({
        value,
        ref: innerRef
      });
    };
  }, [onRegister, onUnregister, innerRef, value]);
  const iconShorthand = react_utilities_1.resolveShorthand(icon);
  const contentShorthand = react_utilities_1.resolveShorthand(content, {
    required: true,
    defaultProps: {
      children: props.children
    }
  });
  return {
    components: {
      root: 'button',
      icon: 'span',
      content: 'span'
    },
    root: react_utilities_1.getNativeElementProps('button', {
      ref: react_utilities_1.useMergedRefs(ref, innerRef),
      role: 'tab',
      type: 'button',
      // aria-selected undefined indicates it is not selectable
      // according to https://www.w3.org/TR/wai-aria-1.1/#aria-selected
      'aria-selected': disabled ? undefined : `${selected}`,
      ...props,
      disabled,
      onClick
    }),
    icon: iconShorthand,
    iconOnly: Boolean((iconShorthand === null || iconShorthand === void 0 ? void 0 : iconShorthand.children) && !contentShorthand.children),
    content: contentShorthand,
    appearance,
    contentReservedSpaceClassName: reserveSelectedTabSpace ? '' : undefined,
    disabled,
    selected,
    size,
    value,
    vertical
  };
};

exports.useTab_unstable = useTab_unstable;
