// @fluentui/react-persona/lib-commonjs/components/Persona/usePersonaStyles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.usePersonaStyles_unstable = exports.personaClassNames = void 0;

const react_1 = /*#__PURE__*/require("@griffel/react");

const react_theme_1 = /*#__PURE__*/require("@fluentui/react-theme");

exports.personaClassNames = {
  root: 'fui-Persona',
  avatar: 'fui-Persona__avatar',
  presence: 'fui-Persona__presence',
  primaryText: 'fui-Persona__primaryText',
  secondaryText: 'fui-Persona__secondaryText',
  tertiaryText: 'fui-Persona__tertiaryText',
  quaternaryText: 'fui-Persona__quaternaryText'
};
const avatarSpacing = `--fui-Persona__avatar--spacing`;
/**
 * Styles for the root slot
 */

const useStyles = /*#__PURE__*/react_1.__styles({
  "base": {
    "mc9l5x": "fwk3njj",
    "v29qe6": "fvtn1bl"
  },
  "after": {
    "Bxotwcr": "f1u07yai",
    "B7hvi0a": "f1m2n5bn",
    "Budl1dq": "f1c1ms5d"
  },
  "before": {
    "Bxotwcr": "f1u07yai",
    "B7hvi0a": "f1tll2w5",
    "Budl1dq": "ffvkwdr"
  },
  "below": {
    "B7hvi0a": "f1oiokrs"
  },
  "coin": {
    "Ijaq50": "f1hek2iy"
  },
  "start": {
    "qb2dma": "f9h729m"
  },
  "center": {
    "qb2dma": "f7nlbp4"
  },
  "afterAlignToPrimary": {
    "qb2dma": "f7nlbp4",
    "Ijaq50": "f1rnkkuc",
    "Bw0ie65": "f1warjpf"
  },
  "beforeAlignToPrimary": {
    "qb2dma": "f7nlbp4",
    "Ijaq50": "f1rnkkuc",
    "Br312pm": "fwu52yu"
  }
}, {
  "d": [".fwk3njj{display:inline-grid;}", ".fvtn1bl{grid-auto-rows:max-content;}", ".f1u07yai{grid-auto-flow:column;}", ".f1m2n5bn{justify-items:start;}", ".f1c1ms5d{grid-template-columns:max-content [middle] auto;}", ".f1tll2w5{justify-items:end;}", ".ffvkwdr{grid-template-columns:auto [middle] max-content;}", ".f1oiokrs{justify-items:center;}", ".f1hek2iy{grid-row-start:span 5;}", ".f9h729m{-webkit-align-self:start;-ms-flex-item-align:start;align-self:start;}", ".f7nlbp4{-webkit-align-self:center;-ms-flex-item-align:center;align-self:center;}", ".f1rnkkuc{grid-row-start:unset;}", ".f1warjpf{grid-column-end:middle;}", ".fwu52yu{grid-column-start:middle;}"]
});

const useAvatarSpacingStyles = /*#__PURE__*/react_1.__styles({
  "extra-small": {
    "Bs1gm4r": "f1e48tse"
  },
  "small": {
    "Bs1gm4r": "f18q9vkd"
  },
  "medium": {
    "Bs1gm4r": "f18q9vkd"
  },
  "large": {
    "Bs1gm4r": "fx34bi6"
  },
  "extra-large": {
    "Bs1gm4r": "fx34bi6"
  },
  "huge": {
    "Bs1gm4r": "f1o96qtm"
  },
  "after": {
    "t21cq0": ["f103ycu4", "f1tao51"]
  },
  "below": {
    "jrapky": "fbo7acy"
  },
  "before": {
    "Frg6f3": ["f1tao51", "f103ycu4"]
  }
}, {
  "d": [".f1e48tse{--fui-Persona__avatar--spacing:var(--spacingHorizontalSNudge);}", ".f18q9vkd{--fui-Persona__avatar--spacing:var(--spacingHorizontalS);}", ".fx34bi6{--fui-Persona__avatar--spacing:var(--spacingHorizontalMNudge);}", ".f1o96qtm{--fui-Persona__avatar--spacing:var(--spacingHorizontalM);}", ".f103ycu4{margin-right:var(--fui-Persona__avatar--spacing);}", ".f1tao51{margin-left:var(--fui-Persona__avatar--spacing);}", ".fbo7acy{margin-bottom:var(--fui-Persona__avatar--spacing);}"]
});

const usePresenceSpacingStyles = /*#__PURE__*/react_1.__styles({
  "small": {
    "Bs1gm4r": "f1e48tse"
  }
}, {
  "d": [".f1e48tse{--fui-Persona__avatar--spacing:var(--spacingHorizontalSNudge);}"]
});
/**
 * Apply styling to the Persona slots based on the state
 */


const usePersonaStyles_unstable = state => {
  const {
    presenceOnly,
    size,
    textAlignment,
    textPosition
  } = state;
  const alignToPrimary = presenceOnly && textAlignment === 'start' && size !== 'extra-large' && size !== 'huge';
  const {
    primaryTextClassName,
    optionalTextClassName
  } = useTextClassNames(state, alignToPrimary);
  const styles = useStyles();
  const avatarSpacingStyles = useAvatarSpacingStyles();
  const presenceSpacingStyles = { ...avatarSpacingStyles,
    ...usePresenceSpacingStyles()
  };
  state.root.className = react_1.mergeClasses(exports.personaClassNames.root, styles.base, styles[textPosition], state.root.className);

  if (state.avatar) {
    state.avatar.className = react_1.mergeClasses(exports.personaClassNames.avatar, styles.coin, styles[textAlignment], avatarSpacingStyles[size], avatarSpacingStyles[textPosition], state.avatar.className);
  }

  if (state.presence) {
    state.presence.className = react_1.mergeClasses(exports.personaClassNames.presence, styles.coin, styles[textAlignment], presenceSpacingStyles[size], presenceSpacingStyles[textPosition], textPosition === 'after' && alignToPrimary && styles.afterAlignToPrimary, textPosition === 'before' && alignToPrimary && styles.beforeAlignToPrimary, state.presence.className);
  }

  if (state.primaryText) {
    state.primaryText.className = react_1.mergeClasses(exports.personaClassNames.primaryText, primaryTextClassName, state.primaryText.className);
  }

  if (state.secondaryText) {
    state.secondaryText.className = react_1.mergeClasses(exports.personaClassNames.secondaryText, optionalTextClassName, state.secondaryText.className);
  }

  if (state.tertiaryText) {
    state.tertiaryText.className = react_1.mergeClasses(exports.personaClassNames.tertiaryText, optionalTextClassName, state.tertiaryText.className);
  }

  if (state.quaternaryText) {
    state.quaternaryText.className = react_1.mergeClasses(exports.personaClassNames.quaternaryText, optionalTextClassName, state.quaternaryText.className);
  }

  return state;
};

exports.usePersonaStyles_unstable = usePersonaStyles_unstable;

const useTextStyles = /*#__PURE__*/react_1.__styles({
  "base": {
    "mc9l5x": "ftgm304"
  },
  "primaryText": {
    "sj55zd": "f19n0e5"
  },
  "optionalText": {
    "sj55zd": "fkfq4zb"
  },
  "beforeAlignToPrimary": {
    "Bw0ie65": "f1warjpf"
  },
  "afterAlignToPrimary": {
    "Br312pm": "fwu52yu"
  },
  "body1": {
    "Bahqtrf": "fk6fouc",
    "Be2twd7": "fkhj508",
    "Bhrd7zp": "figsok6",
    "Bg96gwp": "f1i3iumi"
  },
  "caption1": {
    "Bahqtrf": "fk6fouc",
    "Be2twd7": "fy9rknc",
    "Bhrd7zp": "figsok6",
    "Bg96gwp": "fwrc4pm"
  },
  "subtitle2": {
    "Bahqtrf": "fk6fouc",
    "Be2twd7": "fod5ikn",
    "Bhrd7zp": "fl43uef",
    "Bg96gwp": "faaz57k"
  }
}, {
  "d": [".ftgm304{display:block;}", ".f19n0e5{color:var(--colorNeutralForeground1);}", ".fkfq4zb{color:var(--colorNeutralForeground2);}", ".f1warjpf{grid-column-end:middle;}", ".fwu52yu{grid-column-start:middle;}", ".fk6fouc{font-family:var(--fontFamilyBase);}", ".fkhj508{font-size:var(--fontSizeBase300);}", ".figsok6{font-weight:var(--fontWeightRegular);}", ".f1i3iumi{line-height:var(--lineHeightBase300);}", ".fy9rknc{font-size:var(--fontSizeBase200);}", ".fwrc4pm{line-height:var(--lineHeightBase200);}", ".fod5ikn{font-size:var(--fontSizeBase400);}", ".fl43uef{font-weight:var(--fontWeightSemibold);}", ".faaz57k{line-height:var(--lineHeightBase400);}"]
});

const useTextClassNames = (state, alignToPrimary) => {
  const {
    presenceOnly,
    size,
    textPosition
  } = state;
  const textStyles = useTextStyles();
  let primaryTextSize;
  let alignToPrimaryClassName;

  if (presenceOnly) {
    if (size === 'extra-small') {
      primaryTextSize = state.numTextLines > 1 ? textStyles.body1 : textStyles.caption1;
    } else if (size === 'extra-large' || size === 'huge') {
      primaryTextSize = textStyles.subtitle2;
    } else {
      primaryTextSize = textStyles.body1;
    }

    if (alignToPrimary) {
      if (textPosition === 'before') {
        alignToPrimaryClassName = textStyles.beforeAlignToPrimary;
      } else if (textPosition === 'after') {
        alignToPrimaryClassName = textStyles.afterAlignToPrimary;
      }
    }
  } else {
    if (size === 'huge') {
      primaryTextSize = textStyles.subtitle2;
    } else if (size === 'extra-large') {
      primaryTextSize = textStyles.subtitle2;
    } else {
      primaryTextSize = textStyles.body1;
    }
  }

  return {
    primaryTextClassName: react_1.mergeClasses(textStyles.base, textStyles.primaryText, primaryTextSize, alignToPrimaryClassName),
    optionalTextClassName: react_1.mergeClasses(textStyles.base, textStyles.optionalText, !presenceOnly && size === 'huge' ? textStyles.body1 : textStyles.caption1, alignToPrimaryClassName)
  };
};
