// @fluentui/react-menu/lib-amd/components/MenuItemCheckbox/useMenuItemCheckboxStyles.js
define(["require", "exports", "@griffel/react", "../../selectable/index", "../MenuItem/useMenuItemStyles"], function (require, exports, react_1, index_1, useMenuItemStyles_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useMenuItemCheckboxStyles_unstable = exports.menuItemCheckboxClassNames = void 0;
    exports.menuItemCheckboxClassNames = {
        root: 'fui-MenuItemCheckbox',
        icon: 'fui-MenuItemCheckbox__icon',
        checkmark: 'fui-MenuItemCheckbox__checkmark',
        content: 'fui-MenuItemCheckbox__content',
        secondaryContent: 'fui-MenuItemCheckbox__secondaryContent',
    };
    var useMenuItemCheckboxStyles_unstable = function (state) {
        state.root.className = react_1.mergeClasses(exports.menuItemCheckboxClassNames.root, state.root.className);
        if (state.content) {
            state.content.className = react_1.mergeClasses(exports.menuItemCheckboxClassNames.content, state.content.className);
        }
        if (state.secondaryContent) {
            state.secondaryContent.className = react_1.mergeClasses(exports.menuItemCheckboxClassNames.secondaryContent, state.secondaryContent.className);
        }
        if (state.icon) {
            state.icon.className = react_1.mergeClasses(exports.menuItemCheckboxClassNames.icon, state.icon.className);
        }
        if (state.checkmark) {
            state.checkmark.className = react_1.mergeClasses(exports.menuItemCheckboxClassNames.checkmark, state.checkmark.className);
        }
        useMenuItemStyles_1.useMenuItemStyles_unstable(state);
        index_1.useCheckmarkStyles_unstable(state);
    };
    exports.useMenuItemCheckboxStyles_unstable = useMenuItemCheckboxStyles_unstable;
});
