// @fluentui/react-button/lib/components/Button/useButton.js
import { useARIAButtonShorthand } from '@fluentui/react-aria';
import { getNativeElementProps, resolveShorthand } from '@fluentui/react-utilities';
/**
 * Given user props, defines default props for the Button, calls useButtonState, and returns processed state.
 * @param props - User provided props to the Button component.
 * @param ref - User provided ref to be passed to the Button component.
 */

export const useButton_unstable = (props, ref) => {
  const {
    appearance = 'secondary',
    as = 'button',
    disabled = false,
    disabledFocusable = false,
    icon,
    iconPosition = 'before',
    shape = 'rounded',
    size = 'medium'
  } = props;
  const iconShorthand = resolveShorthand(icon);
  return {
    // Props passed at the top-level
    appearance,
    disabled,
    disabledFocusable,
    iconPosition,
    shape,
    size,
    // State calculated from a set of props
    iconOnly: Boolean((iconShorthand === null || iconShorthand === void 0 ? void 0 : iconShorthand.children) && !props.children),
    // Slots definition
    components: {
      root: 'button',
      icon: 'span'
    },
    root: getNativeElementProps(as, useARIAButtonShorthand(props, {
      required: true,
      defaultProps: {
        ref: ref,
        type: 'button'
      }
    })),
    icon: iconShorthand
  };
};
