// @fluentui/react-badge/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useCounterBadge_unstable = exports.useCounterBadgeStyles_unstable = exports.counterBadgeClassNames = exports.CounterBadge = exports.usePresenceBadge_unstable = exports.usePresenceBadgeStyles_unstable = exports.presenceBadgeClassNames = exports.PresenceBadge = exports.useBadge_unstable = exports.useBadgeStyles_unstable = exports.renderBadge_unstable = exports.badgeClassNames = exports.Badge = void 0;

var Badge_1 = /*#__PURE__*/require("./Badge");

Object.defineProperty(exports, "Badge", {
  enumerable: true,
  get: function () {
    return Badge_1.Badge;
  }
});
Object.defineProperty(exports, "badgeClassNames", {
  enumerable: true,
  get: function () {
    return Badge_1.badgeClassNames;
  }
});
Object.defineProperty(exports, "renderBadge_unstable", {
  enumerable: true,
  get: function () {
    return Badge_1.renderBadge_unstable;
  }
});
Object.defineProperty(exports, "useBadgeStyles_unstable", {
  enumerable: true,
  get: function () {
    return Badge_1.useBadgeStyles_unstable;
  }
});
Object.defineProperty(exports, "useBadge_unstable", {
  enumerable: true,
  get: function () {
    return Badge_1.useBadge_unstable;
  }
});

var PresenceBadge_1 = /*#__PURE__*/require("./PresenceBadge");

Object.defineProperty(exports, "PresenceBadge", {
  enumerable: true,
  get: function () {
    return PresenceBadge_1.PresenceBadge;
  }
});
Object.defineProperty(exports, "presenceBadgeClassNames", {
  enumerable: true,
  get: function () {
    return PresenceBadge_1.presenceBadgeClassNames;
  }
});
Object.defineProperty(exports, "usePresenceBadgeStyles_unstable", {
  enumerable: true,
  get: function () {
    return PresenceBadge_1.usePresenceBadgeStyles_unstable;
  }
});
Object.defineProperty(exports, "usePresenceBadge_unstable", {
  enumerable: true,
  get: function () {
    return PresenceBadge_1.usePresenceBadge_unstable;
  }
});

var CounterBadge_1 = /*#__PURE__*/require("./CounterBadge");

Object.defineProperty(exports, "CounterBadge", {
  enumerable: true,
  get: function () {
    return CounterBadge_1.CounterBadge;
  }
});
Object.defineProperty(exports, "counterBadgeClassNames", {
  enumerable: true,
  get: function () {
    return CounterBadge_1.counterBadgeClassNames;
  }
});
Object.defineProperty(exports, "useCounterBadgeStyles_unstable", {
  enumerable: true,
  get: function () {
    return CounterBadge_1.useCounterBadgeStyles_unstable;
  }
});
Object.defineProperty(exports, "useCounterBadge_unstable", {
  enumerable: true,
  get: function () {
    return CounterBadge_1.useCounterBadge_unstable;
  }
});
