// @fluentui/react-link/lib-amd/components/Link/useLink.js
define(["require", "exports", "tslib", "@fluentui/react-utilities", "./useLinkState"], function (require, exports, tslib_1, react_utilities_1, useLinkState_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useLink_unstable = void 0;
    /**
     * Given user props, defines default props for the Link, calls useLinkState_unstable, and returns processed state.
     * @param props - User provided props to the Link component.
     * @param ref - User provided ref to be passed to the Link component.
     */
    var useLink_unstable = function (props, ref) {
        var _a = props.appearance, appearance = _a === void 0 ? 'default' : _a, _b = props.disabled, disabled = _b === void 0 ? false : _b, _c = props.disabledFocusable, disabledFocusable = _c === void 0 ? false : _c, _d = props.inline, inline = _d === void 0 ? false : _d;
        var as = props.as || (props.href ? 'a' : 'button');
        var type = as === 'button' ? 'button' : undefined;
        var state = {
            // Props passed at the top-level
            appearance: appearance,
            disabled: disabled,
            disabledFocusable: disabledFocusable,
            inline: inline,
            // Slots definition
            components: {
                root: 'a',
            },
            root: react_utilities_1.getNativeElementProps(as, tslib_1.__assign(tslib_1.__assign({ ref: ref, type: type }, props), { as: as })),
        };
        useLinkState_1.useLinkState_unstable(state);
        return state;
    };
    exports.useLink_unstable = useLink_unstable;
});
