// @fluentui/react-menu/lib-amd/components/MenuList/useMenuListContextValues.js
define(["require", "exports"], function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useMenuListContextValues_unstable = void 0;
    function useMenuListContextValues_unstable(state) {
        var checkedValues = state.checkedValues, hasCheckmarks = state.hasCheckmarks, hasIcons = state.hasIcons, selectRadio = state.selectRadio, setFocusByFirstCharacter = state.setFocusByFirstCharacter, toggleCheckbox = state.toggleCheckbox;
        // This context is created with "@fluentui/react-context-selector", these is no sense to memoize it
        var menuList = {
            checkedValues: checkedValues,
            hasCheckmarks: hasCheckmarks,
            hasIcons: hasIcons,
            selectRadio: selectRadio,
            setFocusByFirstCharacter: setFocusByFirstCharacter,
            toggleCheckbox: toggleCheckbox,
        };
        return { menuList: menuList };
    }
    exports.useMenuListContextValues_unstable = useMenuListContextValues_unstable;
});
