// @fluentui/react-menu/lib/components/MenuItemCheckbox/useMenuItemCheckbox.js
import * as React from 'react';
import { resolveShorthand } from '@fluentui/react-utilities';
import { Checkmark16Filled } from '@fluentui/react-icons';
import { useMenuListContext_unstable } from '../../contexts/menuListContext';
import { useMenuItem_unstable } from '../MenuItem/useMenuItem';
/** Returns the props and state required to render the component */

export const useMenuItemCheckbox_unstable = (props, ref) => {
  const toggleCheckbox = useMenuListContext_unstable(context => context.toggleCheckbox);
  const {
    name,
    value
  } = props;
  const checked = useMenuListContext_unstable(context => {
    var _a;

    const checkedItems = ((_a = context.checkedValues) === null || _a === void 0 ? void 0 : _a[name]) || [];
    return checkedItems.indexOf(value) !== -1;
  });
  const state = { ...useMenuItem_unstable({
      role: 'menuitemcheckbox',
      persistOnClick: true,
      ...props,
      'aria-checked': checked,
      checkmark: resolveShorthand(props.checkmark, {
        defaultProps: {
          children: /*#__PURE__*/React.createElement(Checkmark16Filled, null)
        },
        required: true
      }),
      onClick: e => {
        var _a;

        toggleCheckbox === null || toggleCheckbox === void 0 ? void 0 : toggleCheckbox(e, name, value, checked);
        (_a = props.onClick) === null || _a === void 0 ? void 0 : _a.call(props, e);
      }
    }, ref),
    name,
    value,
    checked
  };
  return state;
};
