// @fluentui/react-button/lib-amd/components/MenuButton/useMenuButton.js
define(["require", "exports", "tslib", "react", "@fluentui/react-icons", "@fluentui/react-utilities", "../Button/index"], function (require, exports, tslib_1, React, react_icons_1, react_utilities_1, index_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useMenuButton_unstable = void 0;
    /**
     * Given user props, returns the final state for a MenuButton.
     */
    var useMenuButton_unstable = function (_a, ref) {
        var _b;
        var menuIcon = _a.menuIcon, props = tslib_1.__rest(_a, ["menuIcon"]);
        var buttonState = index_1.useButton_unstable(props, ref);
        buttonState.root['aria-expanded'] = (_b = props['aria-expanded']) !== null && _b !== void 0 ? _b : false;
        return tslib_1.__assign(tslib_1.__assign({}, buttonState), { 
            // State calculated from a set of props
            iconOnly: Boolean(!props.children), 
            // Slots definition
            components: {
                root: 'button',
                icon: 'span',
                menuIcon: 'span',
            }, menuIcon: react_utilities_1.resolveShorthand(menuIcon, {
                defaultProps: {
                    children: React.createElement(react_icons_1.ChevronDownRegular, null),
                },
                required: true,
            }) });
    };
    exports.useMenuButton_unstable = useMenuButton_unstable;
});
