// @fluentui/react-dialog/lib-amd/components/DialogActions/useDialogActionsStyles.js
define(["require", "exports", "tslib", "@griffel/react", "../../contexts/constants"], function (require, exports, tslib_1, react_1, constants_1) {
    "use strict";
    var _a;
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useDialogActionsStyles_unstable = exports.dialogActionsClassNames = void 0;
    exports.dialogActionsClassNames = {
        root: 'fui-DialogActions',
    };
    var useStyles = react_1.makeStyles({
        root: tslib_1.__assign(tslib_1.__assign({ height: 'fit-content', boxSizing: 'border-box', display: 'flex' }, react_1.shorthands.gap(constants_1.DIALOG_GAP)), (_a = {}, _a[constants_1.MEDIA_QUERY_BREAKPOINT_SELECTOR] = {
            flexDirection: 'column',
            justifySelf: 'stretch',
        }, _a)),
        gridPositionEnd: tslib_1.__assign({ justifySelf: 'end' }, react_1.shorthands.gridArea(constants_1.ACTIONS_END_GRID_AREA)),
        gridPositionStart: tslib_1.__assign({ justifySelf: 'start' }, react_1.shorthands.gridArea(constants_1.ACTIONS_START_GRID_AREA)),
    });
    /**
     * Apply styling to the DialogActions slots based on the state
     */
    var useDialogActionsStyles_unstable = function (state) {
        var styles = useStyles();
        state.root.className = react_1.mergeClasses(exports.dialogActionsClassNames.root, styles.root, state.position === 'start' && styles.gridPositionStart, state.position === 'end' && styles.gridPositionEnd, state.root.className);
        return state;
    };
    exports.useDialogActionsStyles_unstable = useDialogActionsStyles_unstable;
});
