// @fluentui/keyboard-keys/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ColorF0Red = exports.CodeInput = exports.ClosedCaptionToggle = exports.Close = exports.Clear = exports.ChannelUp = exports.ChannelDown = exports.CapsLock = exports.Cancel = exports.CameraFocus = exports.Camera = exports.Call = exports.BrowserStop = exports.BrowserSearch = exports.BrowserRefresh = exports.BrowserHome = exports.BrowserForward = exports.BrowserFavorites = exports.BrowserBack = exports.BrightnessUp = exports.BrightnessDown = exports.Backspace = exports.AudioVolumeUp = exports.AudioVolumeMute = exports.AudioVolumeDown = exports.AudioTrebleUp = exports.AudioTrebleDown = exports.AudioSurroundModeNext = exports.AudioFaderRear = exports.AudioFaderFront = exports.AudioBassBoostUp = exports.AudioBassBoostToggle = exports.AudioBassBoostDown = exports.AudioBalanceRight = exports.AudioBalanceLeft = exports.Attn = exports.ArrowUp = exports.ArrowRight = exports.ArrowLeft = exports.ArrowDown = exports.AppSwitch = exports.AltGraph = exports.Alt = exports.Alphanumeric = exports.AllCandidates = exports.Again = exports.Accept = exports.AVRPower = exports.AVRInput = exports.keyCodes = void 0;
exports.FavoriteStore2 = exports.FavoriteStore1 = exports.FavoriteStore0 = exports.FavoriteRecall3 = exports.FavoriteRecall2 = exports.FavoriteRecall1 = exports.FavoriteRecall0 = exports.FavoriteClear3 = exports.FavoriteClear2 = exports.FavoriteClear1 = exports.FavoriteClear0 = exports.F9 = exports.F8 = exports.F7 = exports.F6 = exports.F5 = exports.F4 = exports.F3 = exports.F2 = exports.F12 = exports.F11 = exports.F10 = exports.F1 = exports.Exit = exports.Execute = exports.ExSel = exports.Escape = exports.EraseEof = exports.Enter = exports.EndCall = exports.End = exports.Eject = exports.Eisu = exports.DisplaySwap = exports.Dimmer = exports.Delete = exports.Dead = exports.DVR = exports.Cut = exports.CrSel = exports.Copy = exports.Convert = exports.Control = exports.ContextMenu = exports.Compose = exports.ColorF5Brown = exports.ColorF4Grey = exports.ColorF3Blue = exports.ColorF2Yellow = exports.ColorF1Green = void 0;
exports.LiveContent = exports.ListProgram = exports.Link = exports.LaunchWordProcessor = exports.LaunchWebCam = exports.LaunchWebBrowser = exports.LaunchSpreadsheet = exports.LaunchScreenSaver = exports.LaunchPhone = exports.LaunchMusicPlayer = exports.LaunchMediaPlayer = exports.LaunchMail = exports.LaunchContacts = exports.LaunchCalendar = exports.LaunchApplication2 = exports.LaunchApplication1 = exports.LastNumberRedial = exports.Key12 = exports.Key11 = exports.Katakana = exports.KanjiMode = exports.KanaMode = exports.JunjaMode = exports.InstantReplay = exports.Insert = exports.Info = exports.Hyper = exports.Home = exports.HiraganaKatakana = exports.Hiragana = exports.Hibernate = exports.Help = exports.HeadsetHook = exports.Hankaku = exports.HanjaMode = exports.HangulMode = exports.GuidePreviousDay = exports.GuideNextDay = exports.Guide = exports.GroupPrevious = exports.GroupNext = exports.GroupLast = exports.GroupFirst = exports.GoHome = exports.GoBack = exports.FnLock = exports.Fn = exports.Find = exports.FinalMode = exports.FavoriteStore3 = void 0;
exports.Pause = exports.Paste = exports.Pairing = exports.PageUp = exports.PageDown = exports.Open = exports.OnDemand = exports.NumLock = exports.Notification = exports.NonConvert = exports.NextUserProfile = exports.NextFavoriteChannel = exports.NextCandidate = exports.New = exports.NavigatePrevious = exports.NavigateOut = exports.NavigateNext = exports.NavigateIn = exports.ModeChange = exports.MicrophoneVolumeUp = exports.MicrophoneVolumeMute = exports.MicrophoneVolumeDown = exports.MicrophoneToggle = exports.Meta = exports.MediaTrackPrevious = exports.MediaTrackNext = exports.MediaTopMenu = exports.MediaStop = exports.MediaStepForward = exports.MediaStepBackward = exports.MediaSkipForward = exports.MediaSkipBackward = exports.MediaRewind = exports.MediaRecord = exports.MediaPreviousTrack = exports.MediaPlayPause = exports.MediaPlay = exports.MediaPause = exports.MediaNextTrack = exports.MediaLast = exports.MediaFastForward = exports.MediaClose = exports.MediaAudioTrack = exports.MediaApps = exports.MannerMode = exports.MailSend = exports.MailReply = exports.MailForward = exports.LogOff = exports.Lock = void 0;
exports.TVAudioDescriptionMixDown = exports.TVAudioDescription = exports.TVAntennaCable = exports.TV3DMode = exports.TV = exports.SymbolLock = exports.Symbol = exports.Super = exports.Subtitle = exports.Standby = exports.SplitScreenToggle = exports.SpellCheck = exports.SpeechInputToggle = exports.SpeechCorrectionList = exports.Space = exports.Soft4 = exports.Soft3 = exports.Soft2 = exports.Soft1 = exports.SingleCandidate = exports.Shift = exports.Settings = exports.Select = exports.ScrollLock = exports.ScreenModeNext = exports.ScanChannelsToggle = exports.Save = exports.STBPower = exports.STBInput = exports.Romaji = exports.RfBypass = exports.Redo = exports.RecordSpeedNext = exports.RcLowBattery = exports.RandomToggle = exports.Props = exports.Process = exports.PrintScreen = exports.Print = exports.PreviousCandidate = exports.PowerOff = exports.Power = exports.PlaySpeedUp = exports.PlaySpeedReset = exports.PlaySpeedDown = exports.Play = exports.PinPUp = exports.PinPToggle = exports.PinPMove = exports.PinPDown = void 0;
exports.ZoomToggle = exports.ZoomOut = exports.ZoomIn = exports.ZenkakuHankaku = exports.Zenkaku = exports.Wink = exports.WakeUp = exports.VoiceDial = exports.VideoModeNext = exports.Unidentified = exports.Undo = exports.Teletext = exports.Tab = exports.TVTimer = exports.TVTerrestrialDigital = exports.TVTerrestrialAnalog = exports.TVSatelliteToggle = exports.TVSatelliteCS = exports.TVSatelliteBS = exports.TVSatellite = exports.TVRadioService = exports.TVPower = exports.TVNumberEntry = exports.TVNetwork = exports.TVMediaContext = exports.TVInputVGA1 = exports.TVInputHDMI4 = exports.TVInputHDMI3 = exports.TVInputHDMI2 = exports.TVInputHDMI1 = exports.TVInputComposite2 = exports.TVInputComposite1 = exports.TVInputComponent2 = exports.TVInputComponent1 = exports.TVInput = exports.TVDataService = exports.TVContentsMenu = exports.TVAudioDescriptionMixUp = void 0; // https://github.com/microsoft/rushstack/issues/2780
// API extractor can't support namespace exports
// export * as keyCodes from './keyCodes';

const keyCodes = /*#__PURE__*/require("./keyCodes");

exports.keyCodes = keyCodes;

var keys_1 = /*#__PURE__*/require("./keys");

Object.defineProperty(exports, "AVRInput", {
  enumerable: true,
  get: function () {
    return keys_1.AVRInput;
  }
});
Object.defineProperty(exports, "AVRPower", {
  enumerable: true,
  get: function () {
    return keys_1.AVRPower;
  }
});
Object.defineProperty(exports, "Accept", {
  enumerable: true,
  get: function () {
    return keys_1.Accept;
  }
});
Object.defineProperty(exports, "Again", {
  enumerable: true,
  get: function () {
    return keys_1.Again;
  }
});
Object.defineProperty(exports, "AllCandidates", {
  enumerable: true,
  get: function () {
    return keys_1.AllCandidates;
  }
});
Object.defineProperty(exports, "Alphanumeric", {
  enumerable: true,
  get: function () {
    return keys_1.Alphanumeric;
  }
});
Object.defineProperty(exports, "Alt", {
  enumerable: true,
  get: function () {
    return keys_1.Alt;
  }
});
Object.defineProperty(exports, "AltGraph", {
  enumerable: true,
  get: function () {
    return keys_1.AltGraph;
  }
});
Object.defineProperty(exports, "AppSwitch", {
  enumerable: true,
  get: function () {
    return keys_1.AppSwitch;
  }
});
Object.defineProperty(exports, "ArrowDown", {
  enumerable: true,
  get: function () {
    return keys_1.ArrowDown;
  }
});
Object.defineProperty(exports, "ArrowLeft", {
  enumerable: true,
  get: function () {
    return keys_1.ArrowLeft;
  }
});
Object.defineProperty(exports, "ArrowRight", {
  enumerable: true,
  get: function () {
    return keys_1.ArrowRight;
  }
});
Object.defineProperty(exports, "ArrowUp", {
  enumerable: true,
  get: function () {
    return keys_1.ArrowUp;
  }
});
Object.defineProperty(exports, "Attn", {
  enumerable: true,
  get: function () {
    return keys_1.Attn;
  }
});
Object.defineProperty(exports, "AudioBalanceLeft", {
  enumerable: true,
  get: function () {
    return keys_1.AudioBalanceLeft;
  }
});
Object.defineProperty(exports, "AudioBalanceRight", {
  enumerable: true,
  get: function () {
    return keys_1.AudioBalanceRight;
  }
});
Object.defineProperty(exports, "AudioBassBoostDown", {
  enumerable: true,
  get: function () {
    return keys_1.AudioBassBoostDown;
  }
});
Object.defineProperty(exports, "AudioBassBoostToggle", {
  enumerable: true,
  get: function () {
    return keys_1.AudioBassBoostToggle;
  }
});
Object.defineProperty(exports, "AudioBassBoostUp", {
  enumerable: true,
  get: function () {
    return keys_1.AudioBassBoostUp;
  }
});
Object.defineProperty(exports, "AudioFaderFront", {
  enumerable: true,
  get: function () {
    return keys_1.AudioFaderFront;
  }
});
Object.defineProperty(exports, "AudioFaderRear", {
  enumerable: true,
  get: function () {
    return keys_1.AudioFaderRear;
  }
});
Object.defineProperty(exports, "AudioSurroundModeNext", {
  enumerable: true,
  get: function () {
    return keys_1.AudioSurroundModeNext;
  }
});
Object.defineProperty(exports, "AudioTrebleDown", {
  enumerable: true,
  get: function () {
    return keys_1.AudioTrebleDown;
  }
});
Object.defineProperty(exports, "AudioTrebleUp", {
  enumerable: true,
  get: function () {
    return keys_1.AudioTrebleUp;
  }
});
Object.defineProperty(exports, "AudioVolumeDown", {
  enumerable: true,
  get: function () {
    return keys_1.AudioVolumeDown;
  }
});
Object.defineProperty(exports, "AudioVolumeMute", {
  enumerable: true,
  get: function () {
    return keys_1.AudioVolumeMute;
  }
});
Object.defineProperty(exports, "AudioVolumeUp", {
  enumerable: true,
  get: function () {
    return keys_1.AudioVolumeUp;
  }
});
Object.defineProperty(exports, "Backspace", {
  enumerable: true,
  get: function () {
    return keys_1.Backspace;
  }
});
Object.defineProperty(exports, "BrightnessDown", {
  enumerable: true,
  get: function () {
    return keys_1.BrightnessDown;
  }
});
Object.defineProperty(exports, "BrightnessUp", {
  enumerable: true,
  get: function () {
    return keys_1.BrightnessUp;
  }
});
Object.defineProperty(exports, "BrowserBack", {
  enumerable: true,
  get: function () {
    return keys_1.BrowserBack;
  }
});
Object.defineProperty(exports, "BrowserFavorites", {
  enumerable: true,
  get: function () {
    return keys_1.BrowserFavorites;
  }
});
Object.defineProperty(exports, "BrowserForward", {
  enumerable: true,
  get: function () {
    return keys_1.BrowserForward;
  }
});
Object.defineProperty(exports, "BrowserHome", {
  enumerable: true,
  get: function () {
    return keys_1.BrowserHome;
  }
});
Object.defineProperty(exports, "BrowserRefresh", {
  enumerable: true,
  get: function () {
    return keys_1.BrowserRefresh;
  }
});
Object.defineProperty(exports, "BrowserSearch", {
  enumerable: true,
  get: function () {
    return keys_1.BrowserSearch;
  }
});
Object.defineProperty(exports, "BrowserStop", {
  enumerable: true,
  get: function () {
    return keys_1.BrowserStop;
  }
});
Object.defineProperty(exports, "Call", {
  enumerable: true,
  get: function () {
    return keys_1.Call;
  }
});
Object.defineProperty(exports, "Camera", {
  enumerable: true,
  get: function () {
    return keys_1.Camera;
  }
});
Object.defineProperty(exports, "CameraFocus", {
  enumerable: true,
  get: function () {
    return keys_1.CameraFocus;
  }
});
Object.defineProperty(exports, "Cancel", {
  enumerable: true,
  get: function () {
    return keys_1.Cancel;
  }
});
Object.defineProperty(exports, "CapsLock", {
  enumerable: true,
  get: function () {
    return keys_1.CapsLock;
  }
});
Object.defineProperty(exports, "ChannelDown", {
  enumerable: true,
  get: function () {
    return keys_1.ChannelDown;
  }
});
Object.defineProperty(exports, "ChannelUp", {
  enumerable: true,
  get: function () {
    return keys_1.ChannelUp;
  }
});
Object.defineProperty(exports, "Clear", {
  enumerable: true,
  get: function () {
    return keys_1.Clear;
  }
});
Object.defineProperty(exports, "Close", {
  enumerable: true,
  get: function () {
    return keys_1.Close;
  }
});
Object.defineProperty(exports, "ClosedCaptionToggle", {
  enumerable: true,
  get: function () {
    return keys_1.ClosedCaptionToggle;
  }
});
Object.defineProperty(exports, "CodeInput", {
  enumerable: true,
  get: function () {
    return keys_1.CodeInput;
  }
});
Object.defineProperty(exports, "ColorF0Red", {
  enumerable: true,
  get: function () {
    return keys_1.ColorF0Red;
  }
});
Object.defineProperty(exports, "ColorF1Green", {
  enumerable: true,
  get: function () {
    return keys_1.ColorF1Green;
  }
});
Object.defineProperty(exports, "ColorF2Yellow", {
  enumerable: true,
  get: function () {
    return keys_1.ColorF2Yellow;
  }
});
Object.defineProperty(exports, "ColorF3Blue", {
  enumerable: true,
  get: function () {
    return keys_1.ColorF3Blue;
  }
});
Object.defineProperty(exports, "ColorF4Grey", {
  enumerable: true,
  get: function () {
    return keys_1.ColorF4Grey;
  }
});
Object.defineProperty(exports, "ColorF5Brown", {
  enumerable: true,
  get: function () {
    return keys_1.ColorF5Brown;
  }
});
Object.defineProperty(exports, "Compose", {
  enumerable: true,
  get: function () {
    return keys_1.Compose;
  }
});
Object.defineProperty(exports, "ContextMenu", {
  enumerable: true,
  get: function () {
    return keys_1.ContextMenu;
  }
});
Object.defineProperty(exports, "Control", {
  enumerable: true,
  get: function () {
    return keys_1.Control;
  }
});
Object.defineProperty(exports, "Convert", {
  enumerable: true,
  get: function () {
    return keys_1.Convert;
  }
});
Object.defineProperty(exports, "Copy", {
  enumerable: true,
  get: function () {
    return keys_1.Copy;
  }
});
Object.defineProperty(exports, "CrSel", {
  enumerable: true,
  get: function () {
    return keys_1.CrSel;
  }
});
Object.defineProperty(exports, "Cut", {
  enumerable: true,
  get: function () {
    return keys_1.Cut;
  }
});
Object.defineProperty(exports, "DVR", {
  enumerable: true,
  get: function () {
    return keys_1.DVR;
  }
});
Object.defineProperty(exports, "Dead", {
  enumerable: true,
  get: function () {
    return keys_1.Dead;
  }
});
Object.defineProperty(exports, "Delete", {
  enumerable: true,
  get: function () {
    return keys_1.Delete;
  }
});
Object.defineProperty(exports, "Dimmer", {
  enumerable: true,
  get: function () {
    return keys_1.Dimmer;
  }
});
Object.defineProperty(exports, "DisplaySwap", {
  enumerable: true,
  get: function () {
    return keys_1.DisplaySwap;
  }
});
Object.defineProperty(exports, "Eisu", {
  enumerable: true,
  get: function () {
    return keys_1.Eisu;
  }
});
Object.defineProperty(exports, "Eject", {
  enumerable: true,
  get: function () {
    return keys_1.Eject;
  }
});
Object.defineProperty(exports, "End", {
  enumerable: true,
  get: function () {
    return keys_1.End;
  }
});
Object.defineProperty(exports, "EndCall", {
  enumerable: true,
  get: function () {
    return keys_1.EndCall;
  }
});
Object.defineProperty(exports, "Enter", {
  enumerable: true,
  get: function () {
    return keys_1.Enter;
  }
});
Object.defineProperty(exports, "EraseEof", {
  enumerable: true,
  get: function () {
    return keys_1.EraseEof;
  }
});
Object.defineProperty(exports, "Escape", {
  enumerable: true,
  get: function () {
    return keys_1.Escape;
  }
});
Object.defineProperty(exports, "ExSel", {
  enumerable: true,
  get: function () {
    return keys_1.ExSel;
  }
});
Object.defineProperty(exports, "Execute", {
  enumerable: true,
  get: function () {
    return keys_1.Execute;
  }
});
Object.defineProperty(exports, "Exit", {
  enumerable: true,
  get: function () {
    return keys_1.Exit;
  }
});
Object.defineProperty(exports, "F1", {
  enumerable: true,
  get: function () {
    return keys_1.F1;
  }
});
Object.defineProperty(exports, "F10", {
  enumerable: true,
  get: function () {
    return keys_1.F10;
  }
});
Object.defineProperty(exports, "F11", {
  enumerable: true,
  get: function () {
    return keys_1.F11;
  }
});
Object.defineProperty(exports, "F12", {
  enumerable: true,
  get: function () {
    return keys_1.F12;
  }
});
Object.defineProperty(exports, "F2", {
  enumerable: true,
  get: function () {
    return keys_1.F2;
  }
});
Object.defineProperty(exports, "F3", {
  enumerable: true,
  get: function () {
    return keys_1.F3;
  }
});
Object.defineProperty(exports, "F4", {
  enumerable: true,
  get: function () {
    return keys_1.F4;
  }
});
Object.defineProperty(exports, "F5", {
  enumerable: true,
  get: function () {
    return keys_1.F5;
  }
});
Object.defineProperty(exports, "F6", {
  enumerable: true,
  get: function () {
    return keys_1.F6;
  }
});
Object.defineProperty(exports, "F7", {
  enumerable: true,
  get: function () {
    return keys_1.F7;
  }
});
Object.defineProperty(exports, "F8", {
  enumerable: true,
  get: function () {
    return keys_1.F8;
  }
});
Object.defineProperty(exports, "F9", {
  enumerable: true,
  get: function () {
    return keys_1.F9;
  }
});
Object.defineProperty(exports, "FavoriteClear0", {
  enumerable: true,
  get: function () {
    return keys_1.FavoriteClear0;
  }
});
Object.defineProperty(exports, "FavoriteClear1", {
  enumerable: true,
  get: function () {
    return keys_1.FavoriteClear1;
  }
});
Object.defineProperty(exports, "FavoriteClear2", {
  enumerable: true,
  get: function () {
    return keys_1.FavoriteClear2;
  }
});
Object.defineProperty(exports, "FavoriteClear3", {
  enumerable: true,
  get: function () {
    return keys_1.FavoriteClear3;
  }
});
Object.defineProperty(exports, "FavoriteRecall0", {
  enumerable: true,
  get: function () {
    return keys_1.FavoriteRecall0;
  }
});
Object.defineProperty(exports, "FavoriteRecall1", {
  enumerable: true,
  get: function () {
    return keys_1.FavoriteRecall1;
  }
});
Object.defineProperty(exports, "FavoriteRecall2", {
  enumerable: true,
  get: function () {
    return keys_1.FavoriteRecall2;
  }
});
Object.defineProperty(exports, "FavoriteRecall3", {
  enumerable: true,
  get: function () {
    return keys_1.FavoriteRecall3;
  }
});
Object.defineProperty(exports, "FavoriteStore0", {
  enumerable: true,
  get: function () {
    return keys_1.FavoriteStore0;
  }
});
Object.defineProperty(exports, "FavoriteStore1", {
  enumerable: true,
  get: function () {
    return keys_1.FavoriteStore1;
  }
});
Object.defineProperty(exports, "FavoriteStore2", {
  enumerable: true,
  get: function () {
    return keys_1.FavoriteStore2;
  }
});
Object.defineProperty(exports, "FavoriteStore3", {
  enumerable: true,
  get: function () {
    return keys_1.FavoriteStore3;
  }
});
Object.defineProperty(exports, "FinalMode", {
  enumerable: true,
  get: function () {
    return keys_1.FinalMode;
  }
});
Object.defineProperty(exports, "Find", {
  enumerable: true,
  get: function () {
    return keys_1.Find;
  }
});
Object.defineProperty(exports, "Fn", {
  enumerable: true,
  get: function () {
    return keys_1.Fn;
  }
});
Object.defineProperty(exports, "FnLock", {
  enumerable: true,
  get: function () {
    return keys_1.FnLock;
  }
});
Object.defineProperty(exports, "GoBack", {
  enumerable: true,
  get: function () {
    return keys_1.GoBack;
  }
});
Object.defineProperty(exports, "GoHome", {
  enumerable: true,
  get: function () {
    return keys_1.GoHome;
  }
});
Object.defineProperty(exports, "GroupFirst", {
  enumerable: true,
  get: function () {
    return keys_1.GroupFirst;
  }
});
Object.defineProperty(exports, "GroupLast", {
  enumerable: true,
  get: function () {
    return keys_1.GroupLast;
  }
});
Object.defineProperty(exports, "GroupNext", {
  enumerable: true,
  get: function () {
    return keys_1.GroupNext;
  }
});
Object.defineProperty(exports, "GroupPrevious", {
  enumerable: true,
  get: function () {
    return keys_1.GroupPrevious;
  }
});
Object.defineProperty(exports, "Guide", {
  enumerable: true,
  get: function () {
    return keys_1.Guide;
  }
});
Object.defineProperty(exports, "GuideNextDay", {
  enumerable: true,
  get: function () {
    return keys_1.GuideNextDay;
  }
});
Object.defineProperty(exports, "GuidePreviousDay", {
  enumerable: true,
  get: function () {
    return keys_1.GuidePreviousDay;
  }
});
Object.defineProperty(exports, "HangulMode", {
  enumerable: true,
  get: function () {
    return keys_1.HangulMode;
  }
});
Object.defineProperty(exports, "HanjaMode", {
  enumerable: true,
  get: function () {
    return keys_1.HanjaMode;
  }
});
Object.defineProperty(exports, "Hankaku", {
  enumerable: true,
  get: function () {
    return keys_1.Hankaku;
  }
});
Object.defineProperty(exports, "HeadsetHook", {
  enumerable: true,
  get: function () {
    return keys_1.HeadsetHook;
  }
});
Object.defineProperty(exports, "Help", {
  enumerable: true,
  get: function () {
    return keys_1.Help;
  }
});
Object.defineProperty(exports, "Hibernate", {
  enumerable: true,
  get: function () {
    return keys_1.Hibernate;
  }
});
Object.defineProperty(exports, "Hiragana", {
  enumerable: true,
  get: function () {
    return keys_1.Hiragana;
  }
});
Object.defineProperty(exports, "HiraganaKatakana", {
  enumerable: true,
  get: function () {
    return keys_1.HiraganaKatakana;
  }
});
Object.defineProperty(exports, "Home", {
  enumerable: true,
  get: function () {
    return keys_1.Home;
  }
});
Object.defineProperty(exports, "Hyper", {
  enumerable: true,
  get: function () {
    return keys_1.Hyper;
  }
});
Object.defineProperty(exports, "Info", {
  enumerable: true,
  get: function () {
    return keys_1.Info;
  }
});
Object.defineProperty(exports, "Insert", {
  enumerable: true,
  get: function () {
    return keys_1.Insert;
  }
});
Object.defineProperty(exports, "InstantReplay", {
  enumerable: true,
  get: function () {
    return keys_1.InstantReplay;
  }
});
Object.defineProperty(exports, "JunjaMode", {
  enumerable: true,
  get: function () {
    return keys_1.JunjaMode;
  }
});
Object.defineProperty(exports, "KanaMode", {
  enumerable: true,
  get: function () {
    return keys_1.KanaMode;
  }
});
Object.defineProperty(exports, "KanjiMode", {
  enumerable: true,
  get: function () {
    return keys_1.KanjiMode;
  }
});
Object.defineProperty(exports, "Katakana", {
  enumerable: true,
  get: function () {
    return keys_1.Katakana;
  }
});
Object.defineProperty(exports, "Key11", {
  enumerable: true,
  get: function () {
    return keys_1.Key11;
  }
});
Object.defineProperty(exports, "Key12", {
  enumerable: true,
  get: function () {
    return keys_1.Key12;
  }
});
Object.defineProperty(exports, "LastNumberRedial", {
  enumerable: true,
  get: function () {
    return keys_1.LastNumberRedial;
  }
});
Object.defineProperty(exports, "LaunchApplication1", {
  enumerable: true,
  get: function () {
    return keys_1.LaunchApplication1;
  }
});
Object.defineProperty(exports, "LaunchApplication2", {
  enumerable: true,
  get: function () {
    return keys_1.LaunchApplication2;
  }
});
Object.defineProperty(exports, "LaunchCalendar", {
  enumerable: true,
  get: function () {
    return keys_1.LaunchCalendar;
  }
});
Object.defineProperty(exports, "LaunchContacts", {
  enumerable: true,
  get: function () {
    return keys_1.LaunchContacts;
  }
});
Object.defineProperty(exports, "LaunchMail", {
  enumerable: true,
  get: function () {
    return keys_1.LaunchMail;
  }
});
Object.defineProperty(exports, "LaunchMediaPlayer", {
  enumerable: true,
  get: function () {
    return keys_1.LaunchMediaPlayer;
  }
});
Object.defineProperty(exports, "LaunchMusicPlayer", {
  enumerable: true,
  get: function () {
    return keys_1.LaunchMusicPlayer;
  }
});
Object.defineProperty(exports, "LaunchPhone", {
  enumerable: true,
  get: function () {
    return keys_1.LaunchPhone;
  }
});
Object.defineProperty(exports, "LaunchScreenSaver", {
  enumerable: true,
  get: function () {
    return keys_1.LaunchScreenSaver;
  }
});
Object.defineProperty(exports, "LaunchSpreadsheet", {
  enumerable: true,
  get: function () {
    return keys_1.LaunchSpreadsheet;
  }
});
Object.defineProperty(exports, "LaunchWebBrowser", {
  enumerable: true,
  get: function () {
    return keys_1.LaunchWebBrowser;
  }
});
Object.defineProperty(exports, "LaunchWebCam", {
  enumerable: true,
  get: function () {
    return keys_1.LaunchWebCam;
  }
});
Object.defineProperty(exports, "LaunchWordProcessor", {
  enumerable: true,
  get: function () {
    return keys_1.LaunchWordProcessor;
  }
});
Object.defineProperty(exports, "Link", {
  enumerable: true,
  get: function () {
    return keys_1.Link;
  }
});
Object.defineProperty(exports, "ListProgram", {
  enumerable: true,
  get: function () {
    return keys_1.ListProgram;
  }
});
Object.defineProperty(exports, "LiveContent", {
  enumerable: true,
  get: function () {
    return keys_1.LiveContent;
  }
});
Object.defineProperty(exports, "Lock", {
  enumerable: true,
  get: function () {
    return keys_1.Lock;
  }
});
Object.defineProperty(exports, "LogOff", {
  enumerable: true,
  get: function () {
    return keys_1.LogOff;
  }
});
Object.defineProperty(exports, "MailForward", {
  enumerable: true,
  get: function () {
    return keys_1.MailForward;
  }
});
Object.defineProperty(exports, "MailReply", {
  enumerable: true,
  get: function () {
    return keys_1.MailReply;
  }
});
Object.defineProperty(exports, "MailSend", {
  enumerable: true,
  get: function () {
    return keys_1.MailSend;
  }
});
Object.defineProperty(exports, "MannerMode", {
  enumerable: true,
  get: function () {
    return keys_1.MannerMode;
  }
});
Object.defineProperty(exports, "MediaApps", {
  enumerable: true,
  get: function () {
    return keys_1.MediaApps;
  }
});
Object.defineProperty(exports, "MediaAudioTrack", {
  enumerable: true,
  get: function () {
    return keys_1.MediaAudioTrack;
  }
});
Object.defineProperty(exports, "MediaClose", {
  enumerable: true,
  get: function () {
    return keys_1.MediaClose;
  }
});
Object.defineProperty(exports, "MediaFastForward", {
  enumerable: true,
  get: function () {
    return keys_1.MediaFastForward;
  }
});
Object.defineProperty(exports, "MediaLast", {
  enumerable: true,
  get: function () {
    return keys_1.MediaLast;
  }
});
Object.defineProperty(exports, "MediaNextTrack", {
  enumerable: true,
  get: function () {
    return keys_1.MediaNextTrack;
  }
});
Object.defineProperty(exports, "MediaPause", {
  enumerable: true,
  get: function () {
    return keys_1.MediaPause;
  }
});
Object.defineProperty(exports, "MediaPlay", {
  enumerable: true,
  get: function () {
    return keys_1.MediaPlay;
  }
});
Object.defineProperty(exports, "MediaPlayPause", {
  enumerable: true,
  get: function () {
    return keys_1.MediaPlayPause;
  }
});
Object.defineProperty(exports, "MediaPreviousTrack", {
  enumerable: true,
  get: function () {
    return keys_1.MediaPreviousTrack;
  }
});
Object.defineProperty(exports, "MediaRecord", {
  enumerable: true,
  get: function () {
    return keys_1.MediaRecord;
  }
});
Object.defineProperty(exports, "MediaRewind", {
  enumerable: true,
  get: function () {
    return keys_1.MediaRewind;
  }
});
Object.defineProperty(exports, "MediaSkipBackward", {
  enumerable: true,
  get: function () {
    return keys_1.MediaSkipBackward;
  }
});
Object.defineProperty(exports, "MediaSkipForward", {
  enumerable: true,
  get: function () {
    return keys_1.MediaSkipForward;
  }
});
Object.defineProperty(exports, "MediaStepBackward", {
  enumerable: true,
  get: function () {
    return keys_1.MediaStepBackward;
  }
});
Object.defineProperty(exports, "MediaStepForward", {
  enumerable: true,
  get: function () {
    return keys_1.MediaStepForward;
  }
});
Object.defineProperty(exports, "MediaStop", {
  enumerable: true,
  get: function () {
    return keys_1.MediaStop;
  }
});
Object.defineProperty(exports, "MediaTopMenu", {
  enumerable: true,
  get: function () {
    return keys_1.MediaTopMenu;
  }
});
Object.defineProperty(exports, "MediaTrackNext", {
  enumerable: true,
  get: function () {
    return keys_1.MediaTrackNext;
  }
});
Object.defineProperty(exports, "MediaTrackPrevious", {
  enumerable: true,
  get: function () {
    return keys_1.MediaTrackPrevious;
  }
});
Object.defineProperty(exports, "Meta", {
  enumerable: true,
  get: function () {
    return keys_1.Meta;
  }
});
Object.defineProperty(exports, "MicrophoneToggle", {
  enumerable: true,
  get: function () {
    return keys_1.MicrophoneToggle;
  }
});
Object.defineProperty(exports, "MicrophoneVolumeDown", {
  enumerable: true,
  get: function () {
    return keys_1.MicrophoneVolumeDown;
  }
});
Object.defineProperty(exports, "MicrophoneVolumeMute", {
  enumerable: true,
  get: function () {
    return keys_1.MicrophoneVolumeMute;
  }
});
Object.defineProperty(exports, "MicrophoneVolumeUp", {
  enumerable: true,
  get: function () {
    return keys_1.MicrophoneVolumeUp;
  }
});
Object.defineProperty(exports, "ModeChange", {
  enumerable: true,
  get: function () {
    return keys_1.ModeChange;
  }
});
Object.defineProperty(exports, "NavigateIn", {
  enumerable: true,
  get: function () {
    return keys_1.NavigateIn;
  }
});
Object.defineProperty(exports, "NavigateNext", {
  enumerable: true,
  get: function () {
    return keys_1.NavigateNext;
  }
});
Object.defineProperty(exports, "NavigateOut", {
  enumerable: true,
  get: function () {
    return keys_1.NavigateOut;
  }
});
Object.defineProperty(exports, "NavigatePrevious", {
  enumerable: true,
  get: function () {
    return keys_1.NavigatePrevious;
  }
});
Object.defineProperty(exports, "New", {
  enumerable: true,
  get: function () {
    return keys_1.New;
  }
});
Object.defineProperty(exports, "NextCandidate", {
  enumerable: true,
  get: function () {
    return keys_1.NextCandidate;
  }
});
Object.defineProperty(exports, "NextFavoriteChannel", {
  enumerable: true,
  get: function () {
    return keys_1.NextFavoriteChannel;
  }
});
Object.defineProperty(exports, "NextUserProfile", {
  enumerable: true,
  get: function () {
    return keys_1.NextUserProfile;
  }
});
Object.defineProperty(exports, "NonConvert", {
  enumerable: true,
  get: function () {
    return keys_1.NonConvert;
  }
});
Object.defineProperty(exports, "Notification", {
  enumerable: true,
  get: function () {
    return keys_1.Notification;
  }
});
Object.defineProperty(exports, "NumLock", {
  enumerable: true,
  get: function () {
    return keys_1.NumLock;
  }
});
Object.defineProperty(exports, "OnDemand", {
  enumerable: true,
  get: function () {
    return keys_1.OnDemand;
  }
});
Object.defineProperty(exports, "Open", {
  enumerable: true,
  get: function () {
    return keys_1.Open;
  }
});
Object.defineProperty(exports, "PageDown", {
  enumerable: true,
  get: function () {
    return keys_1.PageDown;
  }
});
Object.defineProperty(exports, "PageUp", {
  enumerable: true,
  get: function () {
    return keys_1.PageUp;
  }
});
Object.defineProperty(exports, "Pairing", {
  enumerable: true,
  get: function () {
    return keys_1.Pairing;
  }
});
Object.defineProperty(exports, "Paste", {
  enumerable: true,
  get: function () {
    return keys_1.Paste;
  }
});
Object.defineProperty(exports, "Pause", {
  enumerable: true,
  get: function () {
    return keys_1.Pause;
  }
});
Object.defineProperty(exports, "PinPDown", {
  enumerable: true,
  get: function () {
    return keys_1.PinPDown;
  }
});
Object.defineProperty(exports, "PinPMove", {
  enumerable: true,
  get: function () {
    return keys_1.PinPMove;
  }
});
Object.defineProperty(exports, "PinPToggle", {
  enumerable: true,
  get: function () {
    return keys_1.PinPToggle;
  }
});
Object.defineProperty(exports, "PinPUp", {
  enumerable: true,
  get: function () {
    return keys_1.PinPUp;
  }
});
Object.defineProperty(exports, "Play", {
  enumerable: true,
  get: function () {
    return keys_1.Play;
  }
});
Object.defineProperty(exports, "PlaySpeedDown", {
  enumerable: true,
  get: function () {
    return keys_1.PlaySpeedDown;
  }
});
Object.defineProperty(exports, "PlaySpeedReset", {
  enumerable: true,
  get: function () {
    return keys_1.PlaySpeedReset;
  }
});
Object.defineProperty(exports, "PlaySpeedUp", {
  enumerable: true,
  get: function () {
    return keys_1.PlaySpeedUp;
  }
});
Object.defineProperty(exports, "Power", {
  enumerable: true,
  get: function () {
    return keys_1.Power;
  }
});
Object.defineProperty(exports, "PowerOff", {
  enumerable: true,
  get: function () {
    return keys_1.PowerOff;
  }
});
Object.defineProperty(exports, "PreviousCandidate", {
  enumerable: true,
  get: function () {
    return keys_1.PreviousCandidate;
  }
});
Object.defineProperty(exports, "Print", {
  enumerable: true,
  get: function () {
    return keys_1.Print;
  }
});
Object.defineProperty(exports, "PrintScreen", {
  enumerable: true,
  get: function () {
    return keys_1.PrintScreen;
  }
});
Object.defineProperty(exports, "Process", {
  enumerable: true,
  get: function () {
    return keys_1.Process;
  }
});
Object.defineProperty(exports, "Props", {
  enumerable: true,
  get: function () {
    return keys_1.Props;
  }
});
Object.defineProperty(exports, "RandomToggle", {
  enumerable: true,
  get: function () {
    return keys_1.RandomToggle;
  }
});
Object.defineProperty(exports, "RcLowBattery", {
  enumerable: true,
  get: function () {
    return keys_1.RcLowBattery;
  }
});
Object.defineProperty(exports, "RecordSpeedNext", {
  enumerable: true,
  get: function () {
    return keys_1.RecordSpeedNext;
  }
});
Object.defineProperty(exports, "Redo", {
  enumerable: true,
  get: function () {
    return keys_1.Redo;
  }
});
Object.defineProperty(exports, "RfBypass", {
  enumerable: true,
  get: function () {
    return keys_1.RfBypass;
  }
});
Object.defineProperty(exports, "Romaji", {
  enumerable: true,
  get: function () {
    return keys_1.Romaji;
  }
});
Object.defineProperty(exports, "STBInput", {
  enumerable: true,
  get: function () {
    return keys_1.STBInput;
  }
});
Object.defineProperty(exports, "STBPower", {
  enumerable: true,
  get: function () {
    return keys_1.STBPower;
  }
});
Object.defineProperty(exports, "Save", {
  enumerable: true,
  get: function () {
    return keys_1.Save;
  }
});
Object.defineProperty(exports, "ScanChannelsToggle", {
  enumerable: true,
  get: function () {
    return keys_1.ScanChannelsToggle;
  }
});
Object.defineProperty(exports, "ScreenModeNext", {
  enumerable: true,
  get: function () {
    return keys_1.ScreenModeNext;
  }
});
Object.defineProperty(exports, "ScrollLock", {
  enumerable: true,
  get: function () {
    return keys_1.ScrollLock;
  }
});
Object.defineProperty(exports, "Select", {
  enumerable: true,
  get: function () {
    return keys_1.Select;
  }
});
Object.defineProperty(exports, "Settings", {
  enumerable: true,
  get: function () {
    return keys_1.Settings;
  }
});
Object.defineProperty(exports, "Shift", {
  enumerable: true,
  get: function () {
    return keys_1.Shift;
  }
});
Object.defineProperty(exports, "SingleCandidate", {
  enumerable: true,
  get: function () {
    return keys_1.SingleCandidate;
  }
});
Object.defineProperty(exports, "Soft1", {
  enumerable: true,
  get: function () {
    return keys_1.Soft1;
  }
});
Object.defineProperty(exports, "Soft2", {
  enumerable: true,
  get: function () {
    return keys_1.Soft2;
  }
});
Object.defineProperty(exports, "Soft3", {
  enumerable: true,
  get: function () {
    return keys_1.Soft3;
  }
});
Object.defineProperty(exports, "Soft4", {
  enumerable: true,
  get: function () {
    return keys_1.Soft4;
  }
});
Object.defineProperty(exports, "Space", {
  enumerable: true,
  get: function () {
    return keys_1.Space;
  }
});
Object.defineProperty(exports, "SpeechCorrectionList", {
  enumerable: true,
  get: function () {
    return keys_1.SpeechCorrectionList;
  }
});
Object.defineProperty(exports, "SpeechInputToggle", {
  enumerable: true,
  get: function () {
    return keys_1.SpeechInputToggle;
  }
});
Object.defineProperty(exports, "SpellCheck", {
  enumerable: true,
  get: function () {
    return keys_1.SpellCheck;
  }
});
Object.defineProperty(exports, "SplitScreenToggle", {
  enumerable: true,
  get: function () {
    return keys_1.SplitScreenToggle;
  }
});
Object.defineProperty(exports, "Standby", {
  enumerable: true,
  get: function () {
    return keys_1.Standby;
  }
});
Object.defineProperty(exports, "Subtitle", {
  enumerable: true,
  get: function () {
    return keys_1.Subtitle;
  }
});
Object.defineProperty(exports, "Super", {
  enumerable: true,
  get: function () {
    return keys_1.Super;
  }
});
Object.defineProperty(exports, "Symbol", {
  enumerable: true,
  get: function () {
    return keys_1.Symbol;
  }
});
Object.defineProperty(exports, "SymbolLock", {
  enumerable: true,
  get: function () {
    return keys_1.SymbolLock;
  }
});
Object.defineProperty(exports, "TV", {
  enumerable: true,
  get: function () {
    return keys_1.TV;
  }
});
Object.defineProperty(exports, "TV3DMode", {
  enumerable: true,
  get: function () {
    return keys_1.TV3DMode;
  }
});
Object.defineProperty(exports, "TVAntennaCable", {
  enumerable: true,
  get: function () {
    return keys_1.TVAntennaCable;
  }
});
Object.defineProperty(exports, "TVAudioDescription", {
  enumerable: true,
  get: function () {
    return keys_1.TVAudioDescription;
  }
});
Object.defineProperty(exports, "TVAudioDescriptionMixDown", {
  enumerable: true,
  get: function () {
    return keys_1.TVAudioDescriptionMixDown;
  }
});
Object.defineProperty(exports, "TVAudioDescriptionMixUp", {
  enumerable: true,
  get: function () {
    return keys_1.TVAudioDescriptionMixUp;
  }
});
Object.defineProperty(exports, "TVContentsMenu", {
  enumerable: true,
  get: function () {
    return keys_1.TVContentsMenu;
  }
});
Object.defineProperty(exports, "TVDataService", {
  enumerable: true,
  get: function () {
    return keys_1.TVDataService;
  }
});
Object.defineProperty(exports, "TVInput", {
  enumerable: true,
  get: function () {
    return keys_1.TVInput;
  }
});
Object.defineProperty(exports, "TVInputComponent1", {
  enumerable: true,
  get: function () {
    return keys_1.TVInputComponent1;
  }
});
Object.defineProperty(exports, "TVInputComponent2", {
  enumerable: true,
  get: function () {
    return keys_1.TVInputComponent2;
  }
});
Object.defineProperty(exports, "TVInputComposite1", {
  enumerable: true,
  get: function () {
    return keys_1.TVInputComposite1;
  }
});
Object.defineProperty(exports, "TVInputComposite2", {
  enumerable: true,
  get: function () {
    return keys_1.TVInputComposite2;
  }
});
Object.defineProperty(exports, "TVInputHDMI1", {
  enumerable: true,
  get: function () {
    return keys_1.TVInputHDMI1;
  }
});
Object.defineProperty(exports, "TVInputHDMI2", {
  enumerable: true,
  get: function () {
    return keys_1.TVInputHDMI2;
  }
});
Object.defineProperty(exports, "TVInputHDMI3", {
  enumerable: true,
  get: function () {
    return keys_1.TVInputHDMI3;
  }
});
Object.defineProperty(exports, "TVInputHDMI4", {
  enumerable: true,
  get: function () {
    return keys_1.TVInputHDMI4;
  }
});
Object.defineProperty(exports, "TVInputVGA1", {
  enumerable: true,
  get: function () {
    return keys_1.TVInputVGA1;
  }
});
Object.defineProperty(exports, "TVMediaContext", {
  enumerable: true,
  get: function () {
    return keys_1.TVMediaContext;
  }
});
Object.defineProperty(exports, "TVNetwork", {
  enumerable: true,
  get: function () {
    return keys_1.TVNetwork;
  }
});
Object.defineProperty(exports, "TVNumberEntry", {
  enumerable: true,
  get: function () {
    return keys_1.TVNumberEntry;
  }
});
Object.defineProperty(exports, "TVPower", {
  enumerable: true,
  get: function () {
    return keys_1.TVPower;
  }
});
Object.defineProperty(exports, "TVRadioService", {
  enumerable: true,
  get: function () {
    return keys_1.TVRadioService;
  }
});
Object.defineProperty(exports, "TVSatellite", {
  enumerable: true,
  get: function () {
    return keys_1.TVSatellite;
  }
});
Object.defineProperty(exports, "TVSatelliteBS", {
  enumerable: true,
  get: function () {
    return keys_1.TVSatelliteBS;
  }
});
Object.defineProperty(exports, "TVSatelliteCS", {
  enumerable: true,
  get: function () {
    return keys_1.TVSatelliteCS;
  }
});
Object.defineProperty(exports, "TVSatelliteToggle", {
  enumerable: true,
  get: function () {
    return keys_1.TVSatelliteToggle;
  }
});
Object.defineProperty(exports, "TVTerrestrialAnalog", {
  enumerable: true,
  get: function () {
    return keys_1.TVTerrestrialAnalog;
  }
});
Object.defineProperty(exports, "TVTerrestrialDigital", {
  enumerable: true,
  get: function () {
    return keys_1.TVTerrestrialDigital;
  }
});
Object.defineProperty(exports, "TVTimer", {
  enumerable: true,
  get: function () {
    return keys_1.TVTimer;
  }
});
Object.defineProperty(exports, "Tab", {
  enumerable: true,
  get: function () {
    return keys_1.Tab;
  }
});
Object.defineProperty(exports, "Teletext", {
  enumerable: true,
  get: function () {
    return keys_1.Teletext;
  }
});
Object.defineProperty(exports, "Undo", {
  enumerable: true,
  get: function () {
    return keys_1.Undo;
  }
});
Object.defineProperty(exports, "Unidentified", {
  enumerable: true,
  get: function () {
    return keys_1.Unidentified;
  }
});
Object.defineProperty(exports, "VideoModeNext", {
  enumerable: true,
  get: function () {
    return keys_1.VideoModeNext;
  }
});
Object.defineProperty(exports, "VoiceDial", {
  enumerable: true,
  get: function () {
    return keys_1.VoiceDial;
  }
});
Object.defineProperty(exports, "WakeUp", {
  enumerable: true,
  get: function () {
    return keys_1.WakeUp;
  }
});
Object.defineProperty(exports, "Wink", {
  enumerable: true,
  get: function () {
    return keys_1.Wink;
  }
});
Object.defineProperty(exports, "Zenkaku", {
  enumerable: true,
  get: function () {
    return keys_1.Zenkaku;
  }
});
Object.defineProperty(exports, "ZenkakuHankaku", {
  enumerable: true,
  get: function () {
    return keys_1.ZenkakuHankaku;
  }
});
Object.defineProperty(exports, "ZoomIn", {
  enumerable: true,
  get: function () {
    return keys_1.ZoomIn;
  }
});
Object.defineProperty(exports, "ZoomOut", {
  enumerable: true,
  get: function () {
    return keys_1.ZoomOut;
  }
});
Object.defineProperty(exports, "ZoomToggle", {
  enumerable: true,
  get: function () {
    return keys_1.ZoomToggle;
  }
});
