// @fluentui/keyboard-keys/lib-commonjs/keyCodes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Digit9 = exports.MultiplicationSign = exports.Digit8 = exports.Ampersand = exports.Digit7 = exports.Caret = exports.Digit6 = exports.PercentSign = exports.Digit5 = exports.DollarSign = exports.Digit4 = exports.PoundSign = exports.Digit3 = exports.AtSign = exports.Digit2 = exports.ExclamationPoint = exports.Digit1 = exports.RightParenthesis = exports.Digit0 = exports.Delete = exports.Insert = exports.PrintScreen = exports.Execute = exports.Print = exports.Select = exports.ArrowDown = exports.ArrowRight = exports.ArrowUp = exports.ArrowLeft = exports.Home = exports.End = exports.PageDown = exports.PageUp = exports.Space = exports.ModeChange = exports.Accept = exports.NonConvert = exports.Convert = exports.Escape = exports.CapsLock = exports.Pause = exports.Alt = exports.Control = exports.Shift = exports.Enter = exports.Clear = exports.Tab = exports.Backspace = exports.Help = exports.Cancel = void 0;
exports.y = exports.X = exports.x = exports.W = exports.w = exports.V = exports.v = exports.U = exports.u = exports.T = exports.t = exports.S = exports.s = exports.R = exports.r = exports.Q = exports.q = exports.P = exports.p = exports.O = exports.o = exports.N = exports.n = exports.M = exports.m = exports.L = exports.l = exports.K = exports.k = exports.J = exports.j = exports.I = exports.i = exports.H = exports.h = exports.G = exports.g = exports.F = exports.f = exports.E = exports.e = exports.D = exports.d = exports.C = exports.c = exports.B = exports.b = exports.A = exports.a = exports.LeftParenthesis = void 0;
exports.BackSlash = exports.LeftCurlyBrace = exports.LeftSquareBracket = exports.Tilde = exports.GraveAccent = exports.QuestionMark = exports.ForwardSlash = exports.RightAngleBracket = exports.Decimal = exports.Underscore = exports.MinusSign = exports.LeftAngleBracket = exports.Comma = exports.PlusSign = exports.EqualsSign = exports.Semicolon = exports.VolumeUp = exports.VolumeDown = exports.VolumeMute = exports.ScrollLock = exports.NumLock = exports.F24 = exports.F23 = exports.F22 = exports.F21 = exports.F20 = exports.F19 = exports.F18 = exports.F17 = exports.F16 = exports.F15 = exports.F14 = exports.F13 = exports.F12 = exports.F11 = exports.F10 = exports.F9 = exports.F8 = exports.F7 = exports.F6 = exports.F5 = exports.F4 = exports.F3 = exports.F2 = exports.F1 = exports.ContextMenu = exports.OS = exports.Z = exports.z = exports.Y = void 0;
exports.ZoomOut = exports.Play = exports.EraseEof = exports.ExSel = exports.CrSel = exports.Attn = exports.AltGraph = exports.Meta = exports.DoubleQuote = exports.SingleQuote = exports.RightCurlyBrace = exports.RightSquareBracket = exports.Pipe = void 0;
exports.Cancel = 3;
exports.Help = 6;
exports.Backspace = 8;
exports.Tab = 9;
exports.Clear = 12;
exports.Enter = 13;
exports.Shift = 16;
exports.Control = 17;
exports.Alt = 18;
exports.Pause = 19;
exports.CapsLock = 20;
exports.Escape = 27;
exports.Convert = 28;
exports.NonConvert = 29;
exports.Accept = 30;
exports.ModeChange = 31;
exports.Space = 32;
exports.PageUp = 33;
exports.PageDown = 34;
exports.End = 35;
exports.Home = 36;
exports.ArrowLeft = 37;
exports.ArrowUp = 38;
exports.ArrowRight = 39;
exports.ArrowDown = 40;
exports.Select = 41;
exports.Print = 42;
exports.Execute = 43;
exports.PrintScreen = 44;
exports.Insert = 45;
exports.Delete = 46;
exports.Digit0 = 48;
exports.RightParenthesis = 48;
exports.Digit1 = 49;
exports.ExclamationPoint = 49;
exports.Digit2 = 50;
exports.AtSign = 50;
exports.Digit3 = 51;
exports.PoundSign = 51;
exports.Digit4 = 52;
exports.DollarSign = 52;
exports.Digit5 = 53;
exports.PercentSign = 53;
exports.Digit6 = 54;
exports.Caret = 54;
exports.Digit7 = 55;
exports.Ampersand = 55;
exports.Digit8 = 56;
exports.MultiplicationSign = 56;
exports.Digit9 = 57;
exports.LeftParenthesis = 57;
exports.a = 65;
exports.A = 65;
exports.b = 66;
exports.B = 66;
exports.c = 67;
exports.C = 67;
exports.d = 68;
exports.D = 68;
exports.e = 69;
exports.E = 69;
exports.f = 70;
exports.F = 70;
exports.g = 71;
exports.G = 71;
exports.h = 72;
exports.H = 72;
exports.i = 73;
exports.I = 73;
exports.j = 74;
exports.J = 74;
exports.k = 75;
exports.K = 75;
exports.l = 76;
exports.L = 76;
exports.m = 77;
exports.M = 77;
exports.n = 78;
exports.N = 78;
exports.o = 79;
exports.O = 79;
exports.p = 80;
exports.P = 80;
exports.q = 81;
exports.Q = 81;
exports.r = 82;
exports.R = 82;
exports.s = 83;
exports.S = 83;
exports.t = 84;
exports.T = 84;
exports.u = 85;
exports.U = 85;
exports.v = 86;
exports.V = 86;
exports.w = 87;
exports.W = 87;
exports.x = 88;
exports.X = 88;
exports.y = 89;
exports.Y = 89;
exports.z = 90;
exports.Z = 90;
exports.OS = 91;
exports.ContextMenu = 93;
exports.F1 = 112;
exports.F2 = 113;
exports.F3 = 114;
exports.F4 = 115;
exports.F5 = 116;
exports.F6 = 117;
exports.F7 = 118;
exports.F8 = 119;
exports.F9 = 120;
exports.F10 = 121;
exports.F11 = 122;
exports.F12 = 123;
exports.F13 = 124;
exports.F14 = 125;
exports.F15 = 126;
exports.F16 = 127;
exports.F17 = 128;
exports.F18 = 129;
exports.F19 = 130;
exports.F20 = 131;
exports.F21 = 132;
exports.F22 = 133;
exports.F23 = 134;
exports.F24 = 135;
exports.NumLock = 144;
exports.ScrollLock = 145;
exports.VolumeMute = 181;
exports.VolumeDown = 182;
exports.VolumeUp = 183;
exports.Semicolon = 186;
exports.EqualsSign = 187;
exports.PlusSign = 187;
exports.Comma = 188;
exports.LeftAngleBracket = 188;
exports.MinusSign = 189;
exports.Underscore = 189;
exports.Decimal = 190;
exports.RightAngleBracket = 190;
exports.ForwardSlash = 191;
exports.QuestionMark = 191;
exports.GraveAccent = 192;
exports.Tilde = 192;
exports.LeftSquareBracket = 219;
exports.LeftCurlyBrace = 219;
exports.BackSlash = 220;
exports.Pipe = 220;
exports.RightSquareBracket = 221;
exports.RightCurlyBrace = 221;
exports.SingleQuote = 222;
exports.DoubleQuote = 222;
exports.Meta = 224;
exports.AltGraph = 225;
exports.Attn = 246;
exports.CrSel = 247;
exports.ExSel = 248;
exports.EraseEof = 249;
exports.Play = 250;
exports.ZoomOut = 251;
