// @fluentui/keyboard-keys/lib-commonjs/keys.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Select = exports.Props = exports.Play = exports.Pause = exports.Help = exports.Find = exports.Execute = exports.Escape = exports.ContextMenu = exports.Cancel = exports.Attn = exports.Again = exports.Accept = exports.Undo = exports.Redo = exports.Paste = exports.Insert = exports.ExSel = exports.EraseEof = exports.Delete = exports.Cut = exports.CrSel = exports.Copy = exports.Clear = exports.Backspace = exports.PageUp = exports.PageDown = exports.Home = exports.End = exports.ArrowUp = exports.ArrowRight = exports.ArrowLeft = exports.ArrowDown = exports.Tab = exports.Space = exports.Enter = exports.Super = exports.Hyper = exports.SymbolLock = exports.Symbol = exports.Shift = exports.ScrollLock = exports.NumLock = exports.Meta = exports.FnLock = exports.Fn = exports.Control = exports.CapsLock = exports.AltGraph = exports.Alt = void 0;
exports.F8 = exports.F7 = exports.F6 = exports.F5 = exports.F4 = exports.F3 = exports.F2 = exports.F1 = exports.ZenkakuHankaku = exports.Zenkaku = exports.Romaji = exports.Katakana = exports.KanjiMode = exports.KanaMode = exports.HiraganaKatakana = exports.Hiragana = exports.Hankaku = exports.Eisu = exports.JunjaMode = exports.HanjaMode = exports.HangulMode = exports.SingleCandidate = exports.Process = exports.PreviousCandidate = exports.NonConvert = exports.NextCandidate = exports.ModeChange = exports.GroupPrevious = exports.GroupNext = exports.GroupLast = exports.GroupFirst = exports.FinalMode = exports.Dead = exports.Convert = exports.Compose = exports.CodeInput = exports.Alphanumeric = exports.AllCandidates = exports.WakeUp = exports.Standby = exports.Hibernate = exports.PrintScreen = exports.PowerOff = exports.Power = exports.LogOff = exports.Eject = exports.BrightnessUp = exports.BrightnessDown = exports.ZoomOut = exports.ZoomIn = void 0;
exports.SpeechInputToggle = exports.SpeechCorrectionList = exports.MicrophoneVolumeMute = exports.MicrophoneVolumeUp = exports.MicrophoneVolumeDown = exports.MicrophoneToggle = exports.AudioVolumeMute = exports.AudioVolumeUp = exports.AudioVolumeDown = exports.AudioTrebleUp = exports.AudioTrebleDown = exports.AudioSurroundModeNext = exports.AudioFaderRear = exports.AudioFaderFront = exports.AudioBassBoostUp = exports.AudioBassBoostToggle = exports.AudioBassBoostDown = exports.AudioBalanceRight = exports.AudioBalanceLeft = exports.Key12 = exports.Key11 = exports.SpellCheck = exports.Save = exports.Print = exports.Open = exports.New = exports.MediaTrackPrevious = exports.MediaTrackNext = exports.MediaStop = exports.MediaRewind = exports.MediaRecord = exports.MediaPlayPause = exports.MediaPlay = exports.MediaPause = exports.MediaFastForward = exports.MediaClose = exports.MailSend = exports.MailReply = exports.MailForward = exports.Close = exports.ChannelUp = exports.ChannelDown = exports.Soft4 = exports.Soft3 = exports.Soft2 = exports.Soft1 = exports.F12 = exports.F11 = exports.F10 = exports.F9 = void 0;
exports.TVInputVGA1 = exports.TVInputHDMI4 = exports.TVInputHDMI3 = exports.TVInputHDMI2 = exports.TVInputHDMI1 = exports.TVInputComposite2 = exports.TVInputComposite1 = exports.TVInputComponent2 = exports.TVInputComponent1 = exports.TVInput = exports.TVDataService = exports.TVContentsMenu = exports.TVAudioDescriptionMixUp = exports.TVAudioDescriptionMixDown = exports.TVAudioDescription = exports.TVAntennaCable = exports.TV3DMode = exports.TV = exports.VoiceDial = exports.MannerMode = exports.Notification = exports.LastNumberRedial = exports.HeadsetHook = exports.GoHome = exports.GoBack = exports.EndCall = exports.CameraFocus = exports.Camera = exports.Call = exports.AppSwitch = exports.BrowserStop = exports.BrowserSearch = exports.BrowserRefresh = exports.BrowserHome = exports.BrowserForward = exports.BrowserFavorites = exports.BrowserBack = exports.LaunchWordProcessor = exports.LaunchWebCam = exports.LaunchWebBrowser = exports.LaunchSpreadsheet = exports.LaunchScreenSaver = exports.LaunchPhone = exports.LaunchMusicPlayer = exports.LaunchMediaPlayer = exports.LaunchMail = exports.LaunchContacts = exports.LaunchCalendar = exports.LaunchApplication2 = exports.LaunchApplication1 = void 0;
exports.MediaSkipBackward = exports.MediaLast = exports.MediaAudioTrack = exports.MediaApps = exports.Lock = exports.LiveContent = exports.ListProgram = exports.Link = exports.InstantReplay = exports.Info = exports.GuidePreviousDay = exports.GuideNextDay = exports.Guide = exports.FavoriteStore3 = exports.FavoriteStore2 = exports.FavoriteStore1 = exports.FavoriteStore0 = exports.FavoriteRecall3 = exports.FavoriteRecall2 = exports.FavoriteRecall1 = exports.FavoriteRecall0 = exports.FavoriteClear3 = exports.FavoriteClear2 = exports.FavoriteClear1 = exports.FavoriteClear0 = exports.Exit = exports.DVR = exports.DisplaySwap = exports.Dimmer = exports.ClosedCaptionToggle = exports.ColorF5Brown = exports.ColorF4Grey = exports.ColorF3Blue = exports.ColorF2Yellow = exports.ColorF1Green = exports.ColorF0Red = exports.AVRPower = exports.AVRInput = exports.TVTimer = exports.TVTerrestrialDigital = exports.TVTerrestrialAnalog = exports.TVSatelliteToggle = exports.TVSatelliteCS = exports.TVSatelliteBS = exports.TVSatellite = exports.TVRadioService = exports.TVPower = exports.TVNumberEntry = exports.TVNetwork = exports.TVMediaContext = void 0;
exports.Unidentified = exports.MediaPreviousTrack = exports.MediaNextTrack = exports.ZoomToggle = exports.Wink = exports.VideoModeNext = exports.Teletext = exports.Subtitle = exports.STBPower = exports.STBInput = exports.SplitScreenToggle = exports.Settings = exports.ScreenModeNext = exports.ScanChannelsToggle = exports.RfBypass = exports.RecordSpeedNext = exports.RcLowBattery = exports.RandomToggle = exports.PlaySpeedUp = exports.PlaySpeedReset = exports.PlaySpeedDown = exports.PinPUp = exports.PinPToggle = exports.PinPMove = exports.PinPDown = exports.Pairing = exports.OnDemand = exports.NextUserProfile = exports.NextFavoriteChannel = exports.NavigatePrevious = exports.NavigateOut = exports.NavigateNext = exports.NavigateIn = exports.MediaTopMenu = exports.MediaStepForward = exports.MediaStepBackward = exports.MediaSkipForward = void 0;
exports.Alt = 'Alt';
exports.AltGraph = 'AltGraph';
exports.CapsLock = 'CapsLock';
exports.Control = 'Control';
exports.Fn = 'Fn';
exports.FnLock = 'FnLock';
exports.Meta = 'Meta';
exports.NumLock = 'NumLock';
exports.ScrollLock = 'ScrollLock';
exports.Shift = 'Shift';
exports.Symbol = 'Symbol';
exports.SymbolLock = 'SymbolLock';
exports.Hyper = 'Hyper';
exports.Super = 'Super';
exports.Enter = 'Enter';
exports.Space = ' ';
exports.Tab = 'Tab';
exports.ArrowDown = 'ArrowDown';
exports.ArrowLeft = 'ArrowLeft';
exports.ArrowRight = 'ArrowRight';
exports.ArrowUp = 'ArrowUp';
exports.End = 'End';
exports.Home = 'Home';
exports.PageDown = 'PageDown';
exports.PageUp = 'PageUp';
exports.Backspace = 'Backspace';
exports.Clear = 'Clear';
exports.Copy = 'Copy';
exports.CrSel = 'CrSel';
exports.Cut = 'Cut';
exports.Delete = 'Delete';
exports.EraseEof = 'EraseEof';
exports.ExSel = 'ExSel';
exports.Insert = 'Insert';
exports.Paste = 'Paste';
exports.Redo = 'Redo';
exports.Undo = 'Undo';
exports.Accept = 'Accept';
exports.Again = 'Again';
exports.Attn = 'Attn';
exports.Cancel = 'Cancel';
exports.ContextMenu = 'ContextMenu';
exports.Escape = 'Escape';
exports.Execute = 'Execute';
exports.Find = 'Find';
exports.Help = 'Help';
exports.Pause = 'Pause';
exports.Play = 'Play';
exports.Props = 'Props';
exports.Select = 'Select';
exports.ZoomIn = 'ZoomIn';
exports.ZoomOut = 'ZoomOut';
exports.BrightnessDown = 'BrightnessDown';
exports.BrightnessUp = 'BrightnessUp';
exports.Eject = 'Eject';
exports.LogOff = 'LogOff';
exports.Power = 'Power';
exports.PowerOff = 'PowerOff';
exports.PrintScreen = 'PrintScreen';
exports.Hibernate = 'Hibernate';
exports.Standby = 'Standby';
exports.WakeUp = 'WakeUp';
exports.AllCandidates = 'AllCandidates';
exports.Alphanumeric = 'Alphanumeric';
exports.CodeInput = 'CodeInput';
exports.Compose = 'Compose';
exports.Convert = 'Convert';
exports.Dead = 'Dead';
exports.FinalMode = 'FinalMode';
exports.GroupFirst = 'GroupFirst';
exports.GroupLast = 'GroupLast';
exports.GroupNext = 'GroupNext';
exports.GroupPrevious = 'GroupPrevious';
exports.ModeChange = 'ModeChange';
exports.NextCandidate = 'NextCandidate';
exports.NonConvert = 'NonConvert';
exports.PreviousCandidate = 'PreviousCandidate';
exports.Process = 'Process';
exports.SingleCandidate = 'SingleCandidate';
exports.HangulMode = 'HangulMode';
exports.HanjaMode = 'HanjaMode';
exports.JunjaMode = 'JunjaMode';
exports.Eisu = 'Eisu';
exports.Hankaku = 'Hankaku';
exports.Hiragana = 'Hiragana';
exports.HiraganaKatakana = 'HiraganaKatakana';
exports.KanaMode = 'KanaMode';
exports.KanjiMode = 'KanjiMode';
exports.Katakana = 'Katakana';
exports.Romaji = 'Romaji';
exports.Zenkaku = 'Zenkaku';
exports.ZenkakuHankaku = 'ZenkakuHankaku';
exports.F1 = 'F1';
exports.F2 = 'F2';
exports.F3 = 'F3';
exports.F4 = 'F4';
exports.F5 = 'F5';
exports.F6 = 'F6';
exports.F7 = 'F7';
exports.F8 = 'F8';
exports.F9 = 'F9';
exports.F10 = 'F10';
exports.F11 = 'F11';
exports.F12 = 'F12';
exports.Soft1 = 'Soft1';
exports.Soft2 = 'Soft2';
exports.Soft3 = 'Soft3';
exports.Soft4 = 'Soft4';
exports.ChannelDown = 'ChannelDown';
exports.ChannelUp = 'ChannelUp';
exports.Close = 'Close';
exports.MailForward = 'MailForward';
exports.MailReply = 'MailReply';
exports.MailSend = 'MailSend';
exports.MediaClose = 'MediaClose';
exports.MediaFastForward = 'MediaFastForward';
exports.MediaPause = 'MediaPause';
exports.MediaPlay = 'MediaPlay';
exports.MediaPlayPause = 'MediaPlayPause';
exports.MediaRecord = 'MediaRecord';
exports.MediaRewind = 'MediaRewind';
exports.MediaStop = 'MediaStop';
exports.MediaTrackNext = 'MediaTrackNext';
exports.MediaTrackPrevious = 'MediaTrackPrevious';
exports.New = 'New';
exports.Open = 'Open';
exports.Print = 'Print';
exports.Save = 'Save';
exports.SpellCheck = 'SpellCheck';
exports.Key11 = 'Key11';
exports.Key12 = 'Key12';
exports.AudioBalanceLeft = 'AudioBalanceLeft';
exports.AudioBalanceRight = 'AudioBalanceRight';
exports.AudioBassBoostDown = 'AudioBassBoostDown';
exports.AudioBassBoostToggle = 'AudioBassBoostToggle';
exports.AudioBassBoostUp = 'AudioBassBoostUp';
exports.AudioFaderFront = 'AudioFaderFront';
exports.AudioFaderRear = 'AudioFaderRear';
exports.AudioSurroundModeNext = 'AudioSurroundModeNext';
exports.AudioTrebleDown = 'AudioTrebleDown';
exports.AudioTrebleUp = 'AudioTrebleUp';
exports.AudioVolumeDown = 'AudioVolumeDown';
exports.AudioVolumeUp = 'AudioVolumeUp';
exports.AudioVolumeMute = 'AudioVolumeMute';
exports.MicrophoneToggle = 'MicrophoneToggle';
exports.MicrophoneVolumeDown = 'MicrophoneVolumeDown';
exports.MicrophoneVolumeUp = 'MicrophoneVolumeUp';
exports.MicrophoneVolumeMute = 'MicrophoneVolumeMute';
exports.SpeechCorrectionList = 'SpeechCorrectionList';
exports.SpeechInputToggle = 'SpeechInputToggle';
exports.LaunchApplication1 = 'LaunchApplication1';
exports.LaunchApplication2 = 'LaunchApplication2';
exports.LaunchCalendar = 'LaunchCalendar';
exports.LaunchContacts = 'LaunchContacts';
exports.LaunchMail = 'LaunchMail';
exports.LaunchMediaPlayer = 'LaunchMediaPlayer';
exports.LaunchMusicPlayer = 'LaunchMusicPlayer';
exports.LaunchPhone = 'LaunchPhone';
exports.LaunchScreenSaver = 'LaunchScreenSaver';
exports.LaunchSpreadsheet = 'LaunchSpreadsheet';
exports.LaunchWebBrowser = 'LaunchWebBrowser';
exports.LaunchWebCam = 'LaunchWebCam';
exports.LaunchWordProcessor = 'LaunchWordProcessor';
exports.BrowserBack = 'BrowserBack';
exports.BrowserFavorites = 'BrowserFavorites';
exports.BrowserForward = 'BrowserForward';
exports.BrowserHome = 'BrowserHome';
exports.BrowserRefresh = 'BrowserRefresh';
exports.BrowserSearch = 'BrowserSearch';
exports.BrowserStop = 'BrowserStop';
exports.AppSwitch = 'AppSwitch';
exports.Call = 'Call';
exports.Camera = 'Camera';
exports.CameraFocus = 'CameraFocus';
exports.EndCall = 'EndCall';
exports.GoBack = 'GoBack';
exports.GoHome = 'GoHome';
exports.HeadsetHook = 'HeadsetHook';
exports.LastNumberRedial = 'LastNumberRedial';
exports.Notification = 'Notification';
exports.MannerMode = 'MannerMode';
exports.VoiceDial = 'VoiceDial';
exports.TV = 'TV';
exports.TV3DMode = 'TV3DMode';
exports.TVAntennaCable = 'TVAntennaCable';
exports.TVAudioDescription = 'TVAudioDescription';
exports.TVAudioDescriptionMixDown = 'TVAudioDescriptionMixDown';
exports.TVAudioDescriptionMixUp = 'TVAudioDescriptionMixUp';
exports.TVContentsMenu = 'TVContentsMenu';
exports.TVDataService = 'TVDataService';
exports.TVInput = 'TVInput';
exports.TVInputComponent1 = 'TVInputComponent1';
exports.TVInputComponent2 = 'TVInputComponent2';
exports.TVInputComposite1 = 'TVInputComposite1';
exports.TVInputComposite2 = 'TVInputComposite2';
exports.TVInputHDMI1 = 'TVInputHDMI1';
exports.TVInputHDMI2 = 'TVInputHDMI2';
exports.TVInputHDMI3 = 'TVInputHDMI3';
exports.TVInputHDMI4 = 'TVInputHDMI4';
exports.TVInputVGA1 = 'TVInputVGA1';
exports.TVMediaContext = 'TVMediaContext';
exports.TVNetwork = 'TVNetwork';
exports.TVNumberEntry = 'TVNumberEntry';
exports.TVPower = 'TVPower';
exports.TVRadioService = 'TVRadioService';
exports.TVSatellite = 'TVSatellite';
exports.TVSatelliteBS = 'TVSatelliteBS';
exports.TVSatelliteCS = 'TVSatelliteCS';
exports.TVSatelliteToggle = 'TVSatelliteToggle';
exports.TVTerrestrialAnalog = 'TVTerrestrialAnalog';
exports.TVTerrestrialDigital = 'TVTerrestrialDigital';
exports.TVTimer = 'TVTimer';
exports.AVRInput = 'AVRInput';
exports.AVRPower = 'AVRPower';
exports.ColorF0Red = 'ColorF0Red';
exports.ColorF1Green = 'ColorF1Green';
exports.ColorF2Yellow = 'ColorF2Yellow';
exports.ColorF3Blue = 'ColorF3Blue';
exports.ColorF4Grey = 'ColorF4Grey';
exports.ColorF5Brown = 'ColorF5Brown';
exports.ClosedCaptionToggle = 'ClosedCaptionToggle';
exports.Dimmer = 'Dimmer';
exports.DisplaySwap = 'DisplaySwap';
exports.DVR = 'DVR';
exports.Exit = 'Exit';
exports.FavoriteClear0 = 'FavoriteClear0';
exports.FavoriteClear1 = 'FavoriteClear1';
exports.FavoriteClear2 = 'FavoriteClear2';
exports.FavoriteClear3 = 'FavoriteClear3';
exports.FavoriteRecall0 = 'FavoriteRecall0';
exports.FavoriteRecall1 = 'FavoriteRecall1';
exports.FavoriteRecall2 = 'FavoriteRecall2';
exports.FavoriteRecall3 = 'FavoriteRecall3';
exports.FavoriteStore0 = 'FavoriteStore0';
exports.FavoriteStore1 = 'FavoriteStore1';
exports.FavoriteStore2 = 'FavoriteStore2';
exports.FavoriteStore3 = 'FavoriteStore3';
exports.Guide = 'Guide';
exports.GuideNextDay = 'GuideNextDay';
exports.GuidePreviousDay = 'GuidePreviousDay';
exports.Info = 'Info';
exports.InstantReplay = 'InstantReplay';
exports.Link = 'Link';
exports.ListProgram = 'ListProgram';
exports.LiveContent = 'LiveContent';
exports.Lock = 'Lock';
exports.MediaApps = 'MediaApps';
exports.MediaAudioTrack = 'MediaAudioTrack';
exports.MediaLast = 'MediaLast';
exports.MediaSkipBackward = 'MediaSkipBackward';
exports.MediaSkipForward = 'MediaSkipForward';
exports.MediaStepBackward = 'MediaStepBackward';
exports.MediaStepForward = 'MediaStepForward';
exports.MediaTopMenu = 'MediaTopMenu';
exports.NavigateIn = 'NavigateIn';
exports.NavigateNext = 'NavigateNext';
exports.NavigateOut = 'NavigateOut';
exports.NavigatePrevious = 'NavigatePrevious';
exports.NextFavoriteChannel = 'NextFavoriteChannel';
exports.NextUserProfile = 'NextUserProfile';
exports.OnDemand = 'OnDemand';
exports.Pairing = 'Pairing';
exports.PinPDown = 'PinPDown';
exports.PinPMove = 'PinPMove';
exports.PinPToggle = 'PinPToggle';
exports.PinPUp = 'PinPUp';
exports.PlaySpeedDown = 'PlaySpeedDown';
exports.PlaySpeedReset = 'PlaySpeedReset';
exports.PlaySpeedUp = 'PlaySpeedUp';
exports.RandomToggle = 'RandomToggle';
exports.RcLowBattery = 'RcLowBattery';
exports.RecordSpeedNext = 'RecordSpeedNext';
exports.RfBypass = 'RfBypass';
exports.ScanChannelsToggle = 'ScanChannelsToggle';
exports.ScreenModeNext = 'ScreenModeNext';
exports.Settings = 'Settings';
exports.SplitScreenToggle = 'SplitScreenToggle';
exports.STBInput = 'STBInput';
exports.STBPower = 'STBPower';
exports.Subtitle = 'Subtitle';
exports.Teletext = 'Teletext';
exports.VideoModeNext = 'VideoModeNext';
exports.Wink = 'Wink';
exports.ZoomToggle = 'ZoomToggle';
exports.MediaNextTrack = 'MediaNextTrack';
exports.MediaPreviousTrack = 'MediaPreviousTrack';
exports.Unidentified = 'Unidentified';
