// @fluentui/react-table/lib-commonjs/components/TableHeaderCell/useTableHeaderCell.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useTableHeaderCell_unstable = void 0;

const React = /*#__PURE__*/require("react");

const react_utilities_1 = /*#__PURE__*/require("@fluentui/react-utilities");

const react_tabster_1 = /*#__PURE__*/require("@fluentui/react-tabster");

const react_icons_1 = /*#__PURE__*/require("@fluentui/react-icons");

const tableContext_1 = /*#__PURE__*/require("../../contexts/tableContext");

const react_aria_1 = /*#__PURE__*/require("@fluentui/react-aria");

const sortIcons = {
  ascending: /*#__PURE__*/React.createElement(react_icons_1.ArrowUpRegular, {
    fontSize: 12
  }),
  descending: /*#__PURE__*/React.createElement(react_icons_1.ArrowDownRegular, {
    fontSize: 12
  })
};
/**
 * Create the state required to render TableHeaderCell.
 *
 * The returned state can be modified with hooks such as useTableHeaderCellStyles_unstable,
 * before being passed to renderTableHeaderCell_unstable.
 *
 * @param props - props from this instance of TableHeaderCell
 * @param ref - reference to root HTMLElement of TableHeaderCell
 */

const useTableHeaderCell_unstable = (props, ref) => {
  var _a, _b;

  const {
    noNativeElements,
    sortable
  } = tableContext_1.useTableContext();
  const rootComponent = ((_a = props.as) !== null && _a !== void 0 ? _a : noNativeElements) ? 'div' : 'th';
  return {
    components: {
      root: rootComponent,
      button: 'button',
      sortIcon: 'span'
    },
    root: react_utilities_1.getNativeElementProps(rootComponent, {
      ref: react_utilities_1.useMergedRefs(ref, react_tabster_1.useFocusWithin()),
      role: rootComponent === 'div' ? 'columnheader' : undefined,
      'aria-sort': sortable ? (_b = props.sortDirection) !== null && _b !== void 0 ? _b : 'none' : undefined,
      ...props
    }),
    sortIcon: react_utilities_1.resolveShorthand(props.sortIcon, {
      required: !!props.sortDirection,
      defaultProps: {
        children: props.sortDirection ? sortIcons[props.sortDirection] : undefined
      }
    }),
    button: react_aria_1.useARIAButtonShorthand(props.button, {
      required: true,
      defaultProps: {
        role: 'presentation',
        tabIndex: -1,
        type: 'button',
        ...(sortable && {
          role: undefined,
          tabIndex: undefined
        })
      }
    }),
    sortable,
    noNativeElements
  };
};

exports.useTableHeaderCell_unstable = useTableHeaderCell_unstable;
