// @fluentui/react-menu/lib/components/MenuItemRadio/useMenuItemRadio.js
import * as React from 'react';
import { resolveShorthand } from '@fluentui/react-utilities';
import { Checkmark16Filled } from '@fluentui/react-icons';
import { useMenuListContext_unstable } from '../../contexts/menuListContext';
import { useMenuItem_unstable } from '../MenuItem/useMenuItem';
/**
 * Given user props, returns state and render function for a MenuItemRadio.
 */

export const useMenuItemRadio_unstable = (props, ref) => {
  const {
    name,
    value
  } = props;
  const checked = useMenuListContext_unstable(context => {
    var _a;

    const checkedItems = ((_a = context.checkedValues) === null || _a === void 0 ? void 0 : _a[name]) || [];
    return checkedItems.indexOf(value) !== -1;
  });
  const selectRadio = useMenuListContext_unstable(context => context.selectRadio);
  return { ...useMenuItem_unstable({ ...props,
      role: 'menuitemradio',
      'aria-checked': checked,
      checkmark: resolveShorthand(props.checkmark, {
        defaultProps: {
          children: /*#__PURE__*/React.createElement(Checkmark16Filled, null)
        },
        required: true
      }),
      onClick: e => {
        var _a;

        selectRadio === null || selectRadio === void 0 ? void 0 : selectRadio(e, name, value, checked);
        (_a = props.onClick) === null || _a === void 0 ? void 0 : _a.call(props, e);
      }
    }, ref),
    checked,
    name,
    value
  };
};
