// @fluentui/react-menu/lib-amd/components/MenuItemRadio/useMenuItemRadioStyles.js
define(["require", "exports", "@griffel/react", "../../selectable/index", "../MenuItem/useMenuItemStyles"], function (require, exports, react_1, index_1, useMenuItemStyles_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useMenuItemRadioStyles_unstable = exports.menuItemRadioClassNames = void 0;
    exports.menuItemRadioClassNames = {
        root: 'fui-MenuItemRadio',
        icon: 'fui-MenuItemRadio__icon',
        checkmark: 'fui-MenuItemRadio__checkmark',
        content: 'fui-MenuItemRadio__content',
        secondaryContent: 'fui-MenuItemRadio__secondaryContent',
    };
    var useMenuItemRadioStyles_unstable = function (state) {
        state.root.className = react_1.mergeClasses(exports.menuItemRadioClassNames.root, state.root.className);
        if (state.content) {
            state.content.className = react_1.mergeClasses(exports.menuItemRadioClassNames.content, state.content.className);
        }
        if (state.secondaryContent) {
            state.secondaryContent.className = react_1.mergeClasses(exports.menuItemRadioClassNames.secondaryContent, state.secondaryContent.className);
        }
        if (state.icon) {
            state.icon.className = react_1.mergeClasses(exports.menuItemRadioClassNames.icon, state.icon.className);
        }
        if (state.checkmark) {
            state.checkmark.className = react_1.mergeClasses(exports.menuItemRadioClassNames.checkmark, state.checkmark.className);
        }
        useMenuItemStyles_1.useMenuItemStyles_unstable(state);
        index_1.useCheckmarkStyles_unstable(state);
    };
    exports.useMenuItemRadioStyles_unstable = useMenuItemRadioStyles_unstable;
});
